{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"PPB_Affinity_processed.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# these columns are created for processing purposes, they will be removed later\n",
    "df[\"ligand\"] = df[\"Ligand Sequences\"].str.replace(\",\", \"\")\n",
    "df[\"receptor\"] = df[\"Receptor Sequences\"].str.replace(\",\", \"\")\n",
    "df[\"combined\"] = df[\"ligand\"] + df[\"receptor\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Source Data Set</th>\n",
       "      <th>Complex ID</th>\n",
       "      <th>PDB</th>\n",
       "      <th>Mutations</th>\n",
       "      <th>Ligand Chains</th>\n",
       "      <th>Receptor Chains</th>\n",
       "      <th>Ligand Name</th>\n",
       "      <th>Receptor Name</th>\n",
       "      <th>KD(M)</th>\n",
       "      <th>Affinity Method</th>\n",
       "      <th>...</th>\n",
       "      <th>PDB PubMed ID</th>\n",
       "      <th>PDB Release Date</th>\n",
       "      <th>Affinity PubMed ID</th>\n",
       "      <th>Affinity Release Date</th>\n",
       "      <th>Subgroup</th>\n",
       "      <th>Ligand Sequences</th>\n",
       "      <th>Receptor Sequences</th>\n",
       "      <th>ligand</th>\n",
       "      <th>receptor</th>\n",
       "      <th>combined</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1A22:A, B::PMID=7504735</td>\n",
       "      <td>1A22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>Human growth hormone</td>\n",
       "      <td>hGH binding protein</td>\n",
       "      <td>9.000000e-10</td>\n",
       "      <td>SPR</td>\n",
       "      <td>...</td>\n",
       "      <td>9571026.0</td>\n",
       "      <td>1998-04-29</td>\n",
       "      <td>7504735</td>\n",
       "      <td>1993 Dec 5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...</td>\n",
       "      <td>PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...</td>\n",
       "      <td>FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...</td>\n",
       "      <td>PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...</td>\n",
       "      <td>FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1A4Y:A, B::PMID=9050852</td>\n",
       "      <td>1A4Y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>Ribonuclease inhibitor</td>\n",
       "      <td>Angiogenin</td>\n",
       "      <td>5.000000e-16</td>\n",
       "      <td>Other</td>\n",
       "      <td>...</td>\n",
       "      <td>9311977.0</td>\n",
       "      <td>1998-10-14</td>\n",
       "      <td>9050852</td>\n",
       "      <td>1997 Mar 4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...</td>\n",
       "      <td>QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...</td>\n",
       "      <td>SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...</td>\n",
       "      <td>QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...</td>\n",
       "      <td>SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1ACB:E, I::PMID=9048543</td>\n",
       "      <td>1ACB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>I</td>\n",
       "      <td>Bovine alpha-chymotrypsin</td>\n",
       "      <td>Eglin c</td>\n",
       "      <td>1.490000e-12</td>\n",
       "      <td>IASP</td>\n",
       "      <td>...</td>\n",
       "      <td>1583684.0</td>\n",
       "      <td>1993-10-31</td>\n",
       "      <td>9048543</td>\n",
       "      <td>1997 Feb 18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...</td>\n",
       "      <td>KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...</td>\n",
       "      <td>CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...</td>\n",
       "      <td>KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...</td>\n",
       "      <td>CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1AHW:A, B, C::PMID=9480775</td>\n",
       "      <td>1AHW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A, B</td>\n",
       "      <td>C</td>\n",
       "      <td>Immunoglobulin fab 5G9</td>\n",
       "      <td>Tissue factor</td>\n",
       "      <td>3.400000e-09</td>\n",
       "      <td>IASP</td>\n",
       "      <td>...</td>\n",
       "      <td>9480775.0</td>\n",
       "      <td>1998-02-25</td>\n",
       "      <td>9480775</td>\n",
       "      <td>1998 Feb 6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...</td>\n",
       "      <td>TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...</td>\n",
       "      <td>DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...</td>\n",
       "      <td>TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...</td>\n",
       "      <td>DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1AK4:A, D::PMID=9223641</td>\n",
       "      <td>1AK4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>D</td>\n",
       "      <td>Cyclophilin A</td>\n",
       "      <td>HIV-1 capsid protein</td>\n",
       "      <td>1.200000e-05</td>\n",
       "      <td>SPR</td>\n",
       "      <td>...</td>\n",
       "      <td>8980234.0</td>\n",
       "      <td>1997-10-15</td>\n",
       "      <td>9223641</td>\n",
       "      <td>1997 Jun 27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...</td>\n",
       "      <td>PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...</td>\n",
       "      <td>MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...</td>\n",
       "      <td>PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...</td>\n",
       "      <td>MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Source Data Set                  Complex ID   PDB Mutations Ligand Chains  \\\n",
       "0     SKEMPI v2.0     1A22:A, B::PMID=7504735  1A22       NaN             A   \n",
       "1     SKEMPI v2.0     1A4Y:A, B::PMID=9050852  1A4Y       NaN             A   \n",
       "2     SKEMPI v2.0     1ACB:E, I::PMID=9048543  1ACB       NaN             E   \n",
       "3     SKEMPI v2.0  1AHW:A, B, C::PMID=9480775  1AHW       NaN          A, B   \n",
       "4     SKEMPI v2.0     1AK4:A, D::PMID=9223641  1AK4       NaN             A   \n",
       "\n",
       "  Receptor Chains                Ligand Name         Receptor Name  \\\n",
       "0               B       Human growth hormone   hGH binding protein   \n",
       "1               B     Ribonuclease inhibitor            Angiogenin   \n",
       "2               I  Bovine alpha-chymotrypsin               Eglin c   \n",
       "3               C     Immunoglobulin fab 5G9         Tissue factor   \n",
       "4               D              Cyclophilin A  HIV-1 capsid protein   \n",
       "\n",
       "          KD(M) Affinity Method  ... PDB PubMed ID PDB Release Date  \\\n",
       "0  9.000000e-10             SPR  ...     9571026.0       1998-04-29   \n",
       "1  5.000000e-16           Other  ...     9311977.0       1998-10-14   \n",
       "2  1.490000e-12            IASP  ...     1583684.0       1993-10-31   \n",
       "3  3.400000e-09            IASP  ...     9480775.0       1998-02-25   \n",
       "4  1.200000e-05             SPR  ...     8980234.0       1997-10-15   \n",
       "\n",
       "   Affinity PubMed ID  Affinity Release Date Subgroup  \\\n",
       "0             7504735             1993 Dec 5      NaN   \n",
       "1             9050852             1997 Mar 4      NaN   \n",
       "2             9048543            1997 Feb 18      NaN   \n",
       "3             9480775             1998 Feb 6      NaN   \n",
       "4             9223641            1997 Jun 27      NaN   \n",
       "\n",
       "                                    Ligand Sequences  \\\n",
       "0  FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...   \n",
       "1  SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...   \n",
       "2  CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...   \n",
       "3  DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...   \n",
       "4  MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...   \n",
       "\n",
       "                                  Receptor Sequences  \\\n",
       "0  PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...   \n",
       "1  QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...   \n",
       "2  KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...   \n",
       "3  TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...   \n",
       "4  PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...   \n",
       "\n",
       "                                              ligand  \\\n",
       "0  FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...   \n",
       "1  SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...   \n",
       "2  CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...   \n",
       "3  DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...   \n",
       "4  MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...   \n",
       "\n",
       "                                            receptor  \\\n",
       "0  PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...   \n",
       "1  QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...   \n",
       "2  KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...   \n",
       "3  TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...   \n",
       "4  PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...   \n",
       "\n",
       "                                            combined  \n",
       "0  FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...  \n",
       "1  SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...  \n",
       "2  CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...  \n",
       "3  DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...  \n",
       "4  MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    12048.000000\n",
       "mean         7.541897\n",
       "std          2.144316\n",
       "min          1.318759\n",
       "25%          5.872895\n",
       "50%          7.602060\n",
       "75%          9.020015\n",
       "max         15.698970\n",
       "Name: pKd, dtype: float64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# convert KD to pKD\n",
    "df[\"pKd\"] = -np.log10(df[\"KD(M)\"])\n",
    "df[\"pKd\"].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dropped 2118 rows with ligand or receptor chain(s) shorter than 20\n",
      "Remaining rows: 9930\n"
     ]
    }
   ],
   "source": [
    "# filter out rows where any ligand or receptor chain is shorter than 20\n",
    "# amino acids\n",
    "minlen = 20\n",
    "f = lambda seqs: any([len(seq) < minlen for seq in seqs])\n",
    "min_len_mask = df[\"Ligand Sequences\"].str.split(\",\").apply(f) | df[\n",
    "    \"Receptor Sequences\"\n",
    "].str.split(\",\").apply(f)\n",
    "df = df[~min_len_mask].reset_index(drop=True)\n",
    "print(\n",
    "    f\"Dropped {min_len_mask.sum()} rows with ligand or receptor chain(s) \"\n",
    "    f\"shorter than {minlen}\"\n",
    ")\n",
    "print(f\"Remaining rows: {len(df)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Affinity Method\n",
       "NaN        3430\n",
       "SPR        2431\n",
       "FL         1090\n",
       "Other       786\n",
       "ITC         555\n",
       "IASP        476\n",
       "SP          336\n",
       "RA          320\n",
       "ELISA       172\n",
       "IARA        125\n",
       "BLI         104\n",
       "Unknown      96\n",
       "IAGE          9\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check affinity method distribution in the dataset, we will use the Affinity\n",
    "# Method for handling duplicated sequences\n",
    "df[\"Affinity Method\"].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# replace nan method with \"Unknown\"\n",
    "df[\"Affinity Method\"] = df[\"Affinity Method\"].fillna(\"Unknown\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_duplicates_by_method(\n",
    "    df,\n",
    "    seq_col=\"combined\",\n",
    "    kd_col=\"KD(M)\",\n",
    "    pkd_col=\"pKd\",\n",
    "    method_col=\"Affinity Method\",\n",
    "):\n",
    "    \"\"\"\n",
    "    Group by 'seq_col' (the combined sequence) and resolve duplicates by method priority:\n",
    "      1. Within each group (all rows sharing the same 'combined' sequence), find the highest-priority method present.\n",
    "      2. Average the KD(M) and pKd values for all rows of that method.\n",
    "      3. Keep the *first row* from that subset for all other columns.\n",
    "    \"\"\"\n",
    "\n",
    "    # Priority list of methods to choose from\n",
    "    method_priority = [\n",
    "        \"SPR\",\n",
    "        \"ITC\",\n",
    "        \"BLI\",\n",
    "        \"RA\",\n",
    "        \"FL\",\n",
    "        \"SP\",\n",
    "        \"ELISA\",\n",
    "        \"IASP\",\n",
    "        \"IARA\",\n",
    "        \"IAGE\",\n",
    "        \"Other\",\n",
    "        \"Unknown\",\n",
    "    ]\n",
    "\n",
    "    def choose_best_method(group):\n",
    "        \"\"\"\n",
    "        For a subset of df with the same 'combined' sequence:\n",
    "          - Iterate over the 'method_priority' list in order.\n",
    "          - If the group contains that method, average KD(M) across those rows.\n",
    "          - Return the *first row* of that subset for all other columns.\n",
    "        \"\"\"\n",
    "        methods_in_group = group[method_col].unique()\n",
    "\n",
    "        for meth in method_priority:\n",
    "            if meth in methods_in_group:\n",
    "                # Get only the rows of the current top-priority method\n",
    "                subset = group[group[method_col] == meth]\n",
    "\n",
    "                # Average KD(M) and pKD for this subset\n",
    "                avg_kd = subset[kd_col].mean()\n",
    "                avg_pkd = subset[pkd_col].mean()\n",
    "\n",
    "                # Take the *first row* from THIS subset\n",
    "                # (the subset that contributes to the average)\n",
    "                first_row = subset.iloc[0].copy()\n",
    "\n",
    "                # Update the KD(M) and method to reflect the average and the chosen method\n",
    "                first_row[kd_col] = avg_kd\n",
    "                first_row[pkd_col] = avg_pkd\n",
    "                first_row[method_col] = meth\n",
    "\n",
    "                return first_row\n",
    "        raise ValueError(\n",
    "            \"None of the methods in the priority list were found in the group!\"\n",
    "        )\n",
    "\n",
    "    # Apply the 'choose_best_method' to each group of duplicated sequences\n",
    "    result_df = df.groupby(seq_col, as_index=False).apply(\n",
    "        choose_best_method, include_groups=False\n",
    "    )\n",
    "    return result_df.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8836"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = filter_duplicates_by_method(df)\n",
    "print(f\"Remaining rows after filtering duplicates: {len(df)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>combined</th>\n",
       "      <th>Source Data Set</th>\n",
       "      <th>Complex ID</th>\n",
       "      <th>PDB</th>\n",
       "      <th>Mutations</th>\n",
       "      <th>Ligand Chains</th>\n",
       "      <th>Receptor Chains</th>\n",
       "      <th>Ligand Name</th>\n",
       "      <th>Receptor Name</th>\n",
       "      <th>KD(M)</th>\n",
       "      <th>...</th>\n",
       "      <th>PDB Release Date</th>\n",
       "      <th>Affinity PubMed ID</th>\n",
       "      <th>Affinity Release Date</th>\n",
       "      <th>Subgroup</th>\n",
       "      <th>Ligand Sequences</th>\n",
       "      <th>Receptor Sequences</th>\n",
       "      <th>ligand</th>\n",
       "      <th>receptor</th>\n",
       "      <th>pKd</th>\n",
       "      <th>fasta_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>2WQZ:B, C::PMID=18093521</td>\n",
       "      <td>2WQZ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B</td>\n",
       "      <td>C</td>\n",
       "      <td>Neuroligin4, X-Linked</td>\n",
       "      <td>Neurexin-1-beta</td>\n",
       "      <td>1.320000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>2009-09-08</td>\n",
       "      <td>18093521</td>\n",
       "      <td>2007 Dec 20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...</td>\n",
       "      <td>HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...</td>\n",
       "      <td>AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...</td>\n",
       "      <td>HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...</td>\n",
       "      <td>6.879426</td>\n",
       "      <td>2WQZ_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>5IIA:A, B, C, D::PMID=28622512</td>\n",
       "      <td>5IIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B, D</td>\n",
       "      <td>A, C</td>\n",
       "      <td>sperm lysinR</td>\n",
       "      <td>red abalone egg VERL repeat 3 (VR3)</td>\n",
       "      <td>1.810000e-09</td>\n",
       "      <td>...</td>\n",
       "      <td>2017-06-14</td>\n",
       "      <td>28622512</td>\n",
       "      <td>2017 Jun 15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...</td>\n",
       "      <td>FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...</td>\n",
       "      <td>AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...</td>\n",
       "      <td>FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...</td>\n",
       "      <td>8.742321</td>\n",
       "      <td>5IIA_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>4ZI2:A, C::PMID=26455799</td>\n",
       "      <td>4ZI2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>BART-like domain of BARTL1/CCDC104 1-113</td>\n",
       "      <td>Arl3FL-GppNHp</td>\n",
       "      <td>4.200000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>2015-11-11</td>\n",
       "      <td>26455799</td>\n",
       "      <td>2015 Nov 3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...</td>\n",
       "      <td>LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...</td>\n",
       "      <td>AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...</td>\n",
       "      <td>LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...</td>\n",
       "      <td>6.376751</td>\n",
       "      <td>4ZI2_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>4UF1:A, B::PMID=26249341</td>\n",
       "      <td>4UF1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>Bak BH3</td>\n",
       "      <td>Deerpox virus DPV022</td>\n",
       "      <td>6.980000e-06</td>\n",
       "      <td>...</td>\n",
       "      <td>2015-08-05</td>\n",
       "      <td>26249341</td>\n",
       "      <td>2015 Aug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...</td>\n",
       "      <td>STMGQVGRQLAIIGDDINRRYD</td>\n",
       "      <td>AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...</td>\n",
       "      <td>STMGQVGRQLAIIGDDINRRYD</td>\n",
       "      <td>5.156145</td>\n",
       "      <td>4UF1_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>6SAZ:A, B::PMID=31604990</td>\n",
       "      <td>6SAZ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>Crayfish Astacin</td>\n",
       "      <td>Cleaved human fetuin-b</td>\n",
       "      <td>1.400000e-10</td>\n",
       "      <td>...</td>\n",
       "      <td>2019-10-23</td>\n",
       "      <td>31604990</td>\n",
       "      <td>2019 Oct 11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...</td>\n",
       "      <td>ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...</td>\n",
       "      <td>AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...</td>\n",
       "      <td>ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...</td>\n",
       "      <td>9.853872</td>\n",
       "      <td>6SAZ_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            combined Source Data Set  \\\n",
       "0  AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...   PDBbind v2020   \n",
       "1  AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...   PDBbind v2020   \n",
       "2  AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...   PDBbind v2020   \n",
       "3  AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...   PDBbind v2020   \n",
       "4  AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...   PDBbind v2020   \n",
       "\n",
       "                       Complex ID   PDB Mutations Ligand Chains  \\\n",
       "0        2WQZ:B, C::PMID=18093521  2WQZ       NaN             B   \n",
       "1  5IIA:A, B, C, D::PMID=28622512  5IIA       NaN          B, D   \n",
       "2        4ZI2:A, C::PMID=26455799  4ZI2       NaN             C   \n",
       "3        4UF1:A, B::PMID=26249341  4UF1       NaN             A   \n",
       "4        6SAZ:A, B::PMID=31604990  6SAZ       NaN             A   \n",
       "\n",
       "  Receptor Chains                               Ligand Name  \\\n",
       "0               C                     Neuroligin4, X-Linked   \n",
       "1            A, C                              sperm lysinR   \n",
       "2               A  BART-like domain of BARTL1/CCDC104 1-113   \n",
       "3               B                                   Bak BH3   \n",
       "4               B                          Crayfish Astacin   \n",
       "\n",
       "                         Receptor Name         KD(M)  ... PDB Release Date  \\\n",
       "0                      Neurexin-1-beta  1.320000e-07  ...       2009-09-08   \n",
       "1  red abalone egg VERL repeat 3 (VR3)  1.810000e-09  ...       2017-06-14   \n",
       "2                        Arl3FL-GppNHp  4.200000e-07  ...       2015-11-11   \n",
       "3                 Deerpox virus DPV022  6.980000e-06  ...       2015-08-05   \n",
       "4               Cleaved human fetuin-b  1.400000e-10  ...       2019-10-23   \n",
       "\n",
       "  Affinity PubMed ID Affinity Release Date  Subgroup  \\\n",
       "0           18093521           2007 Dec 20       NaN   \n",
       "1           28622512           2017 Jun 15       NaN   \n",
       "2           26455799            2015 Nov 3       NaN   \n",
       "3           26249341              2015 Aug       NaN   \n",
       "4           31604990           2019 Oct 11       NaN   \n",
       "\n",
       "                                    Ligand Sequences  \\\n",
       "0  AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...   \n",
       "1  AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...   \n",
       "2  AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...   \n",
       "3  AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...   \n",
       "4  AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...   \n",
       "\n",
       "                                  Receptor Sequences  \\\n",
       "0  HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...   \n",
       "1  FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...   \n",
       "2  LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...   \n",
       "3                             STMGQVGRQLAIIGDDINRRYD   \n",
       "4  ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...   \n",
       "\n",
       "                                              ligand  \\\n",
       "0  AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...   \n",
       "1  AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...   \n",
       "2  AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...   \n",
       "3  AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...   \n",
       "4  AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...   \n",
       "\n",
       "                                            receptor       pKd fasta_id  \n",
       "0  HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...  6.879426   2WQZ_0  \n",
       "1  FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...  8.742321   5IIA_1  \n",
       "2  LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...  6.376751   4ZI2_2  \n",
       "3                             STMGQVGRQLAIIGDDINRRYD  5.156145   4UF1_3  \n",
       "4  ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...  9.853872   6SAZ_4  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fasta_id will be a unique identifier for each row to be used in fasta files\n",
    "# instead of complex id as it may contain spaces\n",
    "df[\"fasta_id\"] = df[\"PDB\"] + \"_\" + df.index.astype(str)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.makedirs(\"data_files\", exist_ok=True)\n",
    "with open(\"data_files/all_seqs.fasta\", \"w\") as f:\n",
    "    for i, row in df.iterrows():\n",
    "        fasta_id = row[\"fasta_id\"]\n",
    "        sequence = row[\"combined\"]\n",
    "        f.write(f\">{fasta_id}\\n{sequence}\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "we cluster all the sequnces into as many clusters as it takes such that for each cluster the minimum sequence identity between the representative of the cluster and any other sequence in the cluster is at least 10%. We use coverage of the smaller sequence with coverage of 20%. Such loose parameters are used to prevent ending up with thousands of clusters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data_files/all_seqs.fasta data_files/cluster tmp --min-seq-id 0.1 --cov-mode 5 -c 0.2 -s 10.0 --remove-tmp-files --max-iterations 50000 --cluster-reassign --cluster-steps 9 \n",
      "\n",
      "MMseqs Version:                     \tf5f780acd64482cd59b46eae0a107f763cd17b4d\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t10\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.2\n",
      "Coverage mode                       \t5\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t8\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.1\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t50000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t9\n",
      "Cluster reassign                    \ttrue\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Database type                       \t0\n",
      "Shuffle input database              \ttrue\n",
      "Createdb mode                       \t1\n",
      "Write lookup file                   \t0\n",
      "Offset of numeric ids               \t0\n",
      "\n",
      "createdb data_files/all_seqs.fasta tmp/9671761839411747810/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mShuffle database cannot be combined with --createdb-mode 0\n",
      "\u001b[39m\u001b[33mWe recompute with --shuffle 0\n",
      "\u001b[39mConverting sequences\n",
      "[8788] 0s 4ms\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\n",
      "Time for merging to input: 0h 0m 0s 0ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 4ms\n",
      "Create directory tmp/9671761839411747810/clu_tmp\n",
      "cluster tmp/9671761839411747810/input tmp/9671761839411747810/clu tmp/9671761839411747810/clu_tmp -s 10 --max-seqs 20 -c 0.2 --cov-mode 5 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.1 --max-iterations 50000 --cluster-steps 9 --cluster-reassign 1 --remove-tmp-files 0 \n",
      "\n",
      "Set cluster mode SET COVER\n",
      "linclust tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 5 -k 0 -c 0.2 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 5 -k 0 -c 0.2 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 8836 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 8.84K 0s 22ms     \n",
      "Sort kmer 0h 0m 0s 27ms\n",
      "Sort by rep. sequence 0h 0m 0s 20ms\n",
      "Time for fill: 0h 0m 0s 1ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 85ms\n",
      "rescorediagonal tmp/9671761839411747810/input tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 5 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 8.84K 0s 7ms                     ] 14.56% 1.29K eta 0s       \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 2ms==================>] 99.77% 8.82K eta 0s       \n",
      "Time for processing: 0h 0m 0s 62ms\n",
      "clust tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref_rescore1 tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pre_clust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 8.84K 0s 8ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 21264 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 8.84K 0s 35ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 8.84K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 74ms\n",
      "Total time: 0h 0m 0s 82ms\n",
      "\n",
      "Size of the sequence database: 8836\n",
      "Size of the alignment database: 8836\n",
      "Number of clusters: 1817\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 96ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/order_redundancy tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 4ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/order_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "filterdb tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref_filter1 tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref_filter2 --filter-file tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/order_redundancy --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 1.82K 0s 6ms      \n",
      "Time for merging to pref_filter2: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 12ms\n",
      "rescorediagonal tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/input_step_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/input_step_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref_filter2 tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.2 -a 0 --cov-mode 5 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mCan not find any score per column for coverage 0.200000 and sequence identity 0.100000. No hit will be filtered.\n",
      "[=================================================================] 100.00% 1.82K 0s 3ms      \n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms===========>       ] 88.27% 1.60K eta 0s       \n",
      "Time for processing: 0h 0m 0s 11ms\n",
      "align tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/input_step_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/input_step_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pref_rescore2 tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1817 type: Aminoacid\n",
      "Target database size: 1817 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 1.82K 1s 72ms     \n",
      "Time for merging to aln: 0h 0m 0s 1ms\n",
      "5075 alignments calculated\n",
      "5075 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "2.793066 hits per query sequence\n",
      "Time for processing: 0h 0m 1s 76ms\n",
      "clust tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/input_step_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/aln tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/clust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 1.82K 0s 3ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3258 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 1.82K 0s 2ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 1.82K 0s 3ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 10ms\n",
      "Total time: 0h 0m 0s 26ms\n",
      "\n",
      "Size of the sequence database: 1817\n",
      "Size of the alignment database: 1817\n",
      "Number of clusters: 915\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 28ms\n",
      "mergeclusters tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/pre_clust tmp/9671761839411747810/clu_tmp/8997417895626108443/linclust/18166716216592373531/clust --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 1.82K 0s 6ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 915 0s 30ms     \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 8.84K 0s 53ms     \n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 66ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_redundancy tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 915 type: Aminoacid\n",
      "Estimated memory consumption: 981M\n",
      "Target database size: 915 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 915 0s 17ms     \n",
      "Index table: Masked residues: 8635\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 915 0s 9ms      \n",
      "Index statistics\n",
      "Entries:          175630\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002744\n",
      "Top 10 k-mers\n",
      "    YWQTTV\t39\n",
      "    SGPRGS\t31\n",
      "    DYGGVT\t31\n",
      "    WYQPAP\t25\n",
      "    SVCLFY\t25\n",
      "    SGGGFT\t23\n",
      "    WNGLGV\t23\n",
      "    RQPKEW\t23\n",
      "    FPVQGL\t21\n",
      "    KGSFAP\t21\n",
      "Time for index table init: 0h 0m 0s 396ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 915\n",
      "Target db start 1 to 915\n",
      "[=================================================================] 100.00% 915 0s 18ms     \n",
      "[===============================================================> ] 98.36% 900 eta 0s       \n",
      "2.366144 k-mers per position\n",
      "330 DB matches per sequence\n",
      "0 overflows\n",
      "3 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 1s 34ms\n",
      "align tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step0 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 915 type: Aminoacid\n",
      "Target database size: 915 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 915 0s 187ms    \n",
      "Time for merging to aln_step0: 0h 0m 0s 0ms\n",
      "3058 alignments calculated\n",
      "3049 sequence pairs passed the thresholds (0.997057 of overall calculated)\n",
      "3.332240 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 204ms\n",
      "clust tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step0 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step0 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 915 0s 9ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 444 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 915 0s 18ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 915 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 114ms\n",
      "Total time: 0h 0m 0s 166ms\n",
      "\n",
      "Size of the sequence database: 915\n",
      "Size of the alignment database: 915\n",
      "Number of clusters: 672\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 183ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step0 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step1 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step1 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2.125 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 672 type: Aminoacid\n",
      "Estimated memory consumption: 980M\n",
      "Target database size: 672 type: Aminoacid\n",
      "Index table k-mer threshold: 144 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 672 0s 16ms     \n",
      "Index table: Masked residues: 6559\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 672 0s 83ms     \n",
      "Index statistics\n",
      "Entries:          206852\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.003232\n",
      "Top 10 k-mers\n",
      "    YWQTTV\t4\n",
      "    PSSNAE\t3\n",
      "    AFEKDE\t3\n",
      "    RKFDTG\t3\n",
      "    ERKLEI\t3\n",
      "    IHDKNI\t3\n",
      "    QLQGVK\t3\n",
      "    WVQPGL\t3\n",
      "    VQGEKP\t3\n",
      "    EDAYAR\t3\n",
      "Time for index table init: 0h 0m 0s 546ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 144\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 672\n",
      "Target db start 1 to 672\n",
      "[=================================================================] 100.00% 672 0s 51ms     \n",
      "\n",
      "10.042144 k-mers per position\n",
      "349 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 380ms\n",
      "align tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step1 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step1 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step1 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 672 type: Aminoacid\n",
      "Target database size: 672 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 672 0s 21ms     \n",
      "Time for merging to aln_step1: 0h 0m 0s 0ms\n",
      "820 alignments calculated\n",
      "788 sequence pairs passed the thresholds (0.960976 of overall calculated)\n",
      "1.172619 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 30ms\n",
      "clust tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step1 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step1 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step1 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 672 0s 17ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 20 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 672 0s 13ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 672 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 103ms\n",
      "Total time: 0h 0m 0s 141ms\n",
      "\n",
      "Size of the sequence database: 672\n",
      "Size of the alignment database: 672\n",
      "Number of clusters: 615\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 145ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step1 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step1 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step2 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step2 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3.25 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 615 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 615 type: Aminoacid\n",
      "Index table k-mer threshold: 134 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 615 0s 14ms     \n",
      "Index table: Masked residues: 6082\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 615 0s 24ms     \n",
      "Index statistics\n",
      "Entries:          215454\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.003366\n",
      "Top 10 k-mers\n",
      "    DPQALA\t3\n",
      "    VRLAAD\t3\n",
      "    PSSNAE\t3\n",
      "    AFEKDE\t3\n",
      "    RKFDTG\t3\n",
      "    IHDKNI\t3\n",
      "    QSNLAK\t3\n",
      "    ANLEKL\t3\n",
      "    NLAALN\t3\n",
      "    AEKLAQ\t3\n",
      "Time for index table init: 0h 0m 0s 539ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 134\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 615\n",
      "Target db start 1 to 615\n",
      "[=================================================================] 100.00% 615 0s 191ms    \n",
      "\n",
      "32.109534 k-mers per position\n",
      "454 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 585ms\n",
      "align tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step2 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step2 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step2 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 615 type: Aminoacid\n",
      "Target database size: 615 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 615 0s 68ms     \n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\n",
      "766 alignments calculated\n",
      "672 sequence pairs passed the thresholds (0.877285 of overall calculated)\n",
      "1.092683 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 110ms\n",
      "clust tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step2 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step2 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step2 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 615 0s 18ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 25 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 615 0s 20ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 615 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 170ms\n",
      "Total time: 0h 0m 0s 236ms\n",
      "\n",
      "Size of the sequence database: 615\n",
      "Size of the alignment database: 615\n",
      "Number of clusters: 579\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 266ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step2 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step2 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step3 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step3: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step3 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step3 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 4.375 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 579 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 579 type: Aminoacid\n",
      "Index table k-mer threshold: 124 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 579 0s 70ms     \n",
      "Index table: Masked residues: 4926\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 579 0s 22ms     \n",
      "Index statistics\n",
      "Entries:          212544\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.003321\n",
      "Top 10 k-mers\n",
      "    AEGLAA\t3\n",
      "    VRLAAD\t3\n",
      "    ELLLLD\t3\n",
      "    AFEKDE\t3\n",
      "    RKFDTG\t3\n",
      "    ELKLIL\t3\n",
      "    GAGLLS\t3\n",
      "    LSLGSV\t3\n",
      "    KDFLAA\t2\n",
      "    GGILAA\t2\n",
      "Time for index table init: 0h 0m 0s 548ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 124\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 579\n",
      "Target db start 1 to 579\n",
      "[=================================================================] 100.00% 579 0s 264ms    \n",
      "[================================================================>] 98.79% 572 eta 0s       \n",
      "98.554084 k-mers per position\n",
      "576 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step3: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 1s 694ms\n",
      "align tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step3 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step3 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step3 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 579 type: Aminoacid\n",
      "Target database size: 579 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 579 0s 33ms     \n",
      "Time for merging to aln_step3: 0h 0m 0s 0ms\n",
      "976 alignments calculated\n",
      "631 sequence pairs passed the thresholds (0.646516 of overall calculated)\n",
      "1.089810 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 47ms\n",
      "clust tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step3 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step3 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step3 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 579 0s 33ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 14 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 579 0s 12ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 579 0s 2ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 123ms\n",
      "Total time: 0h 0m 0s 158ms\n",
      "\n",
      "Size of the sequence database: 579\n",
      "Size of the alignment database: 579\n",
      "Number of clusters: 549\n",
      "\n",
      "Writing results 0h 0m 0s 3ms\n",
      "Time for merging to clu_step3: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 163ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step3 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step3 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step4 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step4: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step4 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step4 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step4 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5.5 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 549 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 549 type: Aminoacid\n",
      "Index table k-mer threshold: 114 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 549 0s 22ms     \n",
      "Index table: Masked residues: 4670\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 549 0s 21ms     \n",
      "Index statistics\n",
      "Entries:          198688\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.003105\n",
      "Top 10 k-mers\n",
      "    AEGLAA\t3\n",
      "    ELLLLD\t3\n",
      "    AFEKDE\t3\n",
      "    ELKLIL\t3\n",
      "    GAGLLS\t3\n",
      "    LSLGSV\t3\n",
      "    GTVGAA\t2\n",
      "    KDFLAA\t2\n",
      "    GGILAA\t2\n",
      "    WASLAA\t2\n",
      "Time for index table init: 0h 0m 0s 505ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 114\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 549\n",
      "Target db start 1 to 549\n",
      "[=================================================================] 100.00% 549 0s 561ms    \n",
      "\n",
      "285.546986 k-mers per position\n",
      "813 DB matches per sequence\n",
      "0 overflows\n",
      "3 sequences passed prefiltering per query sequence\n",
      "3 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step4: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 1s 951ms\n",
      "align tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step4 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step4 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step4 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step4 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 549 type: Aminoacid\n",
      "Target database size: 549 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 549 0s 45ms     \n",
      "Time for merging to aln_step4: 0h 0m 0s 0ms\n",
      "1800 alignments calculated\n",
      "613 sequence pairs passed the thresholds (0.340556 of overall calculated)\n",
      "1.116576 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 57ms\n",
      "clust tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step4 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step4 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step4 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 549 0s 10ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 20 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 549 0s 4ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 549 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 46ms\n",
      "Total time: 0h 0m 0s 67ms\n",
      "\n",
      "Size of the sequence database: 549\n",
      "Size of the alignment database: 549\n",
      "Number of clusters: 510\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step4: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 68ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step4 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step4 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step5 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step5: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step5 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step5 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step5 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 6.625 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 510 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 510 type: Aminoacid\n",
      "Index table k-mer threshold: 104 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 510 0s 15ms     \n",
      "Index table: Masked residues: 3993\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 510 0s 51ms     \n",
      "Index statistics\n",
      "Entries:          183060\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002860\n",
      "Top 10 k-mers\n",
      "    ELLLLD\t3\n",
      "    ELKLIL\t3\n",
      "    GAGLLS\t3\n",
      "    LSLGSV\t3\n",
      "    GSPGAA\t2\n",
      "    GTVGAA\t2\n",
      "    AEGLAA\t2\n",
      "    WASLAA\t2\n",
      "    SSEQAA\t2\n",
      "    LKNQAA\t2\n",
      "Time for index table init: 0h 0m 0s 515ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 104\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 510\n",
      "Target db start 1 to 510\n",
      "[=================================================================] 100.00% 510 1s 463ms    \n",
      "\n",
      "793.550438 k-mers per position\n",
      "1434 DB matches per sequence\n",
      "0 overflows\n",
      "9 sequences passed prefiltering per query sequence\n",
      "8 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step5: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 2s 733ms\n",
      "align tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step5 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step5 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step5 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step5 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 510 type: Aminoacid\n",
      "Target database size: 510 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 510 0s 111ms    \n",
      "Time for merging to aln_step5: 0h 0m 0s 3ms\n",
      "4383 alignments calculated\n",
      "560 sequence pairs passed the thresholds (0.127766 of overall calculated)\n",
      "1.098039 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 125ms\n",
      "clust tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step5 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step5 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step5 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 510 0s 8ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 16 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 510 0s 19ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 510 0s 20ms     \n",
      "\n",
      "Time for read in: 0h 0m 0s 120ms\n",
      "Total time: 0h 0m 0s 147ms\n",
      "\n",
      "Size of the sequence database: 510\n",
      "Size of the alignment database: 510\n",
      "Number of clusters: 478\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step5: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 149ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step5 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step5 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step6 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step6: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step6 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step6 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step6 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 7.75 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 478 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 478 type: Aminoacid\n",
      "Index table k-mer threshold: 94 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 478 0s 23ms     \n",
      "Index table: Masked residues: 3455\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 478 0s 20ms     \n",
      "Index statistics\n",
      "Entries:          171085\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002673\n",
      "Top 10 k-mers\n",
      "    ELLLLD\t3\n",
      "    ELKLIL\t3\n",
      "    GAGLLS\t3\n",
      "    LSLGSV\t3\n",
      "    AEGLAA\t2\n",
      "    WASLAA\t2\n",
      "    SSEQAA\t2\n",
      "    LKNQAA\t2\n",
      "    IQLRAA\t2\n",
      "    TLGSAA\t2\n",
      "Time for index table init: 0h 0m 0s 548ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 94\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 478\n",
      "Target db start 1 to 478\n",
      "[=================================================================] 100.00% 478 3s 8ms      \n",
      "\n",
      "2106.538380 k-mers per position\n",
      "3049 DB matches per sequence\n",
      "0 overflows\n",
      "17 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step6: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 4s 771ms\n",
      "align tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step6 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step6 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step6 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step6 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 478 type: Aminoacid\n",
      "Target database size: 478 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 478 0s 156ms    \n",
      "Time for merging to aln_step6: 0h 0m 0s 1ms\n",
      "7398 alignments calculated\n",
      "496 sequence pairs passed the thresholds (0.067045 of overall calculated)\n",
      "1.037657 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 176ms\n",
      "clust tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step6 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step6 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step6 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 478 0s 28ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 8 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 478 0s 16ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 478 0s 1ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 157ms\n",
      "Total time: 0h 0m 0s 185ms\n",
      "\n",
      "Size of the sequence database: 478\n",
      "Size of the alignment database: 478\n",
      "Number of clusters: 465\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step6: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 215ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step6 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step6 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step7 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step7: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step7 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step7 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step7 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 8.875 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 465 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 465 type: Aminoacid\n",
      "Index table k-mer threshold: 84 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 465 0s 14ms     \n",
      "Index table: Masked residues: 3135\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 465 0s 54ms     \n",
      "Index statistics\n",
      "Entries:          164220\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002566\n",
      "Top 10 k-mers\n",
      "    ELKLIL\t3\n",
      "    GAGLLS\t3\n",
      "    LSLGSV\t3\n",
      "    AEGLAA\t2\n",
      "    SSEQAA\t2\n",
      "    LKNQAA\t2\n",
      "    IQLRAA\t2\n",
      "    TLGSAA\t2\n",
      "    AELDDA\t2\n",
      "    NKESDA\t2\n",
      "Time for index table init: 0h 0m 0s 571ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 84\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 465\n",
      "Target db start 1 to 465\n",
      "[=================================================================] 100.00% 465 7s 873ms    \n",
      "\n",
      "5348.300264 k-mers per position\n",
      "7087 DB matches per sequence\n",
      "0 overflows\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step7: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 9s 400ms\n",
      "align tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step7 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step7 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step7 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step7 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 465 type: Aminoacid\n",
      "Target database size: 465 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 465 0s 87ms     \n",
      "Time for merging to aln_step7: 0h 0m 0s 0ms\n",
      "8043 alignments calculated\n",
      "475 sequence pairs passed the thresholds (0.059058 of overall calculated)\n",
      "1.021505 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 103ms\n",
      "clust tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step7 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step7 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step7 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 465 0s 19ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 2 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 465 0s 15ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 465 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 110ms\n",
      "Total time: 0h 0m 0s 163ms\n",
      "\n",
      "Size of the sequence database: 465\n",
      "Size of the alignment database: 465\n",
      "Number of clusters: 459\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step7: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 165ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step7 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step7 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step8 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step8: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 5ms\n",
      "prefilter tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step8 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step8 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step8 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 10 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 459 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 459 type: Aminoacid\n",
      "Index table k-mer threshold: 74 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 459 0s 16ms     \n",
      "Index table: Masked residues: 3135\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 459 0s 19ms     \n",
      "Index statistics\n",
      "Entries:          160991\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002515\n",
      "Top 10 k-mers\n",
      "    ELKLIL\t3\n",
      "    GAGLLS\t3\n",
      "    LSLGSV\t3\n",
      "    SSEQAA\t2\n",
      "    LKNQAA\t2\n",
      "    IQLRAA\t2\n",
      "    TLGSAA\t2\n",
      "    AELDDA\t2\n",
      "    NKESDA\t2\n",
      "    TSIKEA\t2\n",
      "Time for index table init: 0h 0m 0s 525ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 74\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 459\n",
      "Target db start 1 to 459\n",
      "[=================================================================] 100.00% 459 17s 509ms    \n",
      "\n",
      "13023.612277 k-mers per position\n",
      "16927 DB matches per sequence\n",
      "0 overflows\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step8: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 19s 448ms\n",
      "align tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step8 tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step8 tmp/9671761839411747810/clu_tmp/8997417895626108443/pref_step8 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step8 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 459 type: Aminoacid\n",
      "Target database size: 459 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 459 0s 72ms     \n",
      "Time for merging to aln_step8: 0h 0m 0s 0ms\n",
      "8083 alignments calculated\n",
      "464 sequence pairs passed the thresholds (0.057404 of overall calculated)\n",
      "1.010893 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 79ms\n",
      "clust tmp/9671761839411747810/clu_tmp/8997417895626108443/input_step8 tmp/9671761839411747810/clu_tmp/8997417895626108443/aln_step8 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step8 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 459 0s 4ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 459 0s 8ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 459 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 41ms\n",
      "Total time: 0h 0m 0s 62ms\n",
      "\n",
      "Size of the sequence database: 459\n",
      "Size of the alignment database: 459\n",
      "Number of clusters: 455\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step8: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 64ms\n",
      "mergeclusters tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/clu tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_redundancy tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step0 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step1 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step2 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step3 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step4 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step5 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step6 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step7 tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_step8 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 915 0s 16ms     \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 672 0s 48ms     eta 0s       \n",
      "Clustering step 3\n",
      "[=================================================================] 100.00% 615 0s 71ms     \n",
      "Clustering step 4\n",
      "[=================================================================] 100.00% 579 0s 83ms     \n",
      "Clustering step 5\n",
      "[=================================================================] 100.00% 549 0s 94ms     \n",
      "Clustering step 6\n",
      "[=================================================================] 100.00% 510 0s 111ms    \n",
      "Clustering step 7\n",
      "[=================================================================] 100.00% 478 0s 123ms    \n",
      "Clustering step 8\n",
      "[=================================================================] 100.00% 465 0s 138ms    \n",
      "Clustering step 9\n",
      "[=================================================================] 100.00% 459 0s 149ms    \n",
      "Clustering step 10\n",
      "[=================================================================] 100.00% 455 0s 161ms    \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 8.84K 0s 170ms    \n",
      "Time for merging to clu: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 178ms\n",
      "align tmp/9671761839411747810/input tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/clu tmp/9671761839411747810/clu_tmp/8997417895626108443/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 8836 type: Aminoacid\n",
      "Target database size: 8836 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 455 0s 993ms    \n",
      "Time for merging to aln: 0h 0m 0s 1ms\n",
      "8764 alignments calculated\n",
      "5331 sequence pairs passed the thresholds (0.608284 of overall calculated)\n",
      "11.716483 hits per query sequence\n",
      "Time for processing: 0h 0m 1s 5ms\n",
      "subtractdbs tmp/9671761839411747810/clu_tmp/8997417895626108443/clu tmp/9671761839411747810/clu_tmp/8997417895626108443/aln tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_not_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "subtractdbs tmp/9671761839411747810/clu_tmp/8997417895626108443/clu tmp/9671761839411747810/clu_tmp/8997417895626108443/aln tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_not_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Remove tmp/9671761839411747810/clu_tmp/8997417895626108443/aln ids from tmp/9671761839411747810/clu_tmp/8997417895626108443/clu\n",
      "[=================================================================] 100.00% 455 0s 13ms     \n",
      "Time for merging to clu_not_accepted: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 25ms\n",
      "swapdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_not_accepted tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_not_accepted_swap --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 455 0s 9ms      \n",
      "Computing offsets.\n",
      "[=================================================================] 100.00% 455 0s 8ms      \n",
      "\n",
      "Reading results.\n",
      "[=================================================================] 100.00% 455 0s 9ms      \n",
      "\n",
      "Output database: tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_not_accepted_swap\n",
      "[=================================================================] 100.00% 8.84K 0s 7ms      \n",
      "\n",
      "Time for merging to clu_not_accepted_swap: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 76ms\n",
      "subtractdbs tmp/9671761839411747810/clu_tmp/8997417895626108443/clu tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_not_accepted tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "subtractdbs tmp/9671761839411747810/clu_tmp/8997417895626108443/clu tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_not_accepted tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Remove tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_not_accepted ids from tmp/9671761839411747810/clu_tmp/8997417895626108443/clu\n",
      "[=================================================================] 100.00% 455 0s 6ms      \n",
      "Time for merging to clu_accepted: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 11ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_not_accepted_swap tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned -v 3 \n",
      "\n",
      "Time for merging to seq_wrong_assigned: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 4ms\n",
      "createsubdb tmp/9671761839411747810/clu_tmp/8997417895626108443/clu tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_seeds -v 3 \n",
      "\n",
      "Time for merging to seq_seeds: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_seeds.merged tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned_pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 10 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 3505 type: Aminoacid\n",
      "Estimated memory consumption: 994M\n",
      "Target database size: 3960 type: Aminoacid\n",
      "Index table k-mer threshold: 74 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 3.96K 0s 200ms    \n",
      "Index table: Masked residues: 13893\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 3.96K 0s 84ms     \n",
      "Index statistics\n",
      "Entries:          2096463\n",
      "DB size:          500 MB\n",
      "Avg k-mer size:   0.032757\n",
      "Top 10 k-mers\n",
      "    FPPTWN\t1662\n",
      "    LGLKFP\t1638\n",
      "    VKYPVT\t1632\n",
      "    SGLSHT\t1622\n",
      "    WNGLGV\t1602\n",
      "    LYLSTV\t1562\n",
      "    SVCLFY\t1414\n",
      "    LVDFPV\t1090\n",
      "    PALSLY\t1087\n",
      "    QLEGLV\t1066\n",
      "Time for index table init: 0h 0m 0s 975ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 74\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 3505\n",
      "Target db start 1 to 3960\n",
      "[=================================================================] 100.00% 3.51K 2m 39s 960ms    \n",
      "\n",
      "14003.842983 k-mers per position\n",
      "600258 DB matches per sequence\n",
      "2 overflows\n",
      "20 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to seq_wrong_assigned_pref: 0h 0m 0s 5ms\n",
      "Time for processing: 0h 2m 42s 261ms\n",
      "swapdb tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned_pref tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned_pref_swaped --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 3.51K 0s 6ms      \n",
      "Computing offsets.\n",
      "[=================================================================] 100.00% 3.51K 0s 10ms     ==>                       ] 63.58% 2.23K eta 0s       \n",
      "\n",
      "Reading results.\n",
      "[=================================================================] 100.00% 3.51K 0s 11ms     \n",
      "\n",
      "Output database: tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned_pref_swaped\n",
      "[=================================================================] 100.00% 8.84K 0s 14ms     \n",
      "\n",
      "Time for merging to seq_wrong_assigned_pref_swaped: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 84ms\n",
      "align tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_seeds.merged tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned_pref_swaped tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned_pref_swaped_aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 3960 type: Aminoacid\n",
      "Target database size: 3505 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 3.70K 14s 718ms    \n",
      "Time for merging to seq_wrong_assigned_pref_swaped_aln: 0h 0m 0s 4ms\n",
      "69780 alignments calculated\n",
      "67083 sequence pairs passed the thresholds (0.961350 of overall calculated)\n",
      "18.150162 hits per query sequence\n",
      "Time for processing: 0h 0m 14s 739ms\n",
      "filterdb tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned_pref_swaped_aln tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned_pref_swaped_aln_ocol --trim-to-one-column --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using regular expression\n",
      "[=================================================================] 100.00% 3.70K 0s 49ms     \n",
      "Time for merging to seq_wrong_assigned_pref_swaped_aln_ocol: 0h 0m 0s 10ms\n",
      "Time for processing: 0h 0m 0s 68ms\n",
      "mergedbs tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_seeds.merged tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_accepted_plus_wrong tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_accepted tmp/9671761839411747810/clu_tmp/8997417895626108443/seq_wrong_assigned_pref_swaped_aln_ocol --merge-stop-empty 0 --compressed 0 -v 3 \n",
      "\n",
      "Merging the results to tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_accepted_plus_wrong\n",
      "[=================================================================] 100.00% 3.96K 0s 2ms      \n",
      "Time for merging to clu_accepted_plus_wrong: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 4ms\n",
      "tsv2db tmp/9671761839411747810/clu_tmp/8997417895626108443/missing.single.seqs tmp/9671761839411747810/clu_tmp/8997417895626108443/missing.single.seqs.db --output-dbtype 6 --compressed 0 -v 3 \n",
      "\n",
      "Output database type: Clustering\n",
      "Time for merging to missing.single.seqs.db: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "mergedbs tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_accepted_plus_wrong_plus_single tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_accepted_plus_wrong tmp/9671761839411747810/clu_tmp/8997417895626108443/missing.single.seqs.db --merge-stop-empty 0 --compressed 0 -v 3 \n",
      "\n",
      "Merging the results to tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_accepted_plus_wrong_plus_single\n",
      "[=================================================================] 100.00% 8.84K 0s 3ms      \n",
      "Time for merging to clu_accepted_plus_wrong_plus_single: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 6ms\n",
      "clust tmp/9671761839411747810/input tmp/9671761839411747810/clu_tmp/8997417895626108443/clu_accepted_plus_wrong_plus_single tmp/9671761839411747810/clu --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 8.84K 0s 22ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 37465 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 8.84K 0s 22ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 8.84K 0s 7ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 126ms\n",
      "Total time: 0h 0m 0s 246ms\n",
      "\n",
      "Size of the sequence database: 8836\n",
      "Size of the alignment database: 8836\n",
      "Number of clusters: 669\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 292ms\n",
      "createtsv tmp/9671761839411747810/input tmp/9671761839411747810/input tmp/9671761839411747810/clu tmp/9671761839411747810/cluster.tsv --threads 8 -v 3 \n",
      "\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 28ms\n",
      "result2repseq tmp/9671761839411747810/input tmp/9671761839411747810/clu tmp/9671761839411747810/clu_rep --db-load-mode 0 --compressed 0 --threads 8 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 669 0s 11ms     \n",
      "Time for merging to clu_rep: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 36ms\n",
      "result2flat tmp/9671761839411747810/input tmp/9671761839411747810/input tmp/9671761839411747810/clu_rep tmp/9671761839411747810/rep_seq.fasta --use-fasta-header -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "createseqfiledb tmp/9671761839411747810/input tmp/9671761839411747810/clu tmp/9671761839411747810/clu_seqs --threads 8 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 669 0s 6ms      \n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 19ms\n",
      "result2flat tmp/9671761839411747810/input tmp/9671761839411747810/input tmp/9671761839411747810/clu_seqs tmp/9671761839411747810/all_seqs.fasta -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 13ms\n"
     ]
    }
   ],
   "source": [
    "!mmseqs easy-cluster data_files/all_seqs.fasta data_files/cluster tmp --min-seq-id 0.1 --cov-mode 5 -c 0.2 -s 10.0 --remove-tmp-files --max-iterations 50000 --cluster-reassign --cluster-steps 9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_rep_id</th>\n",
       "      <th>fasta_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5IIA_1</td>\n",
       "      <td>5IIA_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4ETP_9</td>\n",
       "      <td>4ETP_9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4ETP_9</td>\n",
       "      <td>3DCO_7085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6IMF_26</td>\n",
       "      <td>6IMF_26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4EQA_30</td>\n",
       "      <td>4EQA_30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8831</th>\n",
       "      <td>4IOP_8804</td>\n",
       "      <td>1TDQ_3878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8832</th>\n",
       "      <td>4IOP_8804</td>\n",
       "      <td>3WWK_4913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8833</th>\n",
       "      <td>4IOP_8804</td>\n",
       "      <td>3WWK_4914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8834</th>\n",
       "      <td>4IOP_8804</td>\n",
       "      <td>1IJK_6077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8835</th>\n",
       "      <td>4IOP_8804</td>\n",
       "      <td>5TZN_7229</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8836 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     cluster_rep_id   fasta_id\n",
       "0            5IIA_1     5IIA_1\n",
       "1            4ETP_9     4ETP_9\n",
       "2            4ETP_9  3DCO_7085\n",
       "3           6IMF_26    6IMF_26\n",
       "4           4EQA_30    4EQA_30\n",
       "...             ...        ...\n",
       "8831      4IOP_8804  1TDQ_3878\n",
       "8832      4IOP_8804  3WWK_4913\n",
       "8833      4IOP_8804  3WWK_4914\n",
       "8834      4IOP_8804  1IJK_6077\n",
       "8835      4IOP_8804  5TZN_7229\n",
       "\n",
       "[8836 rows x 2 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_df = pd.read_csv(\n",
    "    \"data_files/cluster_cluster.tsv\", sep=\"\\t\", header=None\n",
    ")\n",
    "cluster_df = cluster_df.rename(columns={0: \"cluster_rep_id\", 1: \"fasta_id\"})\n",
    "cluster_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_rep_id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1FLE_8140</th>\n",
       "      <td>607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2FU5_4863</th>\n",
       "      <td>380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3WQB_7929</th>\n",
       "      <td>334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3QHY_8106</th>\n",
       "      <td>318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1SGD_3901</th>\n",
       "      <td>298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4C7N_590</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4C5G_106</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4BKX_7155</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4BD9_4206</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6SAK_3271</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>669 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                cluster_size\n",
       "cluster_rep_id              \n",
       "1FLE_8140                607\n",
       "2FU5_4863                380\n",
       "3WQB_7929                334\n",
       "3QHY_8106                318\n",
       "1SGD_3901                298\n",
       "...                      ...\n",
       "4C7N_590                   1\n",
       "4C5G_106                   1\n",
       "4BKX_7155                  1\n",
       "4BD9_4206                  1\n",
       "6SAK_3271                  1\n",
       "\n",
       "[669 rows x 1 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_df_count = cluster_df.groupby([\"cluster_rep_id\"]).count().sort_values(\n",
    "    \"fasta_id\", ascending=False\n",
    ").rename(columns={\"fasta_id\": \"cluster_size\"})\n",
    "cluster_df_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "split\n",
       "train    0.6\n",
       "test     0.4\n",
       "Name: count, dtype: float64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_fasta_ids = []\n",
    "# Ultimatley, we want 80% of the data to be in the training set, but we know\n",
    "# that we will need to remove some sequences from test to training later to\n",
    "# ensure that the training set is not too similar to the test set. So we will\n",
    "# aim for 60% in the training set for now.\n",
    "target_train_count = round(0.6 * len(df))\n",
    "\n",
    "for cluster_rep_id, cluster_size in cluster_df_count.iterrows():\n",
    "    # add largest cluster first to the training set\n",
    "    train_fasta_ids.extend(\n",
    "        cluster_df[cluster_df[\"cluster_rep_id\"] == cluster_rep_id][\n",
    "            \"fasta_id\"\n",
    "        ].tolist()\n",
    "    )\n",
    "    if len(train_fasta_ids) >= target_train_count:\n",
    "        break\n",
    "\n",
    "df[\"split\"] = df[\"fasta_id\"].apply(\n",
    "    lambda x: \"train\" if x in train_fasta_ids else \"test\"\n",
    ")\n",
    "(df[\"split\"].value_counts() / len(df)).round(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "At this point the train and test splits should be slightly dissimilar to each other. But, we want to guarantee minimal similarity, so we will perform a series of iterations to ensure that the splits are as dissimilar as possible. In each iteration we do the following:\n",
    "1. Compute the pairwise sequence identity between the train and test splits\n",
    "2. Any sequence in the test split that has a sequence identity of at least 30% with any sequence in the train split and the coverage of the smaller sequence is at least 80%, is moved from the test split to the train split to prevent data leakage.\n",
    "3. Repeat from step 1 until no sequence is moved from the test split to the train split."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "from subprocess import run\n",
    "\n",
    "\n",
    "def do_split_iteration(df, splits):\n",
    "    split1, split2 = splits\n",
    "    with open(f\"data_files/{split1}.fasta\", \"w\") as f:\n",
    "        for _, row in df[df[\"split\"] == split1].iterrows():\n",
    "            fasta_id = row[\"fasta_id\"]\n",
    "            seq = row[\"combined\"]\n",
    "            f.write(f\">{fasta_id}\\n{seq}\\n\")\n",
    "\n",
    "    with open(f\"data_files/{split2}.fasta\", \"w\") as f:\n",
    "        for _, row in df[df[\"split\"] == split2].iterrows():\n",
    "            fasta_id = row[\"fasta_id\"]\n",
    "            seq = row[\"combined\"]\n",
    "            f.write(f\">{fasta_id}\\n{seq}\\n\")\n",
    "\n",
    "    command = [\n",
    "        \"mmseqs\",\n",
    "        \"easy-search\",\n",
    "        f\"data_files/{split1}.fasta\",\n",
    "        f\"data_files/{split2}.fasta\",\n",
    "        \"data_files/matches.tsv\",\n",
    "        \"tmp\",\n",
    "        \"-s\",\n",
    "        \"7.5\",\n",
    "        \"--min-seq-id\",\n",
    "        \"0.3\",\n",
    "        \"--cov-mode\",\n",
    "        \"5\",\n",
    "        \"-c\",\n",
    "        \"0.8\",\n",
    "    ]\n",
    "\n",
    "    result = run(command, capture_output=True, text=True)\n",
    "    if result.returncode != 0:\n",
    "        print(result.stderr)\n",
    "        raise ValueError(\"Error running mmseqs\")\n",
    "    try:\n",
    "        matches_df = pd.read_csv(\n",
    "            \"data_files/matches.tsv\", sep=\"\\t\", header=None\n",
    "        )\n",
    "    except pd.errors.EmptyDataError:\n",
    "        print(\"No matches found\")\n",
    "        return df, 0\n",
    "    matches_df = matches_df.rename(\n",
    "        columns={0: f\"{split1}_id\", 1: f\"{split2}_id\", 2: \"pident\"}\n",
    "    )\n",
    "    fasta_ids_to_change_split = matches_df[f\"{split2}_id\"].unique()\n",
    "    print(\n",
    "        f\"Changing {len(fasta_ids_to_change_split)} sequences from {split2} \"\n",
    "        f\"to {split1}\"\n",
    "    )\n",
    "    df[\"split\"] = df.apply(\n",
    "        lambda row: (\n",
    "            split1\n",
    "            if row[\"fasta_id\"] in fasta_ids_to_change_split\n",
    "            else row[\"split\"]\n",
    "        ),\n",
    "        axis=1,\n",
    "    )\n",
    "    return df, len(fasta_ids_to_change_split)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Changing 1256 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.74\n",
      "test     0.26\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 315 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.78\n",
      "test     0.22\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 96 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.79\n",
      "test     0.21\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 38 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.79\n",
      "test     0.21\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 8 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.8\n",
      "test     0.2\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 1 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.8\n",
      "test     0.2\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "No matches found\n",
      "Current split ratios: split\n",
      "train    0.8\n",
      "test     0.2\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n"
     ]
    }
   ],
   "source": [
    "while True:\n",
    "    df, total_changed = do_split_iteration(df, (\"train\", \"test\"))\n",
    "    print(\n",
    "        f\"Current split ratios: {(df['split'].value_counts() / len(df)).round(2)}\"\n",
    "    )\n",
    "    print(\"*\" * 80)\n",
    "    if (df[\"split\"] == \"test\").sum() == 0 or total_changed == 0:\n",
    "        break"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We want to further split the test split into a validation and test split. We will use the same procedure and parameters as above to ensure that the validation and test splits are as dissimilar as possible."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"data_files/all_test_seqs.fasta\", \"w\") as f:\n",
    "    for i, row in df[df[\"split\"] == \"test\"].iterrows():\n",
    "        fasta_id = row[\"fasta_id\"]\n",
    "        sequence = row[\"combined\"]\n",
    "        f.write(f\">{fasta_id}\\n{sequence}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data_files/all_test_seqs.fasta data_files/cluster tmp --min-seq-id 0.1 --cov-mode 5 -c 0.2 -s 10.0 --remove-tmp-files --max-iterations 50000 --cluster-reassign --cluster-steps 9 \n",
      "\n",
      "MMseqs Version:                     \tf5f780acd64482cd59b46eae0a107f763cd17b4d\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t10\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.2\n",
      "Coverage mode                       \t5\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t8\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.1\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t50000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t9\n",
      "Cluster reassign                    \ttrue\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Database type                       \t0\n",
      "Shuffle input database              \ttrue\n",
      "Createdb mode                       \t1\n",
      "Write lookup file                   \t0\n",
      "Offset of numeric ids               \t0\n",
      "\n",
      "createdb data_files/all_test_seqs.fasta tmp/16491769556438291860/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mShuffle database cannot be combined with --createdb-mode 0\n",
      "\u001b[39m\u001b[33mWe recompute with --shuffle 0\n",
      "\u001b[39mConverting sequences\n",
      "[1718] 0s 1ms\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\n",
      "Time for merging to input: 0h 0m 0s 0ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Create directory tmp/16491769556438291860/clu_tmp\n",
      "cluster tmp/16491769556438291860/input tmp/16491769556438291860/clu tmp/16491769556438291860/clu_tmp -s 10 --max-seqs 20 -c 0.2 --cov-mode 5 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.1 --max-iterations 50000 --cluster-steps 9 --cluster-reassign 1 --remove-tmp-files 0 \n",
      "\n",
      "Set cluster mode SET COVER\n",
      "linclust tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 5 -k 0 -c 0.2 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 5 -k 0 -c 0.2 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 1808 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 1.81K 0s 11ms     \n",
      "Sort kmer 0h 0m 0s 41ms\n",
      "Sort by rep. sequence 0h 0m 0s 36ms\n",
      "Time for fill: 0h 0m 0s 0ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 103ms\n",
      "rescorediagonal tmp/16491769556438291860/input tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 5 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 1.81K 0s 2ms      \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 2ms>                  ] 72.16% 1.30K eta 0s       \n",
      "Time for processing: 0h 0m 0s 18ms\n",
      "clust tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref_rescore1 tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pre_clust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 1.81K 0s 27ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 2028 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 1.81K 0s 12ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 1.81K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 135ms\n",
      "Total time: 0h 0m 0s 175ms\n",
      "\n",
      "Size of the sequence database: 1808\n",
      "Size of the alignment database: 1808\n",
      "Number of clusters: 721\n",
      "\n",
      "Writing results 0h 0m 0s 3ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 190ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/order_redundancy tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/order_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "filterdb tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref_filter1 tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref_filter2 --filter-file tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/order_redundancy --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 721 0s 22ms     \n",
      "Time for merging to pref_filter2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 76ms\n",
      "rescorediagonal tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/input_step_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/input_step_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref_filter2 tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.2 -a 0 --cov-mode 5 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mCan not find any score per column for coverage 0.200000 and sequence identity 0.100000. No hit will be filtered.\n",
      "[=================================================================] 100.00% 721 0s 2ms      \n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 67ms\n",
      "align tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/input_step_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/input_step_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pref_rescore2 tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 721 type: Aminoacid\n",
      "Target database size: 721 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 721 0s 59ms     \n",
      "Time for merging to aln: 0h 0m 0s 3ms\n",
      "841 alignments calculated\n",
      "841 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.166435 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 73ms\n",
      "clust tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/input_step_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/aln tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/clust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 721 0s 22ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 120 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 721 0s 11ms     =============>   ] 94.86% 684 eta 0s       \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 721 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 133ms\n",
      "Total time: 0h 0m 0s 188ms\n",
      "\n",
      "Size of the sequence database: 721\n",
      "Size of the alignment database: 721\n",
      "Number of clusters: 616\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 221ms\n",
      "mergeclusters tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/pre_clust tmp/16491769556438291860/clu_tmp/1081386626715268142/linclust/3109238961440592856/clust --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 721 0s 16ms     25% 118 eta 0s       \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 616 0s 48ms     \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 1.81K 0s 61ms                                  ] 36.19% 655 eta 0s       \n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 138ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_redundancy tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "prefilter tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 616 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 616 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 616 0s 20ms     \n",
      "Index table: Masked residues: 4421\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 616 0s 13ms     \n",
      "Index statistics\n",
      "Entries:          110130\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.001721\n",
      "Top 10 k-mers\n",
      "    RKFDTG\t4\n",
      "    CNDVAK\t4\n",
      "    AFNDIR\t4\n",
      "    ITCFPW\t4\n",
      "    TPMSGY\t4\n",
      "    LHTCVA\t3\n",
      "    VYPMVC\t3\n",
      "    GFAGAD\t3\n",
      "    GRLIDD\t3\n",
      "    FHYGGD\t3\n",
      "Time for index table init: 0h 0m 0s 424ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 616\n",
      "Target db start 1 to 616\n",
      "[=================================================================] 100.00% 616 0s 12ms     \n",
      "\n",
      "2.124203 k-mers per position\n",
      "230 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 186ms\n",
      "align tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step0 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 616 type: Aminoacid\n",
      "Target database size: 616 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 616 0s 35ms     \n",
      "Time for merging to aln_step0: 0h 0m 0s 0ms\n",
      "753 alignments calculated\n",
      "751 sequence pairs passed the thresholds (0.997344 of overall calculated)\n",
      "1.219156 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 42ms\n",
      "clust tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step0 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step0 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 616 0s 5ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 1 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 616 0s 4ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 616 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 37ms\n",
      "Total time: 0h 0m 0s 52ms\n",
      "\n",
      "Size of the sequence database: 616\n",
      "Size of the alignment database: 616\n",
      "Number of clusters: 555\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 53ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step0 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step1 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step1 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2.125 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 555 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 555 type: Aminoacid\n",
      "Index table k-mer threshold: 144 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 555 0s 11ms     \n",
      "Index table: Masked residues: 3201\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 555 0s 62ms     \n",
      "Index statistics\n",
      "Entries:          163994\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002562\n",
      "Top 10 k-mers\n",
      "    DPQALA\t3\n",
      "    PSSNAE\t3\n",
      "    VGVPDE\t3\n",
      "    RKFDTG\t3\n",
      "    ERKLEI\t3\n",
      "    SGMEEK\t3\n",
      "    EERTDA\t2\n",
      "    QQNAEA\t2\n",
      "    GSEAFA\t2\n",
      "    NFKLFA\t2\n",
      "Time for index table init: 0h 0m 0s 457ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 144\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 555\n",
      "Target db start 1 to 555\n",
      "[=================================================================] 100.00% 555 0s 39ms     \n",
      "\n",
      "9.575464 k-mers per position\n",
      "332 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 1ms\n",
      "align tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step1 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step1 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step1 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 555 type: Aminoacid\n",
      "Target database size: 555 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 555 0s 15ms     \n",
      "Time for merging to aln_step1: 0h 0m 0s 0ms\n",
      "620 alignments calculated\n",
      "597 sequence pairs passed the thresholds (0.962903 of overall calculated)\n",
      "1.075676 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 17ms\n",
      "clust tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step1 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step1 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step1 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 555 0s 2ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 6 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 555 0s 2ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 555 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 18ms\n",
      "Total time: 0h 0m 0s 25ms\n",
      "\n",
      "Size of the sequence database: 555\n",
      "Size of the alignment database: 555\n",
      "Number of clusters: 532\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 26ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step1 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step1 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step2 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step2 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3.25 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 532 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 532 type: Aminoacid\n",
      "Index table k-mer threshold: 134 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 532 0s 13ms     \n",
      "Index table: Masked residues: 2881\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 532 0s 11ms     \n",
      "Index statistics\n",
      "Entries:          180726\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002824\n",
      "Top 10 k-mers\n",
      "    KLEGKA\t3\n",
      "    DPQALA\t3\n",
      "    PSSNAE\t3\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    ERKLEI\t3\n",
      "    QSNLAK\t3\n",
      "    SGMEEK\t3\n",
      "    ANLEKL\t3\n",
      "Time for index table init: 0h 0m 0s 392ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 134\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 532\n",
      "Target db start 1 to 532\n",
      "[=================================================================] 100.00% 532 0s 67ms     \n",
      "\n",
      "31.110823 k-mers per position\n",
      "432 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 975ms\n",
      "align tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step2 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step2 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step2 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 532 type: Aminoacid\n",
      "Target database size: 532 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 532 0s 26ms     \n",
      "Time for merging to aln_step2: 0h 0m 0s 7ms\n",
      "622 alignments calculated\n",
      "556 sequence pairs passed the thresholds (0.893891 of overall calculated)\n",
      "1.045113 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 71ms\n",
      "clust tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step2 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step2 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step2 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 532 0s 1ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 10 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 532 0s 12ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 532 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 40ms\n",
      "Total time: 0h 0m 0s 41ms\n",
      "\n",
      "Size of the sequence database: 532\n",
      "Size of the alignment database: 532\n",
      "Number of clusters: 517\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 44ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step2 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step2 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step3 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step3: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step3 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step3 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 4.375 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 517 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 517 type: Aminoacid\n",
      "Index table k-mer threshold: 124 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 517 0s 19ms     \n",
      "Index table: Masked residues: 2692\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 517 0s 34ms     \n",
      "Index statistics\n",
      "Entries:          184539\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002883\n",
      "Top 10 k-mers\n",
      "    AEGLAA\t3\n",
      "    KLEGKA\t3\n",
      "    ELLLLD\t3\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "Time for index table init: 0h 0m 0s 415ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 124\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 517\n",
      "Target db start 1 to 517\n",
      "[=================================================================] 100.00% 517 0s 160ms    \n",
      "\n",
      "95.954302 k-mers per position\n",
      "536 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step3: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 119ms\n",
      "align tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step3 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step3 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step3 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 517 type: Aminoacid\n",
      "Target database size: 517 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 517 0s 16ms     \n",
      "Time for merging to aln_step3: 0h 0m 0s 0ms\n",
      "809 alignments calculated\n",
      "547 sequence pairs passed the thresholds (0.676143 of overall calculated)\n",
      "1.058027 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 20ms\n",
      "clust tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step3 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step3 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step3 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 517 0s 3ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 8 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 517 0s 19ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 517 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 76ms\n",
      "Total time: 0h 0m 0s 98ms\n",
      "\n",
      "Size of the sequence database: 517\n",
      "Size of the alignment database: 517\n",
      "Number of clusters: 499\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step3: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 99ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step3 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step3 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step4 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step4: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step4 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step4 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step4 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5.5 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 499 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 499 type: Aminoacid\n",
      "Index table k-mer threshold: 114 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 499 0s 14ms     \n",
      "Index table: Masked residues: 2664\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 499 0s 11ms     \n",
      "Index statistics\n",
      "Entries:          176675\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002761\n",
      "Top 10 k-mers\n",
      "    AEGLAA\t3\n",
      "    KLEGKA\t3\n",
      "    ELLLLD\t3\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "Time for index table init: 0h 0m 0s 397ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 114\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 499\n",
      "Target db start 1 to 499\n",
      "[=================================================================] 100.00% 499 0s 413ms    \n",
      "\n",
      "280.805379 k-mers per position\n",
      "747 DB matches per sequence\n",
      "0 overflows\n",
      "3 sequences passed prefiltering per query sequence\n",
      "3 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step4: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 438ms\n",
      "align tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step4 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step4 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step4 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step4 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 499 type: Aminoacid\n",
      "Target database size: 499 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 499 0s 42ms     \n",
      "Time for merging to aln_step4: 0h 0m 0s 0ms\n",
      "1441 alignments calculated\n",
      "540 sequence pairs passed the thresholds (0.374740 of overall calculated)\n",
      "1.082164 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 73ms\n",
      "clust tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step4 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step4 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step4 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 499 0s 5ms                   ] 73.90% 369 eta 0s       \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 15 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 499 0s 5ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 499 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 39ms\n",
      "Total time: 0h 0m 0s 64ms\n",
      "\n",
      "Size of the sequence database: 499\n",
      "Size of the alignment database: 499\n",
      "Number of clusters: 473\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step4: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 65ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step4 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step4 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step5 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step5: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step5 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step5 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step5 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 6.625 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 473 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 473 type: Aminoacid\n",
      "Index table k-mer threshold: 104 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 473 0s 15ms     \n",
      "Index table: Masked residues: 2517\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 473 0s 19ms     \n",
      "Index statistics\n",
      "Entries:          165710\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002589\n",
      "Top 10 k-mers\n",
      "    KLEGKA\t3\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    SSEQAA\t2\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "    GETEDA\t2\n",
      "Time for index table init: 0h 0m 0s 435ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 104\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 473\n",
      "Target db start 1 to 473\n",
      "[=================================================================] 100.00% 473 1s 196ms    \n",
      "\n",
      "784.764099 k-mers per position\n",
      "1294 DB matches per sequence\n",
      "0 overflows\n",
      "8 sequences passed prefiltering per query sequence\n",
      "8 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step5: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 2s 279ms\n",
      "align tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step5 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step5 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step5 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step5 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 473 type: Aminoacid\n",
      "Target database size: 473 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 473 0s 40ms     \n",
      "Time for merging to aln_step5: 0h 0m 0s 3ms====================>  ] 95.76% 453 eta 0s       \n",
      "3700 alignments calculated\n",
      "506 sequence pairs passed the thresholds (0.136757 of overall calculated)\n",
      "1.069767 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 52ms\n",
      "clust tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step5 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step5 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step5 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 473 0s 10ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 9 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 473 0s 6ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 473 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 55ms\n",
      "Total time: 0h 0m 0s 76ms\n",
      "\n",
      "Size of the sequence database: 473\n",
      "Size of the alignment database: 473\n",
      "Number of clusters: 452\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step5: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 77ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step5 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step5 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step6 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step6: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step6 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step6 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step6 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 7.75 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 452 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 452 type: Aminoacid\n",
      "Index table k-mer threshold: 94 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 452 0s 11ms     \n",
      "Index table: Masked residues: 2302\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 452 0s 17ms     \n",
      "Index statistics\n",
      "Entries:          157958\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002468\n",
      "Top 10 k-mers\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    SSEQAA\t2\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "    GETEDA\t2\n",
      "    NKESDA\t2\n",
      "Time for index table init: 0h 0m 0s 396ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 94\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 452\n",
      "Target db start 1 to 452\n",
      "[=================================================================] 100.00% 452 2s 870ms    \n",
      "\n",
      "2089.278818 k-mers per position\n",
      "2766 DB matches per sequence\n",
      "0 overflows\n",
      "16 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step6: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 3s 908ms\n",
      "align tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step6 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step6 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step6 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step6 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 452 type: Aminoacid\n",
      "Target database size: 452 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 452 0s 88ms     \n",
      "Time for merging to aln_step6: 0h 0m 0s 3ms\n",
      "6876 alignments calculated\n",
      "470 sequence pairs passed the thresholds (0.068354 of overall calculated)\n",
      "1.039823 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 97ms\n",
      "clust tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step6 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step6 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step6 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 452 0s 18ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 8 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 452 0s 21ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 452 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 145ms\n",
      "Total time: 0h 0m 0s 177ms\n",
      "\n",
      "Size of the sequence database: 452\n",
      "Size of the alignment database: 452\n",
      "Number of clusters: 439\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step6: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 178ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step6 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step6 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step7 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step7: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 4ms\n",
      "prefilter tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step7 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step7 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step7 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 8.875 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 439 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 439 type: Aminoacid\n",
      "Index table k-mer threshold: 84 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 439 0s 17ms     \n",
      "Index table: Masked residues: 2044\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 439 0s 19ms     \n",
      "Index statistics\n",
      "Entries:          152570\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002384\n",
      "Top 10 k-mers\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "    GETEDA\t2\n",
      "    NKESDA\t2\n",
      "    LTLQEA\t2\n",
      "Time for index table init: 0h 0m 0s 566ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 84\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 439\n",
      "Target db start 1 to 439\n",
      "[=================================================================] 100.00% 439 6s 584ms    \n",
      "\n",
      "5305.456307 k-mers per position\n",
      "6442 DB matches per sequence\n",
      "0 overflows\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step7: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 8s 238ms\n",
      "align tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step7 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step7 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step7 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step7 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 439 type: Aminoacid\n",
      "Target database size: 439 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 439 0s 88ms     \n",
      "Time for merging to aln_step7: 0h 0m 0s 0ms\n",
      "7559 alignments calculated\n",
      "444 sequence pairs passed the thresholds (0.058738 of overall calculated)\n",
      "1.011389 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 114ms\n",
      "clust tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step7 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step7 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step7 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 439 0s 6ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 439 0s 11ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 439 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 68ms\n",
      "Total time: 0h 0m 0s 113ms\n",
      "\n",
      "Size of the sequence database: 439\n",
      "Size of the alignment database: 439\n",
      "Number of clusters: 435\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step7: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 159ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step7 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step7 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step8 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step8: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step8 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step8 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step8 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 10 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 435 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 435 type: Aminoacid\n",
      "Index table k-mer threshold: 74 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 435 0s 15ms     \n",
      "Index table: Masked residues: 2044\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 435 0s 23ms     \n",
      "Index statistics\n",
      "Entries:          150267\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002348\n",
      "Top 10 k-mers\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "    GETEDA\t2\n",
      "    NKESDA\t2\n",
      "    LTLQEA\t2\n",
      "Time for index table init: 0h 0m 0s 496ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 74\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 435\n",
      "Target db start 1 to 435\n",
      "[=================================================================] 100.00% 435 14s 145ms    \n",
      "\n",
      "12945.693884 k-mers per position\n",
      "15434 DB matches per sequence\n",
      "0 overflows\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step8: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 15s 771ms\n",
      "align tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step8 tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step8 tmp/16491769556438291860/clu_tmp/1081386626715268142/pref_step8 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step8 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 435 type: Aminoacid\n",
      "Target database size: 435 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 435 0s 95ms     \n",
      "Time for merging to aln_step8: 0h 0m 0s 0ms\n",
      "7684 alignments calculated\n",
      "438 sequence pairs passed the thresholds (0.057002 of overall calculated)\n",
      "1.006896 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 115ms\n",
      "clust tmp/16491769556438291860/clu_tmp/1081386626715268142/input_step8 tmp/16491769556438291860/clu_tmp/1081386626715268142/aln_step8 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step8 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 435 0s 16ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 435 0s 5ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 435 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 70ms\n",
      "Total time: 0h 0m 0s 92ms\n",
      "\n",
      "Size of the sequence database: 435\n",
      "Size of the alignment database: 435\n",
      "Number of clusters: 432\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step8: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 93ms\n",
      "mergeclusters tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/clu tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_redundancy tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step0 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step1 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step2 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step3 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step4 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step5 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step6 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step7 tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_step8 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 616 0s 3ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 555 0s 35ms     \n",
      "Clustering step 3\n",
      "[=================================================================] 100.00% 532 0s 61ms     \n",
      "Clustering step 4\n",
      "[=================================================================] 100.00% 517 0s 85ms     \n",
      "Clustering step 5\n",
      "[=================================================================] 100.00% 499 0s 102ms    \n",
      "Clustering step 6\n",
      "[=================================================================] 100.00% 473 0s 118ms    \n",
      "Clustering step 7\n",
      "[=================================================================] 100.00% 452 0s 131ms    \n",
      "Clustering step 8\n",
      "[=================================================================] 100.00% 439 0s 145ms    \n",
      "Clustering step 9\n",
      "[=================================================================] 100.00% 435 0s 158ms    \n",
      "Clustering step 10\n",
      "[=================================================================] 100.00% 432 0s 170ms    \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 1.81K 0s 182ms    \n",
      "Time for merging to clu: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 193ms\n",
      "align tmp/16491769556438291860/input tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/clu tmp/16491769556438291860/clu_tmp/1081386626715268142/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1808 type: Aminoacid\n",
      "Target database size: 1808 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 432 0s 161ms    \n",
      "Time for merging to aln: 0h 0m 0s 0ms\n",
      "1807 alignments calculated\n",
      "1637 sequence pairs passed the thresholds (0.905921 of overall calculated)\n",
      "3.789352 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 183ms\n",
      "subtractdbs tmp/16491769556438291860/clu_tmp/1081386626715268142/clu tmp/16491769556438291860/clu_tmp/1081386626715268142/aln tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_not_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "subtractdbs tmp/16491769556438291860/clu_tmp/1081386626715268142/clu tmp/16491769556438291860/clu_tmp/1081386626715268142/aln tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_not_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Remove tmp/16491769556438291860/clu_tmp/1081386626715268142/aln ids from tmp/16491769556438291860/clu_tmp/1081386626715268142/clu\n",
      "[=================================================================] 100.00% 432 0s 11ms     \n",
      "Time for merging to clu_not_accepted: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 22ms\n",
      "swapdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_not_accepted tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_not_accepted_swap --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 432 0s 4ms      \n",
      "Computing offsets.\n",
      "[=================================================================] 100.00% 432 0s 12ms     \n",
      "\n",
      "Reading results.\n",
      "[=================================================================] 100.00% 432 0s 10ms     \n",
      "\n",
      "Output database: tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_not_accepted_swap\n",
      "[=================================================================] 100.00% 1.81K 0s 12ms     \n",
      "\n",
      "Time for merging to clu_not_accepted_swap: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 110ms\n",
      "subtractdbs tmp/16491769556438291860/clu_tmp/1081386626715268142/clu tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_not_accepted tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "subtractdbs tmp/16491769556438291860/clu_tmp/1081386626715268142/clu tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_not_accepted tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Remove tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_not_accepted ids from tmp/16491769556438291860/clu_tmp/1081386626715268142/clu\n",
      "[=================================================================] 100.00% 432 0s 4ms      \n",
      "Time for merging to clu_accepted: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 11ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_not_accepted_swap tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned -v 3 \n",
      "\n",
      "Time for merging to seq_wrong_assigned: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "createsubdb tmp/16491769556438291860/clu_tmp/1081386626715268142/clu tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_seeds -v 3 \n",
      "\n",
      "Time for merging to seq_seeds: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_seeds.merged tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned_pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 10 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 171 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 603 type: Aminoacid\n",
      "Index table k-mer threshold: 74 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 603 0s 30ms     \n",
      "Index table: Masked residues: 2986\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 603 0s 40ms     \n",
      "Index statistics\n",
      "Entries:          235806\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.003684\n",
      "Top 10 k-mers\n",
      "    TALKKA\t33\n",
      "    VRVRND\t33\n",
      "    SLAHVI\t33\n",
      "    KTELAL\t33\n",
      "    ENDALR\t33\n",
      "    YDNKDS\t33\n",
      "    YVVKSS\t33\n",
      "    PISSEA\t32\n",
      "    IFKDFA\t32\n",
      "    KIAHFA\t32\n",
      "Time for index table init: 0h 0m 0s 581ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 74\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 171\n",
      "Target db start 1 to 603\n",
      "[=================================================================] 100.00% 171 8s 868ms    \n",
      "\n",
      "12198.382654 k-mers per position\n",
      "38286 DB matches per sequence\n",
      "0 overflows\n",
      "20 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to seq_wrong_assigned_pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 10s 574ms\n",
      "swapdb tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned_pref tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned_pref_swaped --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 171 0s 5ms      \n",
      "Computing offsets.\n",
      "[=================================================================] 100.00% 171 0s 6ms      \n",
      "\n",
      "Reading results.\n",
      "[=================================================================] 100.00% 171 0s 11ms     \n",
      "\n",
      "Output database: tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned_pref_swaped\n",
      "[=================================================================] 100.00% 1.81K 0s 14ms     \n",
      "\n",
      "Time for merging to seq_wrong_assigned_pref_swaped: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 68ms\n",
      "align tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_seeds.merged tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned_pref_swaped tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned_pref_swaped_aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 603 type: Aminoacid\n",
      "Target database size: 171 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 441 0s 514ms    \n",
      "Time for merging to seq_wrong_assigned_pref_swaped_aln: 0h 0m 0s 0ms\n",
      "3325 alignments calculated\n",
      "2161 sequence pairs passed the thresholds (0.649925 of overall calculated)\n",
      "4.900227 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 575ms\n",
      "filterdb tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned_pref_swaped_aln tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned_pref_swaped_aln_ocol --trim-to-one-column --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using regular expression\n",
      "[=================================================================] 100.00% 441 0s 12ms     \n",
      "Time for merging to seq_wrong_assigned_pref_swaped_aln_ocol: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 40ms\n",
      "mergedbs tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_seeds.merged tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_accepted_plus_wrong tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_accepted tmp/16491769556438291860/clu_tmp/1081386626715268142/seq_wrong_assigned_pref_swaped_aln_ocol --merge-stop-empty 0 --compressed 0 -v 3 \n",
      "\n",
      "Merging the results to tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_accepted_plus_wrong\n",
      "[=================================================================] 100.00% 603 0s 0ms      \n",
      "Time for merging to clu_accepted_plus_wrong: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "tsv2db tmp/16491769556438291860/clu_tmp/1081386626715268142/missing.single.seqs tmp/16491769556438291860/clu_tmp/1081386626715268142/missing.single.seqs.db --output-dbtype 6 --compressed 0 -v 3 \n",
      "\n",
      "Output database type: Clustering\n",
      "Time for merging to missing.single.seqs.db: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "mergedbs tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_accepted_plus_wrong_plus_single tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_accepted_plus_wrong tmp/16491769556438291860/clu_tmp/1081386626715268142/missing.single.seqs.db --merge-stop-empty 0 --compressed 0 -v 3 \n",
      "\n",
      "Merging the results to tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_accepted_plus_wrong_plus_single\n",
      "[=================================================================] 100.00% 1.81K 0s 0ms      \n",
      "Time for merging to clu_accepted_plus_wrong_plus_single: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "clust tmp/16491769556438291860/input tmp/16491769556438291860/clu_tmp/1081386626715268142/clu_accepted_plus_wrong_plus_single tmp/16491769556438291860/clu --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 1.81K 0s 11ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 1497 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 1.81K 0s 16ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 1.81K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 74ms\n",
      "Total time: 0h 0m 0s 128ms\n",
      "\n",
      "Size of the sequence database: 1808\n",
      "Size of the alignment database: 1808\n",
      "Number of clusters: 448\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 133ms\n",
      "createtsv tmp/16491769556438291860/input tmp/16491769556438291860/input tmp/16491769556438291860/clu tmp/16491769556438291860/cluster.tsv --threads 8 -v 3 \n",
      "\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 89ms\n",
      "result2repseq tmp/16491769556438291860/input tmp/16491769556438291860/clu tmp/16491769556438291860/clu_rep --db-load-mode 0 --compressed 0 --threads 8 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 448 0s 9ms      \n",
      "Time for merging to clu_rep: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 68ms\n",
      "result2flat tmp/16491769556438291860/input tmp/16491769556438291860/input tmp/16491769556438291860/clu_rep tmp/16491769556438291860/rep_seq.fasta --use-fasta-header -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "createseqfiledb tmp/16491769556438291860/input tmp/16491769556438291860/clu tmp/16491769556438291860/clu_seqs --threads 8 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 448 0s 11ms     \n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 70ms\n",
      "result2flat tmp/16491769556438291860/input tmp/16491769556438291860/input tmp/16491769556438291860/clu_seqs tmp/16491769556438291860/all_seqs.fasta -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 4ms\n"
     ]
    }
   ],
   "source": [
    "!mmseqs easy-cluster data_files/all_test_seqs.fasta data_files/cluster tmp --min-seq-id 0.1 --cov-mode 5 -c 0.2 -s 10.0 --remove-tmp-files --max-iterations 50000 --cluster-reassign --cluster-steps 9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_rep_id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5M2O_5685</th>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1IAR_3446</th>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4PWX_8349</th>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3MA2_796</th>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1FCC_6364</th>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4BKX_7155</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4C5G_106</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4C7N_590</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4CT0_105</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3V1C_7348</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>448 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                cluster_size\n",
       "cluster_rep_id              \n",
       "5M2O_5685                 40\n",
       "1IAR_3446                 37\n",
       "4PWX_8349                 34\n",
       "3MA2_796                  34\n",
       "1FCC_6364                 33\n",
       "...                      ...\n",
       "4BKX_7155                  1\n",
       "4C5G_106                   1\n",
       "4C7N_590                   1\n",
       "4CT0_105                   1\n",
       "3V1C_7348                  1\n",
       "\n",
       "[448 rows x 1 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_df = pd.read_csv(\n",
    "    \"data_files/cluster_cluster.tsv\", sep=\"\\t\", header=None\n",
    ")\n",
    "cluster_df = cluster_df.rename(columns={0: \"cluster_rep_id\", 1: \"fasta_id\"})\n",
    "cluster_df_count = (\n",
    "    cluster_df.groupby([\"cluster_rep_id\"])\n",
    "    .count()\n",
    "    .sort_values(\"fasta_id\", ascending=False)\n",
    "    .rename(columns={\"fasta_id\": \"cluster_size\"})\n",
    ")\n",
    "cluster_df_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "split\n",
       "train    0.8\n",
       "val      0.1\n",
       "test     0.1\n",
       "Name: count, dtype: float64"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# evenly distribute the clusters between val and test\n",
    "val_fasta_ids = []\n",
    "add_to_val = True\n",
    "for cluster_rep_id, cluster_size in cluster_df_count.iterrows():\n",
    "    if add_to_val:\n",
    "        val_fasta_ids.extend(\n",
    "            cluster_df[cluster_df[\"cluster_rep_id\"] == cluster_rep_id][\n",
    "                \"fasta_id\"\n",
    "            ].tolist()\n",
    "        )\n",
    "    add_to_val = not add_to_val\n",
    "\n",
    "df[\"split\"] = df.apply(\n",
    "    lambda row: \"val\" if row[\"fasta_id\"] in val_fasta_ids else row[\"split\"],\n",
    "    axis=1,\n",
    ")\n",
    "(df[\"split\"].value_counts() / len(df)).round(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Changing 54 sequences from test to val\n",
      "Current split ratios: split\n",
      "train    0.80\n",
      "val      0.11\n",
      "test     0.09\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 16 sequences from test to val\n",
      "Current split ratios: split\n",
      "train    0.80\n",
      "val      0.11\n",
      "test     0.09\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 6 sequences from test to val\n",
      "Current split ratios: split\n",
      "train    0.80\n",
      "val      0.11\n",
      "test     0.09\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 1 sequences from test to val\n",
      "Current split ratios: split\n",
      "train    0.80\n",
      "val      0.11\n",
      "test     0.09\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "No matches found\n",
      "Current split ratios: split\n",
      "train    0.80\n",
      "val      0.11\n",
      "test     0.09\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "split\n",
       "train    7028\n",
       "val       993\n",
       "test      815\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# perform final split iterations to ensure that the val set is not too similar\n",
    "# to the test set\n",
    "while True:\n",
    "    df, total_changed = do_split_iteration(df, (\"val\", \"test\"))\n",
    "    print(\n",
    "        f\"Current split ratios: {(df['split'].value_counts() / len(df)).round(2)}\"\n",
    "    )\n",
    "    print(\"*\" * 80)\n",
    "    if (df[\"split\"] == \"test\").sum() == 0 or total_changed == 0:\n",
    "        break\n",
    "\n",
    "df[\"split\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.drop(columns=[\"ligand\", \"receptor\", \"combined\", \"fasta_id\"])\n",
    "df.to_csv(\"PPB_Affinity_processed_filtered.csv\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
