{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"PPB_Affinity_processed.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# these columns are created for processing purposes, they will be removed later\n",
    "df[\"ligand\"] = df[\"Ligand Sequences\"].str.replace(\",\", \"\")\n",
    "df[\"receptor\"] = df[\"Receptor Sequences\"].str.replace(\",\", \"\")\n",
    "df[\"combined\"] = df[\"ligand\"] + df[\"receptor\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Source Data Set</th>\n",
       "      <th>Complex ID</th>\n",
       "      <th>PDB</th>\n",
       "      <th>Mutations</th>\n",
       "      <th>Ligand Chains</th>\n",
       "      <th>Receptor Chains</th>\n",
       "      <th>Ligand Name</th>\n",
       "      <th>Receptor Name</th>\n",
       "      <th>KD(M)</th>\n",
       "      <th>Affinity Method</th>\n",
       "      <th>...</th>\n",
       "      <th>PDB PubMed ID</th>\n",
       "      <th>PDB Release Date</th>\n",
       "      <th>Affinity PubMed ID</th>\n",
       "      <th>Affinity Release Date</th>\n",
       "      <th>Subgroup</th>\n",
       "      <th>Ligand Sequences</th>\n",
       "      <th>Receptor Sequences</th>\n",
       "      <th>ligand</th>\n",
       "      <th>receptor</th>\n",
       "      <th>combined</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1A22:A, B::PMID=7504735</td>\n",
       "      <td>1A22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>Human growth hormone</td>\n",
       "      <td>hGH binding protein</td>\n",
       "      <td>9.000000e-10</td>\n",
       "      <td>SPR</td>\n",
       "      <td>...</td>\n",
       "      <td>9571026.0</td>\n",
       "      <td>1998-04-29</td>\n",
       "      <td>7504735</td>\n",
       "      <td>1993 Dec 5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...</td>\n",
       "      <td>PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...</td>\n",
       "      <td>FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...</td>\n",
       "      <td>PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...</td>\n",
       "      <td>FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1A4Y:A, B::PMID=9050852</td>\n",
       "      <td>1A4Y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>Ribonuclease inhibitor</td>\n",
       "      <td>Angiogenin</td>\n",
       "      <td>5.000000e-16</td>\n",
       "      <td>Other</td>\n",
       "      <td>...</td>\n",
       "      <td>9311977.0</td>\n",
       "      <td>1998-10-14</td>\n",
       "      <td>9050852</td>\n",
       "      <td>1997 Mar 4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...</td>\n",
       "      <td>QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...</td>\n",
       "      <td>SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...</td>\n",
       "      <td>QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...</td>\n",
       "      <td>SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1ACB:E, I::PMID=9048543</td>\n",
       "      <td>1ACB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>I</td>\n",
       "      <td>Bovine alpha-chymotrypsin</td>\n",
       "      <td>Eglin c</td>\n",
       "      <td>1.490000e-12</td>\n",
       "      <td>IASP</td>\n",
       "      <td>...</td>\n",
       "      <td>1583684.0</td>\n",
       "      <td>1993-10-31</td>\n",
       "      <td>9048543</td>\n",
       "      <td>1997 Feb 18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...</td>\n",
       "      <td>KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...</td>\n",
       "      <td>CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...</td>\n",
       "      <td>KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...</td>\n",
       "      <td>CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1AHW:A, B, C::PMID=9480775</td>\n",
       "      <td>1AHW</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A, B</td>\n",
       "      <td>C</td>\n",
       "      <td>Immunoglobulin fab 5G9</td>\n",
       "      <td>Tissue factor</td>\n",
       "      <td>3.400000e-09</td>\n",
       "      <td>IASP</td>\n",
       "      <td>...</td>\n",
       "      <td>9480775.0</td>\n",
       "      <td>1998-02-25</td>\n",
       "      <td>9480775</td>\n",
       "      <td>1998 Feb 6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...</td>\n",
       "      <td>TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...</td>\n",
       "      <td>DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...</td>\n",
       "      <td>TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...</td>\n",
       "      <td>DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1AK4:A, D::PMID=9223641</td>\n",
       "      <td>1AK4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>D</td>\n",
       "      <td>Cyclophilin A</td>\n",
       "      <td>HIV-1 capsid protein</td>\n",
       "      <td>1.200000e-05</td>\n",
       "      <td>SPR</td>\n",
       "      <td>...</td>\n",
       "      <td>8980234.0</td>\n",
       "      <td>1997-10-15</td>\n",
       "      <td>9223641</td>\n",
       "      <td>1997 Jun 27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...</td>\n",
       "      <td>PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...</td>\n",
       "      <td>MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...</td>\n",
       "      <td>PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...</td>\n",
       "      <td>MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Source Data Set                  Complex ID   PDB Mutations Ligand Chains  \\\n",
       "0     SKEMPI v2.0     1A22:A, B::PMID=7504735  1A22       NaN             A   \n",
       "1     SKEMPI v2.0     1A4Y:A, B::PMID=9050852  1A4Y       NaN             A   \n",
       "2     SKEMPI v2.0     1ACB:E, I::PMID=9048543  1ACB       NaN             E   \n",
       "3     SKEMPI v2.0  1AHW:A, B, C::PMID=9480775  1AHW       NaN          A, B   \n",
       "4     SKEMPI v2.0     1AK4:A, D::PMID=9223641  1AK4       NaN             A   \n",
       "\n",
       "  Receptor Chains                Ligand Name         Receptor Name  \\\n",
       "0               B       Human growth hormone   hGH binding protein   \n",
       "1               B     Ribonuclease inhibitor            Angiogenin   \n",
       "2               I  Bovine alpha-chymotrypsin               Eglin c   \n",
       "3               C     Immunoglobulin fab 5G9         Tissue factor   \n",
       "4               D              Cyclophilin A  HIV-1 capsid protein   \n",
       "\n",
       "          KD(M) Affinity Method  ... PDB PubMed ID PDB Release Date  \\\n",
       "0  9.000000e-10             SPR  ...     9571026.0       1998-04-29   \n",
       "1  5.000000e-16           Other  ...     9311977.0       1998-10-14   \n",
       "2  1.490000e-12            IASP  ...     1583684.0       1993-10-31   \n",
       "3  3.400000e-09            IASP  ...     9480775.0       1998-02-25   \n",
       "4  1.200000e-05             SPR  ...     8980234.0       1997-10-15   \n",
       "\n",
       "   Affinity PubMed ID  Affinity Release Date Subgroup  \\\n",
       "0             7504735             1993 Dec 5      NaN   \n",
       "1             9050852             1997 Mar 4      NaN   \n",
       "2             9048543            1997 Feb 18      NaN   \n",
       "3             9480775             1998 Feb 6      NaN   \n",
       "4             9223641            1997 Jun 27      NaN   \n",
       "\n",
       "                                    Ligand Sequences  \\\n",
       "0  FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...   \n",
       "1  SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...   \n",
       "2  CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...   \n",
       "3  DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...   \n",
       "4  MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...   \n",
       "\n",
       "                                  Receptor Sequences  \\\n",
       "0  PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...   \n",
       "1  QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...   \n",
       "2  KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...   \n",
       "3  TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...   \n",
       "4  PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...   \n",
       "\n",
       "                                              ligand  \\\n",
       "0  FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...   \n",
       "1  SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...   \n",
       "2  CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...   \n",
       "3  DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...   \n",
       "4  MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...   \n",
       "\n",
       "                                            receptor  \\\n",
       "0  PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...   \n",
       "1  QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...   \n",
       "2  KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...   \n",
       "3  TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...   \n",
       "4  PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...   \n",
       "\n",
       "                                            combined  \n",
       "0  FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...  \n",
       "1  SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...  \n",
       "2  CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...  \n",
       "3  DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...  \n",
       "4  MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    12048.000000\n",
       "mean         7.541897\n",
       "std          2.144316\n",
       "min          1.318759\n",
       "25%          5.872895\n",
       "50%          7.602060\n",
       "75%          9.020015\n",
       "max         15.698970\n",
       "Name: pKd, dtype: float64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# convert KD to pKD\n",
    "df[\"pKd\"] = -np.log10(df[\"KD(M)\"])\n",
    "df[\"pKd\"].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dropped 2118 rows with ligand or receptor chain(s) shorter than 20\n",
      "Remaining rows: 9930\n"
     ]
    }
   ],
   "source": [
    "# filter out rows where any ligand or receptor chain is shorter than 20\n",
    "# amino acids\n",
    "minlen = 20\n",
    "f = lambda seqs: any([len(seq) < minlen for seq in seqs])\n",
    "min_len_mask = df[\"Ligand Sequences\"].str.split(\",\").apply(f) | df[\n",
    "    \"Receptor Sequences\"\n",
    "].str.split(\",\").apply(f)\n",
    "df = df[~min_len_mask].reset_index(drop=True)\n",
    "print(\n",
    "    f\"Dropped {min_len_mask.sum()} rows with ligand or receptor chain(s) \"\n",
    "    f\"shorter than {minlen}\"\n",
    ")\n",
    "print(f\"Remaining rows: {len(df)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Affinity Method\n",
       "NaN        3430\n",
       "SPR        2431\n",
       "FL         1090\n",
       "Other       786\n",
       "ITC         555\n",
       "IASP        476\n",
       "SP          336\n",
       "RA          320\n",
       "ELISA       172\n",
       "IARA        125\n",
       "BLI         104\n",
       "Unknown      96\n",
       "IAGE          9\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check affinity method distribution in the dataset, we will use the Affinity\n",
    "# Method for handling duplicated sequences\n",
    "df[\"Affinity Method\"].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# replace nan method with \"Unknown\"\n",
    "df[\"Affinity Method\"] = df[\"Affinity Method\"].fillna(\"Unknown\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_duplicates_by_method(\n",
    "    df,\n",
    "    seq_col=\"combined\",\n",
    "    kd_col=\"KD(M)\",\n",
    "    pkd_col=\"pKd\",\n",
    "    method_col=\"Affinity Method\",\n",
    "):\n",
    "    \"\"\"\n",
    "    Group by 'seq_col' (the combined sequence) and resolve duplicates by method priority:\n",
    "      1. Within each group (all rows sharing the same 'combined' sequence), find the highest-priority method present.\n",
    "      2. Average the KD(M) and pKd values for all rows of that method.\n",
    "      3. Keep the *first row* from that subset for all other columns.\n",
    "    \"\"\"\n",
    "\n",
    "    # Priority list of methods to choose from\n",
    "    method_priority = [\n",
    "        \"SPR\",\n",
    "        \"ITC\",\n",
    "        \"BLI\",\n",
    "        \"RA\",\n",
    "        \"FL\",\n",
    "        \"SP\",\n",
    "        \"ELISA\",\n",
    "        \"IASP\",\n",
    "        \"IARA\",\n",
    "        \"IAGE\",\n",
    "        \"Other\",\n",
    "        \"Unknown\",\n",
    "    ]\n",
    "\n",
    "    def choose_best_method(group):\n",
    "        \"\"\"\n",
    "        For a subset of df with the same 'combined' sequence:\n",
    "          - Iterate over the 'method_priority' list in order.\n",
    "          - If the group contains that method, average KD(M) across those rows.\n",
    "          - Return the *first row* of that subset for all other columns.\n",
    "        \"\"\"\n",
    "        methods_in_group = group[method_col].unique()\n",
    "\n",
    "        for meth in method_priority:\n",
    "            if meth in methods_in_group:\n",
    "                # Get only the rows of the current top-priority method\n",
    "                subset = group[group[method_col] == meth]\n",
    "\n",
    "                # Average KD(M) and pKD for this subset\n",
    "                avg_kd = subset[kd_col].mean()\n",
    "                avg_pkd = subset[pkd_col].mean()\n",
    "\n",
    "                # Take the *first row* from THIS subset\n",
    "                # (the subset that contributes to the average)\n",
    "                first_row = subset.iloc[0].copy()\n",
    "\n",
    "                # Update the KD(M) and method to reflect the average and the chosen method\n",
    "                first_row[kd_col] = avg_kd\n",
    "                first_row[pkd_col] = avg_pkd\n",
    "                first_row[method_col] = meth\n",
    "\n",
    "                return first_row\n",
    "        raise ValueError(\n",
    "            \"None of the methods in the priority list were found in the group!\"\n",
    "        )\n",
    "\n",
    "    # Apply the 'choose_best_method' to each group of duplicated sequences\n",
    "    result_df = df.groupby(seq_col, as_index=False).apply(\n",
    "        choose_best_method, include_groups=False\n",
    "    )\n",
    "    return result_df.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Remaining rows after filtering duplicates: 8814\n"
     ]
    }
   ],
   "source": [
    "df = filter_duplicates_by_method(df)\n",
    "print(f\"Remaining rows after filtering duplicates: {len(df)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>combined</th>\n",
       "      <th>Source Data Set</th>\n",
       "      <th>Complex ID</th>\n",
       "      <th>PDB</th>\n",
       "      <th>Mutations</th>\n",
       "      <th>Ligand Chains</th>\n",
       "      <th>Receptor Chains</th>\n",
       "      <th>Ligand Name</th>\n",
       "      <th>Receptor Name</th>\n",
       "      <th>KD(M)</th>\n",
       "      <th>...</th>\n",
       "      <th>PDB Release Date</th>\n",
       "      <th>Affinity PubMed ID</th>\n",
       "      <th>Affinity Release Date</th>\n",
       "      <th>Subgroup</th>\n",
       "      <th>Ligand Sequences</th>\n",
       "      <th>Receptor Sequences</th>\n",
       "      <th>ligand</th>\n",
       "      <th>receptor</th>\n",
       "      <th>pKd</th>\n",
       "      <th>fasta_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>2WQZ:B, C::PMID=18093521</td>\n",
       "      <td>2WQZ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B</td>\n",
       "      <td>C</td>\n",
       "      <td>Neuroligin4, X-Linked</td>\n",
       "      <td>Neurexin-1-beta</td>\n",
       "      <td>1.320000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>2009-09-08</td>\n",
       "      <td>18093521</td>\n",
       "      <td>2007 Dec 20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...</td>\n",
       "      <td>HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...</td>\n",
       "      <td>AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...</td>\n",
       "      <td>HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...</td>\n",
       "      <td>6.879426</td>\n",
       "      <td>2WQZ_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>5IIA:A, B, C, D::PMID=28622512</td>\n",
       "      <td>5IIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B, D</td>\n",
       "      <td>A, C</td>\n",
       "      <td>sperm lysinR</td>\n",
       "      <td>red abalone egg VERL repeat 3 (VR3)</td>\n",
       "      <td>1.810000e-09</td>\n",
       "      <td>...</td>\n",
       "      <td>2017-06-14</td>\n",
       "      <td>28622512</td>\n",
       "      <td>2017 Jun 15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...</td>\n",
       "      <td>FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...</td>\n",
       "      <td>AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...</td>\n",
       "      <td>FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...</td>\n",
       "      <td>8.742321</td>\n",
       "      <td>5IIA_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>4ZI2:A, C::PMID=26455799</td>\n",
       "      <td>4ZI2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>BART-like domain of BARTL1/CCDC104 1-113</td>\n",
       "      <td>Arl3FL-GppNHp</td>\n",
       "      <td>4.200000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>2015-11-11</td>\n",
       "      <td>26455799</td>\n",
       "      <td>2015 Nov 3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...</td>\n",
       "      <td>LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...</td>\n",
       "      <td>AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...</td>\n",
       "      <td>LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...</td>\n",
       "      <td>6.376751</td>\n",
       "      <td>4ZI2_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>4UF1:A, B::PMID=26249341</td>\n",
       "      <td>4UF1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>Bak BH3</td>\n",
       "      <td>Deerpox virus DPV022</td>\n",
       "      <td>6.980000e-06</td>\n",
       "      <td>...</td>\n",
       "      <td>2015-08-05</td>\n",
       "      <td>26249341</td>\n",
       "      <td>2015 Aug</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...</td>\n",
       "      <td>STMGQVGRQLAIIGDDINRRYD</td>\n",
       "      <td>AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...</td>\n",
       "      <td>STMGQVGRQLAIIGDDINRRYD</td>\n",
       "      <td>5.156145</td>\n",
       "      <td>4UF1_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>6SAZ:A, B::PMID=31604990</td>\n",
       "      <td>6SAZ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>Crayfish Astacin</td>\n",
       "      <td>Cleaved human fetuin-b</td>\n",
       "      <td>1.400000e-10</td>\n",
       "      <td>...</td>\n",
       "      <td>2019-10-23</td>\n",
       "      <td>31604990</td>\n",
       "      <td>2019 Oct 11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...</td>\n",
       "      <td>ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...</td>\n",
       "      <td>AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...</td>\n",
       "      <td>ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...</td>\n",
       "      <td>9.853872</td>\n",
       "      <td>6SAZ_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            combined Source Data Set  \\\n",
       "0  AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...   PDBbind v2020   \n",
       "1  AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...   PDBbind v2020   \n",
       "2  AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...   PDBbind v2020   \n",
       "3  AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...   PDBbind v2020   \n",
       "4  AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...   PDBbind v2020   \n",
       "\n",
       "                       Complex ID   PDB Mutations Ligand Chains  \\\n",
       "0        2WQZ:B, C::PMID=18093521  2WQZ       NaN             B   \n",
       "1  5IIA:A, B, C, D::PMID=28622512  5IIA       NaN          B, D   \n",
       "2        4ZI2:A, C::PMID=26455799  4ZI2       NaN             C   \n",
       "3        4UF1:A, B::PMID=26249341  4UF1       NaN             A   \n",
       "4        6SAZ:A, B::PMID=31604990  6SAZ       NaN             A   \n",
       "\n",
       "  Receptor Chains                               Ligand Name  \\\n",
       "0               C                     Neuroligin4, X-Linked   \n",
       "1            A, C                              sperm lysinR   \n",
       "2               A  BART-like domain of BARTL1/CCDC104 1-113   \n",
       "3               B                                   Bak BH3   \n",
       "4               B                          Crayfish Astacin   \n",
       "\n",
       "                         Receptor Name         KD(M)  ... PDB Release Date  \\\n",
       "0                      Neurexin-1-beta  1.320000e-07  ...       2009-09-08   \n",
       "1  red abalone egg VERL repeat 3 (VR3)  1.810000e-09  ...       2017-06-14   \n",
       "2                        Arl3FL-GppNHp  4.200000e-07  ...       2015-11-11   \n",
       "3                 Deerpox virus DPV022  6.980000e-06  ...       2015-08-05   \n",
       "4               Cleaved human fetuin-b  1.400000e-10  ...       2019-10-23   \n",
       "\n",
       "  Affinity PubMed ID Affinity Release Date  Subgroup  \\\n",
       "0           18093521           2007 Dec 20       NaN   \n",
       "1           28622512           2017 Jun 15       NaN   \n",
       "2           26455799            2015 Nov 3       NaN   \n",
       "3           26249341              2015 Aug       NaN   \n",
       "4           31604990           2019 Oct 11       NaN   \n",
       "\n",
       "                                    Ligand Sequences  \\\n",
       "0  AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...   \n",
       "1  AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...   \n",
       "2  AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...   \n",
       "3  AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...   \n",
       "4  AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...   \n",
       "\n",
       "                                  Receptor Sequences  \\\n",
       "0  HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...   \n",
       "1  FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...   \n",
       "2  LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...   \n",
       "3                             STMGQVGRQLAIIGDDINRRYD   \n",
       "4  ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...   \n",
       "\n",
       "                                              ligand  \\\n",
       "0  AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...   \n",
       "1  AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...   \n",
       "2  AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...   \n",
       "3  AAIEFDEIVKKLLNIYINDICTTGEKRLLNNYEKSILDRIYKSCEY...   \n",
       "4  AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...   \n",
       "\n",
       "                                            receptor       pKd fasta_id  \n",
       "0  HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...  6.879426   2WQZ_0  \n",
       "1  FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...  8.742321   5IIA_1  \n",
       "2  LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...  6.376751   4ZI2_2  \n",
       "3                             STMGQVGRQLAIIGDDINRRYD  5.156145   4UF1_3  \n",
       "4  ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...  9.853872   6SAZ_4  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fasta_id will be a unique identifier for each row to be used in fasta files\n",
    "# instead of complex id as it may contain spaces\n",
    "df[\"fasta_id\"] = df[\"PDB\"] + \"_\" + df.index.astype(str)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.makedirs(\"data_files\", exist_ok=True)\n",
    "with open(\"data_files/all_seqs.fasta\", \"w\") as f:\n",
    "    for i, row in df.iterrows():\n",
    "        fasta_id = row[\"fasta_id\"]\n",
    "        sequence = row[\"combined\"]\n",
    "        f.write(f\">{fasta_id}\\n{sequence}\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "we cluster all the sequnces into as many clusters as it takes such that for each cluster the minimum sequence identity between the representative of the cluster and any other sequence in the cluster is at least 10%. We use coverage of the smaller sequence with coverage of 20%. Such loose parameters are used to prevent ending up with thousands of clusters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data_files/all_seqs.fasta data_files/cluster tmp --min-seq-id 0.1 --cov-mode 5 -c 0.2 -s 10.0 --remove-tmp-files --max-iterations 50000 --cluster-reassign --cluster-steps 9 \n",
      "\n",
      "MMseqs Version:                     \tf5f780acd64482cd59b46eae0a107f763cd17b4d\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t10\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.2\n",
      "Coverage mode                       \t5\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t8\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.1\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t50000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t9\n",
      "Cluster reassign                    \ttrue\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Database type                       \t0\n",
      "Shuffle input database              \ttrue\n",
      "Createdb mode                       \t1\n",
      "Write lookup file                   \t0\n",
      "Offset of numeric ids               \t0\n",
      "\n",
      "createdb data_files/all_seqs.fasta tmp/15545192122171757246/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mShuffle database cannot be combined with --createdb-mode 0\n",
      "\u001b[39m\u001b[33mWe recompute with --shuffle 0\n",
      "\u001b[39mConverting sequences\n",
      "[8788] 0s 4ms\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\n",
      "Time for merging to input: 0h 0m 0s 0ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 5ms\n",
      "Create directory tmp/15545192122171757246/clu_tmp\n",
      "cluster tmp/15545192122171757246/input tmp/15545192122171757246/clu tmp/15545192122171757246/clu_tmp -s 10 --max-seqs 20 -c 0.2 --cov-mode 5 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.1 --max-iterations 50000 --cluster-steps 9 --cluster-reassign 1 --remove-tmp-files 0 \n",
      "\n",
      "Set cluster mode SET COVER\n",
      "linclust tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 5 -k 0 -c 0.2 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 5 -k 0 -c 0.2 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 8814 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 8.81K 0s 20ms     \n",
      "Sort kmer 0h 0m 0s 10ms\n",
      "Sort by rep. sequence 0h 0m 0s 6ms\n",
      "Time for fill: 0h 0m 0s 1ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 52ms\n",
      "rescorediagonal tmp/15545192122171757246/input tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 5 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 8.81K 0s 8ms      \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 4ms================>  ] 95.65% 8.43K eta 0s       \n",
      "Time for processing: 0h 0m 0s 57ms\n",
      "clust tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref_rescore1 tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pre_clust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 8.81K 0s 8ms                          ] 65.55% 5.78K eta 0s       \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 21150 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 8.81K 0s 11ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 8.81K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 45ms\n",
      "Total time: 0h 0m 0s 102ms\n",
      "\n",
      "Size of the sequence database: 8814\n",
      "Size of the alignment database: 8814\n",
      "Number of clusters: 1828\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 112ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/order_redundancy tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/order_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "filterdb tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref_filter1 tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref_filter2 --filter-file tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/order_redundancy --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 1.83K 0s 9ms      \n",
      "Time for merging to pref_filter2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 13ms\n",
      "rescorediagonal tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/input_step_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/input_step_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref_filter2 tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.2 -a 0 --cov-mode 5 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mCan not find any score per column for coverage 0.200000 and sequence identity 0.100000. No hit will be filtered.\n",
      "[=================================================================] 100.00% 1.83K 0s 4ms      \n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms==================>] 99.51% 1.82K eta 0s       \n",
      "Time for processing: 0h 0m 0s 15ms\n",
      "align tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/input_step_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/input_step_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pref_rescore2 tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1828 type: Aminoacid\n",
      "Target database size: 1828 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 1.83K 0s 849ms    \n",
      "Time for merging to aln: 0h 0m 0s 1ms\n",
      "5248 alignments calculated\n",
      "5248 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "2.870897 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 856ms\n",
      "clust tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/input_step_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/aln tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/clust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 1.83K 0s 9ms      ] 76.74% 1.40K eta 0s       \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3420 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 1.83K 0s 9ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 1.83K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 46ms\n",
      "Total time: 0h 0m 0s 68ms\n",
      "\n",
      "Size of the sequence database: 1828\n",
      "Size of the alignment database: 1828\n",
      "Number of clusters: 917\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 69ms\n",
      "mergeclusters tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/pre_clust tmp/15545192122171757246/clu_tmp/5318361636495855060/linclust/8072989200786753440/clust --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 1.83K 0s 8ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 917 0s 24ms     \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 8.81K 0s 37ms     \n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 43ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_redundancy tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 917 type: Aminoacid\n",
      "Estimated memory consumption: 981M\n",
      "Target database size: 917 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 917 0s 17ms     \n",
      "Index table: Masked residues: 8716\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 917 0s 9ms      \n",
      "Index statistics\n",
      "Entries:          175667\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002745\n",
      "Top 10 k-mers\n",
      "    YWQTTV\t39\n",
      "    FGGKIK\t32\n",
      "    SGPRGS\t32\n",
      "    WGGLVS\t31\n",
      "    VHFAQS\t30\n",
      "    DYGGVT\t30\n",
      "    SVCLFY\t26\n",
      "    SGGGFT\t23\n",
      "    YFEVSW\t23\n",
      "    RQPKEW\t22\n",
      "Time for index table init: 0h 0m 0s 417ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 917\n",
      "Target db start 1 to 917\n",
      "[=================================================================] 100.00% 917 0s 19ms     \n",
      "\n",
      "2.371524 k-mers per position\n",
      "336 DB matches per sequence\n",
      "0 overflows\n",
      "3 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 906ms\n",
      "align tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step0 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 917 type: Aminoacid\n",
      "Target database size: 917 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 917 0s 136ms    \n",
      "Time for merging to aln_step0: 0h 0m 0s 0ms\n",
      "3093 alignments calculated\n",
      "3081 sequence pairs passed the thresholds (0.996120 of overall calculated)\n",
      "3.359869 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 145ms\n",
      "clust tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step0 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step0 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 917 0s 4ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 454 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 917 0s 9ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 917 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 47ms\n",
      "Total time: 0h 0m 0s 85ms\n",
      "\n",
      "Size of the sequence database: 917\n",
      "Size of the alignment database: 917\n",
      "Number of clusters: 674\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 86ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step0 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step1 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step1 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2.125 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 674 type: Aminoacid\n",
      "Estimated memory consumption: 980M\n",
      "Target database size: 674 type: Aminoacid\n",
      "Index table k-mer threshold: 144 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 674 0s 25ms     \n",
      "Index table: Masked residues: 6995\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 674 0s 18ms     \n",
      "Index statistics\n",
      "Entries:          208617\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.003260\n",
      "Top 10 k-mers\n",
      "    ERKLEI\t4\n",
      "    WVQPGL\t4\n",
      "    YWQTTV\t4\n",
      "    PSSNAE\t3\n",
      "    AFEKDE\t3\n",
      "    DVRLAF\t3\n",
      "    GYGRNF\t3\n",
      "    SLLCSG\t3\n",
      "    RKFDTG\t3\n",
      "    IHDKNI\t3\n",
      "Time for index table init: 0h 0m 0s 444ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 144\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 674\n",
      "Target db start 1 to 674\n",
      "[=================================================================] 100.00% 674 0s 40ms     \n",
      "\n",
      "10.000324 k-mers per position\n",
      "356 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 967ms\n",
      "align tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step1 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step1 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step1 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 674 type: Aminoacid\n",
      "Target database size: 674 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 674 0s 21ms     \n",
      "Time for merging to aln_step1: 0h 0m 0s 0ms\n",
      "832 alignments calculated\n",
      "795 sequence pairs passed the thresholds (0.955529 of overall calculated)\n",
      "1.179525 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 27ms\n",
      "clust tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step1 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step1 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step1 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 674 0s 7ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 23 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 674 0s 5ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 674 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 36ms\n",
      "Total time: 0h 0m 0s 53ms\n",
      "\n",
      "Size of the sequence database: 674\n",
      "Size of the alignment database: 674\n",
      "Number of clusters: 614\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 55ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step1 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step1 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step2 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step2 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3.25 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 614 type: Aminoacid\n",
      "Estimated memory consumption: 980M\n",
      "Target database size: 614 type: Aminoacid\n",
      "Index table k-mer threshold: 134 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 614 0s 44ms     eta 0s       \n",
      "Index table: Masked residues: 6507\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 614 0s 18ms     \n",
      "Index statistics\n",
      "Entries:          216506\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.003383\n",
      "Top 10 k-mers\n",
      "    DPQALA\t3\n",
      "    VRLAAD\t3\n",
      "    PSSNAE\t3\n",
      "    AFEKDE\t3\n",
      "    RKFDTG\t3\n",
      "    ERKLEI\t3\n",
      "    QSNLAK\t3\n",
      "    ANLEKL\t3\n",
      "    TQLKKL\t3\n",
      "    NLAALN\t3\n",
      "Time for index table init: 0h 0m 0s 488ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 134\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 614\n",
      "Target db start 1 to 614\n",
      "[=================================================================] 100.00% 614 0s 85ms     \n",
      "\n",
      "31.894182 k-mers per position\n",
      "462 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 88ms\n",
      "align tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step2 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step2 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step2 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 614 type: Aminoacid\n",
      "Target database size: 614 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 614 0s 48ms     \n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\n",
      "775 alignments calculated\n",
      "677 sequence pairs passed the thresholds (0.873548 of overall calculated)\n",
      "1.102606 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 54ms\n",
      "clust tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step2 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step2 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step2 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 614 0s 10ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 27 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 614 0s 10ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 614 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 56ms\n",
      "Total time: 0h 0m 0s 74ms\n",
      "\n",
      "Size of the sequence database: 614\n",
      "Size of the alignment database: 614\n",
      "Number of clusters: 574\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 75ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step2 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step2 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step3 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step3: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step3 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step3 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 4.375 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 574 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 574 type: Aminoacid\n",
      "Index table k-mer threshold: 124 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 574 0s 34ms     \n",
      "Index table: Masked residues: 5371\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 574 0s 59ms     \n",
      "Index statistics\n",
      "Entries:          210757\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.003293\n",
      "Top 10 k-mers\n",
      "    AEGLAA\t3\n",
      "    VRLAAD\t3\n",
      "    ELLLLD\t3\n",
      "    AFEKDE\t3\n",
      "    RKFDTG\t3\n",
      "    ELKLIL\t3\n",
      "    TQLKKL\t3\n",
      "    RGKLKQ\t3\n",
      "    GAGLLS\t3\n",
      "    LSLGSV\t3\n",
      "Time for index table init: 0h 0m 0s 584ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 124\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 574\n",
      "Target db start 1 to 574\n",
      "[=================================================================] 100.00% 574 0s 187ms    \n",
      "\n",
      "97.857620 k-mers per position\n",
      "573 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step3: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 266ms\n",
      "align tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step3 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step3 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step3 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 574 type: Aminoacid\n",
      "Target database size: 574 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 574 0s 19ms     \n",
      "Time for merging to aln_step3: 0h 0m 0s 0ms\n",
      "976 alignments calculated\n",
      "627 sequence pairs passed the thresholds (0.642418 of overall calculated)\n",
      "1.092335 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 25ms\n",
      "clust tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step3 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step3 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step3 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 574 0s 6ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 13 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 574 0s 4ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 574 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 32ms\n",
      "Total time: 0h 0m 0s 50ms\n",
      "\n",
      "Size of the sequence database: 574\n",
      "Size of the alignment database: 574\n",
      "Number of clusters: 544\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step3: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 51ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step3 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step3 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step4 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step4: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step4 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step4 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step4 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5.5 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 544 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 544 type: Aminoacid\n",
      "Index table k-mer threshold: 114 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 544 0s 19ms     \n",
      "Index table: Masked residues: 4959\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 544 0s 16ms     \n",
      "Index statistics\n",
      "Entries:          197613\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.003088\n",
      "Top 10 k-mers\n",
      "    AEGLAA\t3\n",
      "    ELLLLD\t3\n",
      "    AFEKDE\t3\n",
      "    RKFDTG\t3\n",
      "    ELKLIL\t3\n",
      "    TQLKKL\t3\n",
      "    RGKLKQ\t3\n",
      "    GAGLLS\t3\n",
      "    LSLGSV\t3\n",
      "    SSEQAA\t2\n",
      "Time for index table init: 0h 0m 0s 430ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 114\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 544\n",
      "Target db start 1 to 544\n",
      "[=================================================================] 100.00% 544 0s 464ms    \n",
      "\n",
      "284.100478 k-mers per position\n",
      "810 DB matches per sequence\n",
      "0 overflows\n",
      "3 sequences passed prefiltering per query sequence\n",
      "3 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step4: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 430ms\n",
      "align tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step4 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step4 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step4 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step4 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 544 type: Aminoacid\n",
      "Target database size: 544 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 544 0s 25ms     \n",
      "Time for merging to aln_step4: 0h 0m 0s 0ms\n",
      "1767 alignments calculated\n",
      "607 sequence pairs passed the thresholds (0.343520 of overall calculated)\n",
      "1.115809 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 35ms\n",
      "clust tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step4 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step4 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step4 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 544 0s 4ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 21 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 544 0s 7ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 544 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 45ms\n",
      "Total time: 0h 0m 0s 65ms\n",
      "\n",
      "Size of the sequence database: 544\n",
      "Size of the alignment database: 544\n",
      "Number of clusters: 504\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step4: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 67ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step4 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step4 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step5 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step5: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step5 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step5 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step5 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 6.625 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 504 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 504 type: Aminoacid\n",
      "Index table k-mer threshold: 104 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 504 0s 23ms     \n",
      "Index table: Masked residues: 4333\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 504 0s 16ms     \n",
      "Index statistics\n",
      "Entries:          181997\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002844\n",
      "Top 10 k-mers\n",
      "    AEGLAA\t3\n",
      "    ELLLLD\t3\n",
      "    TQLKKL\t3\n",
      "    RGKLKQ\t3\n",
      "    GAGLLS\t3\n",
      "    WASLAA\t2\n",
      "    SSEQAA\t2\n",
      "    LKNQAA\t2\n",
      "    EGQQAA\t2\n",
      "    IQLRAA\t2\n",
      "Time for index table init: 0h 0m 0s 444ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 104\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 504\n",
      "Target db start 1 to 504\n",
      "[=================================================================] 100.00% 504 1s 50ms     \n",
      "\n",
      "787.295927 k-mers per position\n",
      "1437 DB matches per sequence\n",
      "0 overflows\n",
      "9 sequences passed prefiltering per query sequence\n",
      "8 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step5: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 2s 11ms\n",
      "align tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step5 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step5 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step5 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step5 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 504 type: Aminoacid\n",
      "Target database size: 504 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 504 0s 45ms     \n",
      "Time for merging to aln_step5: 0h 0m 0s 0ms\n",
      "4267 alignments calculated\n",
      "551 sequence pairs passed the thresholds (0.129131 of overall calculated)\n",
      "1.093254 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 54ms\n",
      "clust tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step5 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step5 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step5 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 504 0s 7ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 15 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 504 0s 2ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 504 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 39ms\n",
      "Total time: 0h 0m 0s 53ms\n",
      "\n",
      "Size of the sequence database: 504\n",
      "Size of the alignment database: 504\n",
      "Number of clusters: 474\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step5: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 54ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step5 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step5 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step6 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step6: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step6 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step6 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step6 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 7.75 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 474 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 474 type: Aminoacid\n",
      "Index table k-mer threshold: 94 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 474 0s 22ms     \n",
      "Index table: Masked residues: 3773\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 474 0s 20ms     \n",
      "Index statistics\n",
      "Entries:          170928\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002671\n",
      "Top 10 k-mers\n",
      "    ELLLLD\t3\n",
      "    TQLKKL\t3\n",
      "    RGKLKQ\t3\n",
      "    GAGLLS\t3\n",
      "    AEGLAA\t2\n",
      "    GGILAA\t2\n",
      "    WASLAA\t2\n",
      "    SSEQAA\t2\n",
      "    LKNQAA\t2\n",
      "    EGQQAA\t2\n",
      "Time for index table init: 0h 0m 0s 440ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 94\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 474\n",
      "Target db start 1 to 474\n",
      "[=================================================================] 100.00% 474 2s 542ms    \n",
      "\n",
      "2101.096293 k-mers per position\n",
      "3075 DB matches per sequence\n",
      "0 overflows\n",
      "17 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step6: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 3s 618ms\n",
      "align tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step6 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step6 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step6 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step6 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 474 type: Aminoacid\n",
      "Target database size: 474 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 474 0s 86ms     \n",
      "Time for merging to aln_step6: 0h 0m 0s 4ms\n",
      "7269 alignments calculated\n",
      "492 sequence pairs passed the thresholds (0.067685 of overall calculated)\n",
      "1.037975 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 97ms\n",
      "clust tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step6 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step6 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step6 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 474 0s 10ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 8 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 474 0s 13ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 474 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 77ms\n",
      "Total time: 0h 0m 0s 105ms\n",
      "\n",
      "Size of the sequence database: 474\n",
      "Size of the alignment database: 474\n",
      "Number of clusters: 461\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step6: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 108ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step6 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step6 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step7 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step7: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step7 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step7 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step7 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 8.875 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 461 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 461 type: Aminoacid\n",
      "Index table k-mer threshold: 84 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 461 0s 28ms     \n",
      "Index table: Masked residues: 3453\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 461 0s 18ms     \n",
      "Index statistics\n",
      "Entries:          164063\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002563\n",
      "Top 10 k-mers\n",
      "    TQLKKL\t3\n",
      "    RGKLKQ\t3\n",
      "    GAGLLS\t3\n",
      "    AEGLAA\t2\n",
      "    GGILAA\t2\n",
      "    SSEQAA\t2\n",
      "    LKNQAA\t2\n",
      "    EGQQAA\t2\n",
      "    IQLRAA\t2\n",
      "    GVYTCA\t2\n",
      "Time for index table init: 0h 0m 0s 603ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 84\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 461\n",
      "Target db start 1 to 461\n",
      "[=================================================================] 100.00% 461 6s 715ms    \n",
      "\n",
      "5335.674547 k-mers per position\n",
      "7138 DB matches per sequence\n",
      "0 overflows\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step7: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 8s 203ms\n",
      "align tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step7 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step7 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step7 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step7 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 461 type: Aminoacid\n",
      "Target database size: 461 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 461 0s 87ms     \n",
      "Time for merging to aln_step7: 0h 0m 0s 4ms\n",
      "7924 alignments calculated\n",
      "473 sequence pairs passed the thresholds (0.059692 of overall calculated)\n",
      "1.026030 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 97ms\n",
      "clust tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step7 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step7 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step7 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 461 0s 11ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 2 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 461 0s 7ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 461 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 63ms\n",
      "Total time: 0h 0m 0s 81ms\n",
      "\n",
      "Size of the sequence database: 461\n",
      "Size of the alignment database: 461\n",
      "Number of clusters: 454\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step7: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 83ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step7 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step7 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step8 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step8: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step8 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step8 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step8 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 10 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 454 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 454 type: Aminoacid\n",
      "Index table k-mer threshold: 74 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 454 0s 20ms     \n",
      "Index table: Masked residues: 3178\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 454 0s 16ms     \n",
      "Index statistics\n",
      "Entries:          159957\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002499\n",
      "Top 10 k-mers\n",
      "    TQLKKL\t3\n",
      "    RGKLKQ\t3\n",
      "    GAGLLS\t3\n",
      "    GGILAA\t2\n",
      "    SSEQAA\t2\n",
      "    LKNQAA\t2\n",
      "    EGQQAA\t2\n",
      "    IQLRAA\t2\n",
      "    GVYTCA\t2\n",
      "    AELDDA\t2\n",
      "Time for index table init: 0h 0m 0s 540ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 74\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 454\n",
      "Target db start 1 to 454\n",
      "[=================================================================] 100.00% 454 14s 177ms    \n",
      "\n",
      "12994.610686 k-mers per position\n",
      "16886 DB matches per sequence\n",
      "0 overflows\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step8: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 15s 824ms\n",
      "align tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step8 tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step8 tmp/15545192122171757246/clu_tmp/5318361636495855060/pref_step8 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step8 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 454 type: Aminoacid\n",
      "Target database size: 454 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 454 0s 77ms     \n",
      "Time for merging to aln_step8: 0h 0m 0s 0ms\n",
      "7956 alignments calculated\n",
      "460 sequence pairs passed the thresholds (0.057818 of overall calculated)\n",
      "1.013216 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 106ms\n",
      "clust tmp/15545192122171757246/clu_tmp/5318361636495855060/input_step8 tmp/15545192122171757246/clu_tmp/5318361636495855060/aln_step8 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step8 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 454 0s 8ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 2 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 454 0s 5ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 454 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 35ms\n",
      "Total time: 0h 0m 0s 41ms\n",
      "\n",
      "Size of the sequence database: 454\n",
      "Size of the alignment database: 454\n",
      "Number of clusters: 450\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step8: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 43ms\n",
      "mergeclusters tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/clu tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_redundancy tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step0 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step1 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step2 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step3 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step4 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step5 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step6 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step7 tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_step8 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 917 0s 5ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 674 0s 20ms     \n",
      "Clustering step 3\n",
      "[=================================================================] 100.00% 614 0s 35ms     \n",
      "Clustering step 4\n",
      "[=================================================================] 100.00% 574 0s 53ms     \n",
      "Clustering step 5\n",
      "[=================================================================] 100.00% 544 0s 65ms     \n",
      "Clustering step 6\n",
      "[=================================================================] 100.00% 504 0s 72ms     \n",
      "Clustering step 7\n",
      "[=================================================================] 100.00% 474 0s 80ms     \n",
      "Clustering step 8\n",
      "[=================================================================] 100.00% 461 0s 93ms     \n",
      "Clustering step 9\n",
      "[=================================================================] 100.00% 454 0s 107ms    \n",
      "Clustering step 10\n",
      "[=================================================================] 100.00% 450 0s 119ms    \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 8.81K 0s 143ms    \n",
      "Time for merging to clu: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 155ms\n",
      "align tmp/15545192122171757246/input tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/clu tmp/15545192122171757246/clu_tmp/5318361636495855060/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 8814 type: Aminoacid\n",
      "Target database size: 8814 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 450 0s 725ms    \n",
      "Time for merging to aln: 0h 0m 0s 3ms\n",
      "8602 alignments calculated\n",
      "6340 sequence pairs passed the thresholds (0.737038 of overall calculated)\n",
      "14.088889 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 744ms\n",
      "subtractdbs tmp/15545192122171757246/clu_tmp/5318361636495855060/clu tmp/15545192122171757246/clu_tmp/5318361636495855060/aln tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_not_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "subtractdbs tmp/15545192122171757246/clu_tmp/5318361636495855060/clu tmp/15545192122171757246/clu_tmp/5318361636495855060/aln tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_not_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Remove tmp/15545192122171757246/clu_tmp/5318361636495855060/aln ids from tmp/15545192122171757246/clu_tmp/5318361636495855060/clu\n",
      "[=================================================================] 100.00% 450 0s 5ms      \n",
      "Time for merging to clu_not_accepted: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 9ms\n",
      "swapdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_not_accepted tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_not_accepted_swap --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 450 0s 5ms      \n",
      "Computing offsets.\n",
      "[=================================================================] 100.00% 450 0s 3ms      \n",
      "\n",
      "Reading results.\n",
      "[=================================================================] 100.00% 450 0s 7ms      \n",
      "\n",
      "Output database: tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_not_accepted_swap\n",
      "[=================================================================] 100.00% 8.81K 0s 10ms     \n",
      "\n",
      "Time for merging to clu_not_accepted_swap: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 50ms\n",
      "subtractdbs tmp/15545192122171757246/clu_tmp/5318361636495855060/clu tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_not_accepted tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "subtractdbs tmp/15545192122171757246/clu_tmp/5318361636495855060/clu tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_not_accepted tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Remove tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_not_accepted ids from tmp/15545192122171757246/clu_tmp/5318361636495855060/clu\n",
      "[=================================================================] 100.00% 450 0s 6ms      \n",
      "Time for merging to clu_accepted: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 11ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_not_accepted_swap tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned -v 3 \n",
      "\n",
      "Time for merging to seq_wrong_assigned: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "createsubdb tmp/15545192122171757246/clu_tmp/5318361636495855060/clu tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_seeds -v 3 \n",
      "\n",
      "Time for merging to seq_seeds: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_seeds.merged tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned_pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 10 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2474 type: Aminoacid\n",
      "Estimated memory consumption: 990M\n",
      "Target database size: 2924 type: Aminoacid\n",
      "Index table k-mer threshold: 74 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 2.92K 0s 86ms     \n",
      "Index table: Masked residues: 12443\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 2.92K 0s 56ms     \n",
      "Index statistics\n",
      "Entries:          1483219\n",
      "DB size:          496 MB\n",
      "Avg k-mer size:   0.023175\n",
      "Top 10 k-mers\n",
      "    FPPTWN\t808\n",
      "    VHFAQS\t808\n",
      "    PEVVNS\t775\n",
      "    VKYPVT\t764\n",
      "    WNGLGV\t750\n",
      "    LYLSTV\t730\n",
      "    YWQTTV\t612\n",
      "    RQPKEW\t587\n",
      "    AAGLDY\t551\n",
      "    QLEGLV\t549\n",
      "Time for index table init: 0h 0m 0s 641ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 74\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2474\n",
      "Target db start 1 to 2924\n",
      "[=================================================================] 100.00% 2.47K 1m 47s 718ms   \n",
      "\n",
      "14513.963957 k-mers per position\n",
      "323309 DB matches per sequence\n",
      "0 overflows\n",
      "20 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to seq_wrong_assigned_pref: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 1m 49s 376ms\n",
      "swapdb tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned_pref tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned_pref_swaped --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 2.47K 0s 6ms      \n",
      "Computing offsets.\n",
      "[=================================================================] 100.00% 2.47K 0s 9ms      \n",
      "\n",
      "Reading results.\n",
      "[=================================================================] 100.00% 2.47K 0s 6ms                     ] 50.91% 1.26K eta 0s       \n",
      "\n",
      "Output database: tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned_pref_swaped\n",
      "[=================================================================] 100.00% 8.81K 0s 10ms     \n",
      "\n",
      "Time for merging to seq_wrong_assigned_pref_swaped: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 68ms\n",
      "align tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_seeds.merged tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned_pref_swaped tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned_pref_swaped_aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 2924 type: Aminoacid\n",
      "Target database size: 2474 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 2.70K 8s 532ms    \n",
      "Time for merging to seq_wrong_assigned_pref_swaped_aln: 0h 0m 0s 1ms\n",
      "49100 alignments calculated\n",
      "46183 sequence pairs passed the thresholds (0.940591 of overall calculated)\n",
      "17.111153 hits per query sequence\n",
      "Time for processing: 0h 0m 8s 554ms\n",
      "filterdb tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned_pref_swaped_aln tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned_pref_swaped_aln_ocol --trim-to-one-column --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using regular expression\n",
      "[=================================================================] 100.00% 2.70K 0s 33ms     \n",
      "Time for merging to seq_wrong_assigned_pref_swaped_aln_ocol: 0h 0m 0s 4ms\n",
      "Time for processing: 0h 0m 0s 47ms\n",
      "mergedbs tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_seeds.merged tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_accepted_plus_wrong tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_accepted tmp/15545192122171757246/clu_tmp/5318361636495855060/seq_wrong_assigned_pref_swaped_aln_ocol --merge-stop-empty 0 --compressed 0 -v 3 \n",
      "\n",
      "Merging the results to tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_accepted_plus_wrong\n",
      "[=================================================================] 100.00% 2.92K 0s 1ms      \n",
      "Time for merging to clu_accepted_plus_wrong: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "tsv2db tmp/15545192122171757246/clu_tmp/5318361636495855060/missing.single.seqs tmp/15545192122171757246/clu_tmp/5318361636495855060/missing.single.seqs.db --output-dbtype 6 --compressed 0 -v 3 \n",
      "\n",
      "Output database type: Clustering\n",
      "Time for merging to missing.single.seqs.db: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "mergedbs tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_accepted_plus_wrong_plus_single tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_accepted_plus_wrong tmp/15545192122171757246/clu_tmp/5318361636495855060/missing.single.seqs.db --merge-stop-empty 0 --compressed 0 -v 3 \n",
      "\n",
      "Merging the results to tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_accepted_plus_wrong_plus_single\n",
      "[=================================================================] 100.00% 8.81K 0s 2ms      \n",
      "Time for merging to clu_accepted_plus_wrong_plus_single: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 4ms\n",
      "clust tmp/15545192122171757246/input tmp/15545192122171757246/clu_tmp/5318361636495855060/clu_accepted_plus_wrong_plus_single tmp/15545192122171757246/clu --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 8.81K 0s 10ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 27655 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 8.81K 0s 12ms     =================>             ] 79.66% 7.02K eta 0s       \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 8.81K 0s 2ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 65ms\n",
      "Total time: 0h 0m 0s 116ms\n",
      "\n",
      "Size of the sequence database: 8814\n",
      "Size of the alignment database: 8814\n",
      "Number of clusters: 632\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 154ms\n",
      "createtsv tmp/15545192122171757246/input tmp/15545192122171757246/input tmp/15545192122171757246/clu tmp/15545192122171757246/cluster.tsv --threads 8 -v 3 \n",
      "\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 18ms\n",
      "result2repseq tmp/15545192122171757246/input tmp/15545192122171757246/clu tmp/15545192122171757246/clu_rep --db-load-mode 0 --compressed 0 --threads 8 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 632 0s 9ms      \n",
      "Time for merging to clu_rep: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 18ms\n",
      "result2flat tmp/15545192122171757246/input tmp/15545192122171757246/input tmp/15545192122171757246/clu_rep tmp/15545192122171757246/rep_seq.fasta --use-fasta-header -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "createseqfiledb tmp/15545192122171757246/input tmp/15545192122171757246/clu tmp/15545192122171757246/clu_seqs --threads 8 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 632 0s 5ms      \n",
      "Time for merging to clu_seqs: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 16ms\n",
      "result2flat tmp/15545192122171757246/input tmp/15545192122171757246/input tmp/15545192122171757246/clu_seqs tmp/15545192122171757246/all_seqs.fasta -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 13ms\n"
     ]
    }
   ],
   "source": [
    "!mmseqs easy-cluster data_files/all_seqs.fasta data_files/cluster tmp --min-seq-id 0.1 --cov-mode 5 -c 0.2 -s 10.0 --remove-tmp-files --max-iterations 50000 --cluster-reassign --cluster-steps 9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_rep_id</th>\n",
       "      <th>fasta_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5IIA_1</td>\n",
       "      <td>5IIA_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4ETP_9</td>\n",
       "      <td>4ETP_9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4ETP_9</td>\n",
       "      <td>3DCO_7071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6IMF_26</td>\n",
       "      <td>6IMF_26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4EQA_30</td>\n",
       "      <td>4EQA_30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8809</th>\n",
       "      <td>3TO4_8796</td>\n",
       "      <td>3QUX_5860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8810</th>\n",
       "      <td>3TO4_8796</td>\n",
       "      <td>4ELM_6458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8811</th>\n",
       "      <td>3TO4_8796</td>\n",
       "      <td>4P2O_6580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8812</th>\n",
       "      <td>3TO4_8796</td>\n",
       "      <td>3TVM_8210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8813</th>\n",
       "      <td>3TO4_8796</td>\n",
       "      <td>3HE7_8794</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8814 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     cluster_rep_id   fasta_id\n",
       "0            5IIA_1     5IIA_1\n",
       "1            4ETP_9     4ETP_9\n",
       "2            4ETP_9  3DCO_7071\n",
       "3           6IMF_26    6IMF_26\n",
       "4           4EQA_30    4EQA_30\n",
       "...             ...        ...\n",
       "8809      3TO4_8796  3QUX_5860\n",
       "8810      3TO4_8796  4ELM_6458\n",
       "8811      3TO4_8796  4P2O_6580\n",
       "8812      3TO4_8796  3TVM_8210\n",
       "8813      3TO4_8796  3HE7_8794\n",
       "\n",
       "[8814 rows x 2 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_df = pd.read_csv(\n",
    "    \"data_files/cluster_cluster.tsv\", sep=\"\\t\", header=None\n",
    ")\n",
    "cluster_df = cluster_df.rename(columns={0: \"cluster_rep_id\", 1: \"fasta_id\"})\n",
    "cluster_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_rep_id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6UMT_4919</th>\n",
       "      <td>950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1FLE_8116</th>\n",
       "      <td>607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2FU5_4850</th>\n",
       "      <td>380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3WQB_7909</th>\n",
       "      <td>334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3QHY_8082</th>\n",
       "      <td>318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4A94_8509</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3ZWL_7069</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3ZWL_7068</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1AVA_62</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3WN7_5160</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>632 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                cluster_size\n",
       "cluster_rep_id              \n",
       "6UMT_4919                950\n",
       "1FLE_8116                607\n",
       "2FU5_4850                380\n",
       "3WQB_7909                334\n",
       "3QHY_8082                318\n",
       "...                      ...\n",
       "4A94_8509                  1\n",
       "3ZWL_7069                  1\n",
       "3ZWL_7068                  1\n",
       "1AVA_62                    1\n",
       "3WN7_5160                  1\n",
       "\n",
       "[632 rows x 1 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_df_count = cluster_df.groupby([\"cluster_rep_id\"]).count().sort_values(\n",
    "    \"fasta_id\", ascending=False\n",
    ").rename(columns={\"fasta_id\": \"cluster_size\"})\n",
    "cluster_df_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "split\n",
       "train    0.6\n",
       "test     0.4\n",
       "Name: count, dtype: float64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_fasta_ids = []\n",
    "# Ultimatley, we want 80% of the data to be in the training set, but we know\n",
    "# that we will need to remove some sequences from test to training later to\n",
    "# ensure that the training set is not too similar to the test set. So we will\n",
    "# aim for 60% in the training set for now.\n",
    "target_train_count = round(0.6 * len(df))\n",
    "\n",
    "for cluster_rep_id, cluster_size in cluster_df_count.iterrows():\n",
    "    # add largest cluster first to the training set\n",
    "    train_fasta_ids.extend(\n",
    "        cluster_df[cluster_df[\"cluster_rep_id\"] == cluster_rep_id][\n",
    "            \"fasta_id\"\n",
    "        ].tolist()\n",
    "    )\n",
    "    if len(train_fasta_ids) >= target_train_count:\n",
    "        break\n",
    "\n",
    "df[\"split\"] = df[\"fasta_id\"].apply(\n",
    "    lambda x: \"train\" if x in train_fasta_ids else \"test\"\n",
    ")\n",
    "(df[\"split\"].value_counts() / len(df)).round(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "At this point the train and test splits should be slightly dissimilar to each other. But, we want to guarantee minimal similarity, so we will perform a series of iterations to ensure that the splits are as dissimilar as possible. In each iteration we do the following:\n",
    "1. Compute the pairwise sequence identity between the train and test splits\n",
    "2. Any sequence in the test split that has a sequence identity of at least 30% with any sequence in the train split and the coverage of the smaller sequence is at least 80%, is moved from the test split to the train split to prevent data leakage.\n",
    "3. Repeat from step 1 until no sequence is moved from the test split to the train split."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "from subprocess import run\n",
    "\n",
    "\n",
    "def do_split_iteration(df, splits):\n",
    "    split1, split2 = splits\n",
    "    with open(f\"data_files/{split1}.fasta\", \"w\") as f:\n",
    "        for _, row in df[df[\"split\"] == split1].iterrows():\n",
    "            fasta_id = row[\"fasta_id\"]\n",
    "            seq = row[\"combined\"]\n",
    "            f.write(f\">{fasta_id}\\n{seq}\\n\")\n",
    "\n",
    "    with open(f\"data_files/{split2}.fasta\", \"w\") as f:\n",
    "        for _, row in df[df[\"split\"] == split2].iterrows():\n",
    "            fasta_id = row[\"fasta_id\"]\n",
    "            seq = row[\"combined\"]\n",
    "            f.write(f\">{fasta_id}\\n{seq}\\n\")\n",
    "\n",
    "    command = [\n",
    "        \"mmseqs\",\n",
    "        \"easy-search\",\n",
    "        f\"data_files/{split1}.fasta\",\n",
    "        f\"data_files/{split2}.fasta\",\n",
    "        \"data_files/matches.tsv\",\n",
    "        \"tmp\",\n",
    "        \"-s\",\n",
    "        \"7.5\",\n",
    "        \"--min-seq-id\",\n",
    "        \"0.3\",\n",
    "        \"--cov-mode\",\n",
    "        \"5\",\n",
    "        \"-c\",\n",
    "        \"0.8\",\n",
    "    ]\n",
    "\n",
    "    result = run(command, capture_output=True, text=True)\n",
    "    if result.returncode != 0:\n",
    "        print(result.stderr)\n",
    "        raise ValueError(\"Error running mmseqs\")\n",
    "    try:\n",
    "        matches_df = pd.read_csv(\n",
    "            \"data_files/matches.tsv\", sep=\"\\t\", header=None\n",
    "        )\n",
    "    except pd.errors.EmptyDataError:\n",
    "        print(\"No matches found\")\n",
    "        return df, 0\n",
    "    matches_df = matches_df.rename(\n",
    "        columns={0: f\"{split1}_id\", 1: f\"{split2}_id\", 2: \"pident\"}\n",
    "    )\n",
    "    fasta_ids_to_change_split = matches_df[f\"{split2}_id\"].unique()\n",
    "    print(\n",
    "        f\"Changing {len(fasta_ids_to_change_split)} sequences from {split2} \"\n",
    "        f\"to {split1}\"\n",
    "    )\n",
    "    df[\"split\"] = df.apply(\n",
    "        lambda row: (\n",
    "            split1\n",
    "            if row[\"fasta_id\"] in fasta_ids_to_change_split\n",
    "            else row[\"split\"]\n",
    "        ),\n",
    "        axis=1,\n",
    "    )\n",
    "    return df, len(fasta_ids_to_change_split)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Changing 1113 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.73\n",
      "test     0.27\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 268 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.76\n",
      "test     0.24\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 95 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.77\n",
      "test     0.23\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 120 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.78\n",
      "test     0.22\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 14 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.78\n",
      "test     0.22\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 4 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.78\n",
      "test     0.22\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 1 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.78\n",
      "test     0.22\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "No matches found\n",
      "Current split ratios: split\n",
      "train    0.78\n",
      "test     0.22\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n"
     ]
    }
   ],
   "source": [
    "while True:\n",
    "    df, total_changed = do_split_iteration(df, (\"train\", \"test\"))\n",
    "    print(\n",
    "        f\"Current split ratios: {(df['split'].value_counts() / len(df)).round(2)}\"\n",
    "    )\n",
    "    print(\"*\" * 80)\n",
    "    if (df[\"split\"] == \"test\").sum() == 0 or total_changed == 0:\n",
    "        break"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We want to further split the test split into a validation and test split. We will use the same procedure and parameters as above to ensure that the validation and test splits are as dissimilar as possible."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"data_files/all_test_seqs.fasta\", \"w\") as f:\n",
    "    for i, row in df[df[\"split\"] == \"test\"].iterrows():\n",
    "        fasta_id = row[\"fasta_id\"]\n",
    "        sequence = row[\"combined\"]\n",
    "        f.write(f\">{fasta_id}\\n{sequence}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data_files/all_test_seqs.fasta data_files/cluster tmp --min-seq-id 0.1 --cov-mode 5 -c 0.2 -s 10.0 --remove-tmp-files --max-iterations 50000 --cluster-reassign --cluster-steps 9 \n",
      "\n",
      "MMseqs Version:                     \tf5f780acd64482cd59b46eae0a107f763cd17b4d\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t10\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.2\n",
      "Coverage mode                       \t5\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t8\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.1\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t50000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t9\n",
      "Cluster reassign                    \ttrue\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Database type                       \t0\n",
      "Shuffle input database              \ttrue\n",
      "Createdb mode                       \t1\n",
      "Write lookup file                   \t0\n",
      "Offset of numeric ids               \t0\n",
      "\n",
      "createdb data_files/all_test_seqs.fasta tmp/6478405549615959369/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mShuffle database cannot be combined with --createdb-mode 0\n",
      "\u001b[39m\u001b[33mWe recompute with --shuffle 0\n",
      "\u001b[39mConverting sequences\n",
      "[1819] 0s 0ms\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\n",
      "Time for merging to input: 0h 0m 0s 0ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Create directory tmp/6478405549615959369/clu_tmp\n",
      "cluster tmp/6478405549615959369/input tmp/6478405549615959369/clu tmp/6478405549615959369/clu_tmp -s 10 --max-seqs 20 -c 0.2 --cov-mode 5 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.1 --max-iterations 50000 --cluster-steps 9 --cluster-reassign 1 --remove-tmp-files 0 \n",
      "\n",
      "Set cluster mode SET COVER\n",
      "linclust tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 0 --force-reuse 0 \n",
      "\n",
      "kmermatcher tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 5 -k 0 -c 0.2 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.1 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 5 -k 0 -c 0.2 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 1908 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 1.91K 0s 10ms     \n",
      "Sort kmer 0h 0m 0s 2ms\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\n",
      "Time for fill: 0h 0m 0s 0ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 28ms\n",
      "rescorediagonal tmp/6478405549615959369/input tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.5 -a 0 --cov-mode 5 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 1.91K 0s 3ms      \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 10ms\n",
      "clust tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref_rescore1 tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pre_clust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 1.91K 0s 6ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 2215 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 1.91K 0s 7ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 1.91K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 26ms\n",
      "Total time: 0h 0m 0s 59ms\n",
      "\n",
      "Size of the sequence database: 1908\n",
      "Size of the alignment database: 1908\n",
      "Number of clusters: 725\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 60ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/order_redundancy tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/order_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "filterdb tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref_filter1 tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref_filter2 --filter-file tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/order_redundancy --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 725 0s 5ms      \n",
      "Time for merging to pref_filter2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 10ms\n",
      "rescorediagonal tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/input_step_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/input_step_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref_filter2 tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.2 -a 0 --cov-mode 5 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mCan not find any score per column for coverage 0.200000 and sequence identity 0.100000. No hit will be filtered.\n",
      "[=================================================================] 100.00% 725 0s 4ms      \n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms==================>] 99.59% 722 eta 0s       \n",
      "Time for processing: 0h 0m 0s 8ms\n",
      "align tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/input_step_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/input_step_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pref_rescore2 tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 725 type: Aminoacid\n",
      "Target database size: 725 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 725 0s 36ms     \n",
      "Time for merging to aln: 0h 0m 0s 0ms\n",
      "841 alignments calculated\n",
      "841 sequence pairs passed the thresholds (1.000000 of overall calculated)\n",
      "1.160000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 39ms\n",
      "clust tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/input_step_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/aln tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/clust --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 725 0s 6ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 116 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 725 0s 7ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 725 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 50ms\n",
      "Total time: 0h 0m 0s 66ms\n",
      "\n",
      "Size of the sequence database: 725\n",
      "Size of the alignment database: 725\n",
      "Number of clusters: 622\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 67ms\n",
      "mergeclusters tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/pre_clust tmp/6478405549615959369/clu_tmp/2715299047738836632/linclust/4735513278985964044/clust --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 725 0s 6ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 622 0s 14ms     \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 1.91K 0s 23ms     \n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 27ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_redundancy tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 622 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 622 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 622 0s 15ms     \n",
      "Index table: Masked residues: 4500\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 622 0s 14ms     \n",
      "Index statistics\n",
      "Entries:          110230\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.001722\n",
      "Top 10 k-mers\n",
      "    RKFDTG\t4\n",
      "    CNDVAK\t4\n",
      "    AFNDIR\t4\n",
      "    ITCFPW\t4\n",
      "    TPMSGY\t4\n",
      "    LHTCVA\t3\n",
      "    VYPMVC\t3\n",
      "    GFAGAD\t3\n",
      "    GRLIDD\t3\n",
      "    FHYGGD\t3\n",
      "Time for index table init: 0h 0m 0s 406ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 622\n",
      "Target db start 1 to 622\n",
      "[=================================================================] 100.00% 622 0s 9ms      \n",
      "[=======================================================>         ] 85.83% 534 eta 0s       \n",
      "2.135087 k-mers per position\n",
      "225 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 931ms\n",
      "align tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step0 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 622 type: Aminoacid\n",
      "Target database size: 622 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 622 0s 18ms     \n",
      "Time for merging to aln_step0: 0h 0m 0s 0ms\n",
      "763 alignments calculated\n",
      "761 sequence pairs passed the thresholds (0.997379 of overall calculated)\n",
      "1.223473 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 26ms\n",
      "clust tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step0 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step0 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 622 0s 8ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 1 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 622 0s 7ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 622 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 45ms\n",
      "Total time: 0h 0m 0s 63ms\n",
      "\n",
      "Size of the sequence database: 622\n",
      "Size of the alignment database: 622\n",
      "Number of clusters: 559\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 64ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step0 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step1 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step1 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2.125 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 559 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 559 type: Aminoacid\n",
      "Index table k-mer threshold: 144 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 559 0s 20ms     \n",
      "Index table: Masked residues: 3231\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 559 0s 13ms     \n",
      "Index statistics\n",
      "Entries:          163940\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002562\n",
      "Top 10 k-mers\n",
      "    DPQALA\t3\n",
      "    PSSNAE\t3\n",
      "    VGVPDE\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    APDQLY\t3\n",
      "    QQNAEA\t2\n",
      "    GSEAFA\t2\n",
      "    NFKLFA\t2\n",
      "    QQPLFA\t2\n",
      "Time for index table init: 0h 0m 0s 430ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 144\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 559\n",
      "Target db start 1 to 559\n",
      "[=================================================================] 100.00% 559 0s 28ms     \n",
      "\n",
      "9.598424 k-mers per position\n",
      "324 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 998ms\n",
      "align tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step1 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step1 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step1 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 559 type: Aminoacid\n",
      "Target database size: 559 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 559 0s 14ms     \n",
      "Time for merging to aln_step1: 0h 0m 0s 0ms\n",
      "630 alignments calculated\n",
      "609 sequence pairs passed the thresholds (0.966667 of overall calculated)\n",
      "1.089445 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 19ms\n",
      "clust tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step1 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step1 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step1 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 559 0s 8ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 8 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 559 0s 7ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 559 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 45ms\n",
      "Total time: 0h 0m 0s 61ms\n",
      "\n",
      "Size of the sequence database: 559\n",
      "Size of the alignment database: 559\n",
      "Number of clusters: 531\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 62ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step1 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step1 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step2 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step2 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step2 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3.25 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 531 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 531 type: Aminoacid\n",
      "Index table k-mer threshold: 134 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 531 0s 18ms     \n",
      "Index table: Masked residues: 2905\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 531 0s 17ms     \n",
      "Index statistics\n",
      "Entries:          179243\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002801\n",
      "Top 10 k-mers\n",
      "    KLEGKA\t3\n",
      "    DPQALA\t3\n",
      "    PSSNAE\t3\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    QSNLAK\t3\n",
      "    SGMEEK\t3\n",
      "    ANLEKL\t3\n",
      "    NLAALN\t3\n",
      "Time for index table init: 0h 0m 0s 412ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 134\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 531\n",
      "Target db start 1 to 531\n",
      "[=================================================================] 100.00% 531 0s 61ms     \n",
      "\n",
      "31.079975 k-mers per position\n",
      "422 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 10ms\n",
      "align tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step2 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step2 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step2 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 531 type: Aminoacid\n",
      "Target database size: 531 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 531 0s 13ms     \n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\n",
      "618 alignments calculated\n",
      "553 sequence pairs passed the thresholds (0.894822 of overall calculated)\n",
      "1.041431 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 17ms\n",
      "clust tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step2 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step2 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step2 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 531 0s 11ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 8 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 531 0s 12ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 531 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 83ms\n",
      "Total time: 0h 0m 0s 127ms\n",
      "\n",
      "Size of the sequence database: 531\n",
      "Size of the alignment database: 531\n",
      "Number of clusters: 518\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 128ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step2 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step2 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step3 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step3: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step3 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step3 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 4.375 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 518 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 518 type: Aminoacid\n",
      "Index table k-mer threshold: 124 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 518 0s 17ms     \n",
      "Index table: Masked residues: 2716\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 518 0s 18ms     \n",
      "Index statistics\n",
      "Entries:          184288\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002880\n",
      "Top 10 k-mers\n",
      "    AEGLAA\t3\n",
      "    KLEGKA\t3\n",
      "    ELLLLD\t3\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "Time for index table init: 0h 0m 0s 426ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 124\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 518\n",
      "Target db start 1 to 518\n",
      "[=================================================================] 100.00% 518 0s 158ms    \n",
      "\n",
      "95.874014 k-mers per position\n",
      "531 DB matches per sequence\n",
      "0 overflows\n",
      "1 sequences passed prefiltering per query sequence\n",
      "1 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step3: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 95ms\n",
      "align tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step3 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step3 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step3 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step3 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 518 type: Aminoacid\n",
      "Target database size: 518 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 518 0s 20ms     \n",
      "Time for merging to aln_step3: 0h 0m 0s 0ms\n",
      "812 alignments calculated\n",
      "544 sequence pairs passed the thresholds (0.669951 of overall calculated)\n",
      "1.050193 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 26ms\n",
      "clust tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step3 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step3 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step3 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 518 0s 8ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 8 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 518 0s 10ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 518 0s 2ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 55ms\n",
      "Total time: 0h 0m 0s 67ms\n",
      "\n",
      "Size of the sequence database: 518\n",
      "Size of the alignment database: 518\n",
      "Number of clusters: 501\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step3: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 68ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step3 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step3 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step4 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step4: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step4 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step4 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step4 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5.5 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 501 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 501 type: Aminoacid\n",
      "Index table k-mer threshold: 114 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 501 0s 22ms     \n",
      "Index table: Masked residues: 2688\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 501 0s 18ms     \n",
      "Index statistics\n",
      "Entries:          176936\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002765\n",
      "Top 10 k-mers\n",
      "    AEGLAA\t3\n",
      "    KLEGKA\t3\n",
      "    ELLLLD\t3\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "Time for index table init: 0h 0m 0s 538ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 114\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 501\n",
      "Target db start 1 to 501\n",
      "[=================================================================] 100.00% 501 0s 394ms    \n",
      "\n",
      "281.116904 k-mers per position\n",
      "740 DB matches per sequence\n",
      "0 overflows\n",
      "3 sequences passed prefiltering per query sequence\n",
      "3 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step4: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 738ms\n",
      "align tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step4 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step4 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step4 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step4 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 501 type: Aminoacid\n",
      "Target database size: 501 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 501 0s 53ms     \n",
      "Time for merging to aln_step4: 0h 0m 0s 0ms\n",
      "1470 alignments calculated\n",
      "543 sequence pairs passed the thresholds (0.369388 of overall calculated)\n",
      "1.083832 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 75ms\n",
      "clust tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step4 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step4 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step4 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 501 0s 12ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 14 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 501 0s 7ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 501 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 70ms\n",
      "Total time: 0h 0m 0s 105ms\n",
      "\n",
      "Size of the sequence database: 501\n",
      "Size of the alignment database: 501\n",
      "Number of clusters: 475\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step4: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 111ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step4 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step4 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step5 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step5: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step5 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step5 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step5 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 6.625 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 475 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 475 type: Aminoacid\n",
      "Index table k-mer threshold: 104 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 475 0s 22ms     \n",
      "Index table: Masked residues: 2541\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 475 0s 21ms     \n",
      "Index statistics\n",
      "Entries:          166240\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002597\n",
      "Top 10 k-mers\n",
      "    KLEGKA\t3\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    SSEQAA\t2\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "    GETEDA\t2\n",
      "Time for index table init: 0h 0m 0s 591ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 104\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 475\n",
      "Target db start 1 to 475\n",
      "[=================================================================] 100.00% 475 1s 152ms    \n",
      "\n",
      "784.655178 k-mers per position\n",
      "1288 DB matches per sequence\n",
      "0 overflows\n",
      "8 sequences passed prefiltering per query sequence\n",
      "8 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step5: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 2s 554ms\n",
      "align tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step5 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step5 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step5 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step5 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 475 type: Aminoacid\n",
      "Target database size: 475 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 475 0s 60ms     \n",
      "Time for merging to aln_step5: 0h 0m 0s 0ms\n",
      "3778 alignments calculated\n",
      "506 sequence pairs passed the thresholds (0.133933 of overall calculated)\n",
      "1.065263 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 74ms\n",
      "clust tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step5 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step5 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step5 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 475 0s 8ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 7 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 475 0s 7ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 475 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 49ms\n",
      "Total time: 0h 0m 0s 65ms\n",
      "\n",
      "Size of the sequence database: 475\n",
      "Size of the alignment database: 475\n",
      "Number of clusters: 456\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step5: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 74ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step5 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step5 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step6 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step6: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step6 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step6 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step6 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 7.75 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 456 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 456 type: Aminoacid\n",
      "Index table k-mer threshold: 94 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 456 0s 29ms     \n",
      "Index table: Masked residues: 2326\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 456 0s 21ms     \n",
      "Index statistics\n",
      "Entries:          159844\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002498\n",
      "Top 10 k-mers\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    SSEQAA\t2\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "    GETEDA\t2\n",
      "    NKESDA\t2\n",
      "Time for index table init: 0h 0m 0s 605ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 94\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 456\n",
      "Target db start 1 to 456\n",
      "[=================================================================] 100.00% 456 2s 431ms    \n",
      "\n",
      "2086.918513 k-mers per position\n",
      "2765 DB matches per sequence\n",
      "0 overflows\n",
      "16 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step6: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 3s 959ms\n",
      "align tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step6 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step6 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step6 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step6 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 456 type: Aminoacid\n",
      "Target database size: 456 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 456 0s 100ms    \n",
      "Time for merging to aln_step6: 0h 0m 0s 0ms\n",
      "6994 alignments calculated\n",
      "474 sequence pairs passed the thresholds (0.067772 of overall calculated)\n",
      "1.039474 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 106ms\n",
      "clust tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step6 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step6 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step6 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 456 0s 13ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 8 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 456 0s 8ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 456 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 70ms\n",
      "Total time: 0h 0m 0s 86ms\n",
      "\n",
      "Size of the sequence database: 456\n",
      "Size of the alignment database: 456\n",
      "Number of clusters: 443\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step6: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 88ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step6 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step6 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step7 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step7: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step7 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step7 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step7 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 8.875 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 443 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 443 type: Aminoacid\n",
      "Index table k-mer threshold: 84 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 443 0s 30ms     \n",
      "Index table: Masked residues: 2068\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 443 0s 13ms     \n",
      "Index statistics\n",
      "Entries:          153832\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002404\n",
      "Top 10 k-mers\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "    GETEDA\t2\n",
      "    NKESDA\t2\n",
      "    LTLQEA\t2\n",
      "Time for index table init: 0h 0m 0s 571ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 84\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 443\n",
      "Target db start 1 to 443\n",
      "[=================================================================] 100.00% 443 5s 880ms    \n",
      "\n",
      "5298.930428 k-mers per position\n",
      "6408 DB matches per sequence\n",
      "0 overflows\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step7: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 7s 437ms\n",
      "align tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step7 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step7 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step7 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step7 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 443 type: Aminoacid\n",
      "Target database size: 443 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 443 0s 61ms     \n",
      "Time for merging to aln_step7: 0h 0m 0s 0ms\n",
      "7677 alignments calculated\n",
      "448 sequence pairs passed the thresholds (0.058356 of overall calculated)\n",
      "1.011287 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 65ms\n",
      "clust tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step7 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step7 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step7 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 443 0s 11ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 443 0s 11ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 443 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 56ms\n",
      "Total time: 0h 0m 0s 63ms\n",
      "\n",
      "Size of the sequence database: 443\n",
      "Size of the alignment database: 443\n",
      "Number of clusters: 439\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step7: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 64ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step7 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step7 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step8 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step8: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step8 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step8 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step8 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 10 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 439 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 439 type: Aminoacid\n",
      "Index table k-mer threshold: 74 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 439 0s 14ms     \n",
      "Index table: Masked residues: 2068\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 439 0s 18ms     \n",
      "Index statistics\n",
      "Entries:          151529\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.002368\n",
      "Top 10 k-mers\n",
      "    VGVPDE\t3\n",
      "    LLLERG\t3\n",
      "    RKFDTG\t3\n",
      "    SGMEEK\t3\n",
      "    GAGLLS\t3\n",
      "    IQLRAA\t2\n",
      "    AELDDA\t2\n",
      "    GETEDA\t2\n",
      "    NKESDA\t2\n",
      "    LTLQEA\t2\n",
      "Time for index table init: 0h 0m 0s 508ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 74\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 439\n",
      "Target db start 1 to 439\n",
      "[=================================================================] 100.00% 439 13s 230ms    \n",
      "\n",
      "12928.794688 k-mers per position\n",
      "15365 DB matches per sequence\n",
      "0 overflows\n",
      "19 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step8: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 14s 656ms\n",
      "align tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step8 tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step8 tmp/6478405549615959369/clu_tmp/2715299047738836632/pref_step8 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step8 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 439 type: Aminoacid\n",
      "Target database size: 439 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 439 0s 57ms     \n",
      "Time for merging to aln_step8: 0h 0m 0s 0ms\n",
      "7782 alignments calculated\n",
      "442 sequence pairs passed the thresholds (0.056798 of overall calculated)\n",
      "1.006834 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 64ms\n",
      "clust tmp/6478405549615959369/clu_tmp/2715299047738836632/input_step8 tmp/6478405549615959369/clu_tmp/2715299047738836632/aln_step8 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step8 --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 439 0s 10ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 439 0s 5ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 439 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 45ms\n",
      "Total time: 0h 0m 0s 56ms\n",
      "\n",
      "Size of the sequence database: 439\n",
      "Size of the alignment database: 439\n",
      "Number of clusters: 436\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step8: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 58ms\n",
      "mergeclusters tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/clu tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_redundancy tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step0 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step1 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step2 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step3 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step4 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step5 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step6 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step7 tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_step8 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 622 0s 9ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 559 0s 29ms     \n",
      "Clustering step 3\n",
      "[=================================================================] 100.00% 531 0s 41ms     \n",
      "Clustering step 4\n",
      "[=================================================================] 100.00% 518 0s 60ms     \n",
      "Clustering step 5\n",
      "[=================================================================] 100.00% 501 0s 74ms     \n",
      "Clustering step 6\n",
      "[=================================================================] 100.00% 475 0s 90ms     \n",
      "Clustering step 7\n",
      "[=================================================================] 100.00% 456 0s 99ms     \n",
      "Clustering step 8\n",
      "[=================================================================] 100.00% 443 0s 109ms    \n",
      "Clustering step 9\n",
      "[=================================================================] 100.00% 439 0s 119ms    \n",
      "Clustering step 10\n",
      "[=================================================================] 100.00% 436 0s 131ms    \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 1.91K 0s 143ms    \n",
      "Time for merging to clu: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 153ms\n",
      "align tmp/6478405549615959369/input tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/clu tmp/6478405549615959369/clu_tmp/2715299047738836632/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1908 type: Aminoacid\n",
      "Target database size: 1908 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 436 0s 129ms    \n",
      "Time for merging to aln: 0h 0m 0s 0ms\n",
      "1908 alignments calculated\n",
      "1751 sequence pairs passed the thresholds (0.917715 of overall calculated)\n",
      "4.016055 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 143ms\n",
      "subtractdbs tmp/6478405549615959369/clu_tmp/2715299047738836632/clu tmp/6478405549615959369/clu_tmp/2715299047738836632/aln tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_not_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "subtractdbs tmp/6478405549615959369/clu_tmp/2715299047738836632/clu tmp/6478405549615959369/clu_tmp/2715299047738836632/aln tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_not_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Remove tmp/6478405549615959369/clu_tmp/2715299047738836632/aln ids from tmp/6478405549615959369/clu_tmp/2715299047738836632/clu\n",
      "[=================================================================] 100.00% 436 0s 9ms      \n",
      "Time for merging to clu_not_accepted: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 17ms\n",
      "swapdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_not_accepted tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_not_accepted_swap --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 436 0s 3ms      \n",
      "Computing offsets.\n",
      "[=================================================================] 100.00% 436 0s 7ms      \n",
      "\n",
      "Reading results.\n",
      "[=================================================================] 100.00% 436 0s 4ms      \n",
      "\n",
      "Output database: tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_not_accepted_swap\n",
      "[=================================================================] 100.00% 1.91K 0s 5ms      \n",
      "\n",
      "Time for merging to clu_not_accepted_swap: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 36ms\n",
      "subtractdbs tmp/6478405549615959369/clu_tmp/2715299047738836632/clu tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_not_accepted tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "subtractdbs tmp/6478405549615959369/clu_tmp/2715299047738836632/clu tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_not_accepted tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_accepted --e-profile 100000000 -e 100000000 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Remove tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_not_accepted ids from tmp/6478405549615959369/clu_tmp/2715299047738836632/clu\n",
      "[=================================================================] 100.00% 436 0s 5ms      \n",
      "Time for merging to clu_accepted: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 8ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_not_accepted_swap tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned -v 3 \n",
      "\n",
      "Time for merging to seq_wrong_assigned: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createsubdb tmp/6478405549615959369/clu_tmp/2715299047738836632/clu tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_seeds -v 3 \n",
      "\n",
      "Time for merging to seq_seeds: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_seeds.merged tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned_pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 10 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.2 --cov-mode 5 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 157 type: Aminoacid\n",
      "Estimated memory consumption: 979M\n",
      "Target database size: 593 type: Aminoacid\n",
      "Index table k-mer threshold: 74 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 593 0s 22ms     \n",
      "Index table: Masked residues: 2961\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 593 0s 18ms     \n",
      "Index statistics\n",
      "Entries:          223706\n",
      "DB size:          489 MB\n",
      "Avg k-mer size:   0.003495\n",
      "Top 10 k-mers\n",
      "    TALKKA\t33\n",
      "    VRVRND\t33\n",
      "    SLAHVI\t33\n",
      "    KTELAL\t33\n",
      "    ENDALR\t33\n",
      "    YDNKDS\t33\n",
      "    YVVKSS\t33\n",
      "    PISSEA\t32\n",
      "    IFKDFA\t32\n",
      "    KIAHFA\t32\n",
      "Time for index table init: 0h 0m 0s 507ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 74\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 157\n",
      "Target db start 1 to 593\n",
      "[=================================================================] 100.00% 157 5s 167ms    \n",
      "\n",
      "12322.130015 k-mers per position\n",
      "34190 DB matches per sequence\n",
      "0 overflows\n",
      "20 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to seq_wrong_assigned_pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 6s 561ms\n",
      "swapdb tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned_pref tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned_pref_swaped --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 157 0s 3ms      \n",
      "Computing offsets.\n",
      "[=================================================================] 100.00% 157 0s 2ms      \n",
      "\n",
      "Reading results.\n",
      "[=================================================================] 100.00% 157 0s 3ms      \n",
      "\n",
      "Output database: tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned_pref_swaped\n",
      "[=================================================================] 100.00% 1.91K 0s 5ms      \n",
      "\n",
      "Time for merging to seq_wrong_assigned_pref_swaped: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 41ms\n",
      "align tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_seeds.merged tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned_pref_swaped tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned_pref_swaped_aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.1 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.2 --cov-mode 5 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 593 type: Aminoacid\n",
      "Target database size: 157 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 436 0s 425ms    \n",
      "Time for merging to seq_wrong_assigned_pref_swaped_aln: 0h 0m 0s 0ms\n",
      "3053 alignments calculated\n",
      "1926 sequence pairs passed the thresholds (0.630855 of overall calculated)\n",
      "4.417431 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 470ms\n",
      "filterdb tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned_pref_swaped_aln tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned_pref_swaped_aln_ocol --trim-to-one-column --threads 8 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using regular expression\n",
      "[=================================================================] 100.00% 436 0s 6ms      \n",
      "Time for merging to seq_wrong_assigned_pref_swaped_aln_ocol: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 18ms\n",
      "mergedbs tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_seeds.merged tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_accepted_plus_wrong tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_accepted tmp/6478405549615959369/clu_tmp/2715299047738836632/seq_wrong_assigned_pref_swaped_aln_ocol --merge-stop-empty 0 --compressed 0 -v 3 \n",
      "\n",
      "Merging the results to tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_accepted_plus_wrong\n",
      "[=================================================================] 100.00% 593 0s 0ms      \n",
      "Time for merging to clu_accepted_plus_wrong: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "tsv2db tmp/6478405549615959369/clu_tmp/2715299047738836632/missing.single.seqs tmp/6478405549615959369/clu_tmp/2715299047738836632/missing.single.seqs.db --output-dbtype 6 --compressed 0 -v 3 \n",
      "\n",
      "Output database type: Clustering\n",
      "Time for merging to missing.single.seqs.db: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "mergedbs tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_accepted_plus_wrong_plus_single tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_accepted_plus_wrong tmp/6478405549615959369/clu_tmp/2715299047738836632/missing.single.seqs.db --merge-stop-empty 0 --compressed 0 -v 3 \n",
      "\n",
      "Merging the results to tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_accepted_plus_wrong_plus_single\n",
      "[=================================================================] 100.00% 1.91K 0s 0ms      \n",
      "Time for merging to clu_accepted_plus_wrong_plus_single: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "clust tmp/6478405549615959369/input tmp/6478405549615959369/clu_tmp/2715299047738836632/clu_accepted_plus_wrong_plus_single tmp/6478405549615959369/clu --cluster-mode 0 --max-iterations 50000 --similarity-type 2 --threads 8 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 1.91K 0s 7ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 1630 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 1.91K 0s 7ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 1.91K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 36ms\n",
      "Total time: 0h 0m 0s 45ms\n",
      "\n",
      "Size of the sequence database: 1908\n",
      "Size of the alignment database: 1908\n",
      "Number of clusters: 451\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 47ms\n",
      "createtsv tmp/6478405549615959369/input tmp/6478405549615959369/input tmp/6478405549615959369/clu tmp/6478405549615959369/cluster.tsv --threads 8 -v 3 \n",
      "\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 11ms\n",
      "result2repseq tmp/6478405549615959369/input tmp/6478405549615959369/clu tmp/6478405549615959369/clu_rep --db-load-mode 0 --compressed 0 --threads 8 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 451 0s 4ms      \n",
      "Time for merging to clu_rep: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 10ms\n",
      "result2flat tmp/6478405549615959369/input tmp/6478405549615959369/input tmp/6478405549615959369/clu_rep tmp/6478405549615959369/rep_seq.fasta --use-fasta-header -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "createseqfiledb tmp/6478405549615959369/input tmp/6478405549615959369/clu tmp/6478405549615959369/clu_seqs --threads 8 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 451 0s 8ms      \n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 18ms\n",
      "result2flat tmp/6478405549615959369/input tmp/6478405549615959369/input tmp/6478405549615959369/clu_seqs tmp/6478405549615959369/all_seqs.fasta -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 3ms\n"
     ]
    }
   ],
   "source": [
    "!mmseqs easy-cluster data_files/all_test_seqs.fasta data_files/cluster tmp --min-seq-id 0.1 --cov-mode 5 -c 0.2 -s 10.0 --remove-tmp-files --max-iterations 50000 --cluster-reassign --cluster-steps 9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_rep_id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2C5D_1015</th>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4RS1_1910</th>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4BFI_6035</th>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5M2O_5671</th>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1IAR_3435</th>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1ZHI_3416</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4IU3_5847</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1Z92_7910</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4JO6_3119</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1AVA_62</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>451 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                cluster_size\n",
       "cluster_rep_id              \n",
       "2C5D_1015                 50\n",
       "4RS1_1910                 48\n",
       "4BFI_6035                 41\n",
       "5M2O_5671                 40\n",
       "1IAR_3435                 37\n",
       "...                      ...\n",
       "1ZHI_3416                  1\n",
       "4IU3_5847                  1\n",
       "1Z92_7910                  1\n",
       "4JO6_3119                  1\n",
       "1AVA_62                    1\n",
       "\n",
       "[451 rows x 1 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_df = pd.read_csv(\n",
    "    \"data_files/cluster_cluster.tsv\", sep=\"\\t\", header=None\n",
    ")\n",
    "cluster_df = cluster_df.rename(columns={0: \"cluster_rep_id\", 1: \"fasta_id\"})\n",
    "cluster_df_count = (\n",
    "    cluster_df.groupby([\"cluster_rep_id\"])\n",
    "    .count()\n",
    "    .sort_values(\"fasta_id\", ascending=False)\n",
    "    .rename(columns={\"fasta_id\": \"cluster_size\"})\n",
    ")\n",
    "cluster_df_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "split\n",
       "train    0.78\n",
       "val      0.11\n",
       "test     0.11\n",
       "Name: count, dtype: float64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# evenly distribute the clusters between val and test\n",
    "val_fasta_ids = []\n",
    "add_to_val = True\n",
    "for cluster_rep_id, cluster_size in cluster_df_count.iterrows():\n",
    "    if add_to_val:\n",
    "        val_fasta_ids.extend(\n",
    "            cluster_df[cluster_df[\"cluster_rep_id\"] == cluster_rep_id][\n",
    "                \"fasta_id\"\n",
    "            ].tolist()\n",
    "        )\n",
    "    add_to_val = not add_to_val\n",
    "\n",
    "df[\"split\"] = df.apply(\n",
    "    lambda row: \"val\" if row[\"fasta_id\"] in val_fasta_ids else row[\"split\"],\n",
    "    axis=1,\n",
    ")\n",
    "(df[\"split\"].value_counts() / len(df)).round(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Changing 51 sequences from test to val\n",
      "Current split ratios: split\n",
      "train    0.78\n",
      "val      0.12\n",
      "test     0.10\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 36 sequences from test to val\n",
      "Current split ratios: split\n",
      "train    0.78\n",
      "val      0.12\n",
      "test     0.10\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 8 sequences from test to val\n",
      "Current split ratios: split\n",
      "train    0.78\n",
      "val      0.12\n",
      "test     0.10\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "No matches found\n",
      "Current split ratios: split\n",
      "train    0.78\n",
      "val      0.12\n",
      "test     0.10\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "split\n",
       "train    6906\n",
       "val      1059\n",
       "test      849\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# perform final split iterations to ensure that the val set is not too similar\n",
    "# to the test set\n",
    "while True:\n",
    "    df, total_changed = do_split_iteration(df, (\"val\", \"test\"))\n",
    "    print(\n",
    "        f\"Current split ratios: {(df['split'].value_counts() / len(df)).round(2)}\"\n",
    "    )\n",
    "    print(\"*\" * 80)\n",
    "    if (df[\"split\"] == \"test\").sum() == 0 or total_changed == 0:\n",
    "        break\n",
    "\n",
    "df[\"split\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.drop(columns=[\"ligand\", \"receptor\", \"combined\", \"fasta_id\"])\n",
    "df.to_csv(\"PPB_Affinity_processed_filtered.csv\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
