{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "from datasets import load_dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = load_dataset(\n",
    "    \"proteinea/ppb_affinity\",\n",
    "    \"raw\",\n",
    "    trust_remote_code=True,\n",
    "    split=\"train\",\n",
    ").to_pandas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# these columns are created for processing purposes, they will be removed later\n",
    "df[\"ligand\"] = df[\"Ligand Sequences\"].str.replace(\",\", \"\")\n",
    "df[\"receptor\"] = df[\"Receptor Sequences\"].str.replace(\",\", \"\")\n",
    "df[\"combined\"] = df[\"ligand\"] + df[\"receptor\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Source Data Set</th>\n",
       "      <th>Complex ID</th>\n",
       "      <th>PDB</th>\n",
       "      <th>Mutations</th>\n",
       "      <th>Ligand Chains</th>\n",
       "      <th>Receptor Chains</th>\n",
       "      <th>Ligand Name</th>\n",
       "      <th>Receptor Name</th>\n",
       "      <th>KD(M)</th>\n",
       "      <th>Affinity Method</th>\n",
       "      <th>...</th>\n",
       "      <th>PDB PubMed ID</th>\n",
       "      <th>PDB Release Date</th>\n",
       "      <th>Affinity PubMed ID</th>\n",
       "      <th>Affinity Release Date</th>\n",
       "      <th>Subgroup</th>\n",
       "      <th>Ligand Sequences</th>\n",
       "      <th>Receptor Sequences</th>\n",
       "      <th>ligand</th>\n",
       "      <th>receptor</th>\n",
       "      <th>combined</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1A22:A, B::PMID=7504735</td>\n",
       "      <td>1A22</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>Human growth hormone</td>\n",
       "      <td>hGH binding protein</td>\n",
       "      <td>9e-10</td>\n",
       "      <td>SPR</td>\n",
       "      <td>...</td>\n",
       "      <td>9571026.0</td>\n",
       "      <td>1998-04-29</td>\n",
       "      <td>7504735</td>\n",
       "      <td>1993 Dec 5</td>\n",
       "      <td></td>\n",
       "      <td>FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...</td>\n",
       "      <td>PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...</td>\n",
       "      <td>FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...</td>\n",
       "      <td>PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...</td>\n",
       "      <td>FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1A4Y:A, B::PMID=9050852</td>\n",
       "      <td>1A4Y</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>Ribonuclease inhibitor</td>\n",
       "      <td>Angiogenin</td>\n",
       "      <td>5e-16</td>\n",
       "      <td>Other</td>\n",
       "      <td>...</td>\n",
       "      <td>9311977.0</td>\n",
       "      <td>1998-10-14</td>\n",
       "      <td>9050852</td>\n",
       "      <td>1997 Mar 4</td>\n",
       "      <td></td>\n",
       "      <td>SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...</td>\n",
       "      <td>QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...</td>\n",
       "      <td>SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...</td>\n",
       "      <td>QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...</td>\n",
       "      <td>SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1ACB:E, I::PMID=9048543</td>\n",
       "      <td>1ACB</td>\n",
       "      <td></td>\n",
       "      <td>E</td>\n",
       "      <td>I</td>\n",
       "      <td>Bovine alpha-chymotrypsin</td>\n",
       "      <td>Eglin c</td>\n",
       "      <td>1.49e-12</td>\n",
       "      <td>IASP</td>\n",
       "      <td>...</td>\n",
       "      <td>1583684.0</td>\n",
       "      <td>1993-10-31</td>\n",
       "      <td>9048543</td>\n",
       "      <td>1997 Feb 18</td>\n",
       "      <td></td>\n",
       "      <td>CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...</td>\n",
       "      <td>KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...</td>\n",
       "      <td>CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...</td>\n",
       "      <td>KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...</td>\n",
       "      <td>CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1AHW:A, B, C::PMID=9480775</td>\n",
       "      <td>1AHW</td>\n",
       "      <td></td>\n",
       "      <td>A, B</td>\n",
       "      <td>C</td>\n",
       "      <td>Immunoglobulin fab 5G9</td>\n",
       "      <td>Tissue factor</td>\n",
       "      <td>3.4e-09</td>\n",
       "      <td>IASP</td>\n",
       "      <td>...</td>\n",
       "      <td>9480775.0</td>\n",
       "      <td>1998-02-25</td>\n",
       "      <td>9480775</td>\n",
       "      <td>1998 Feb 6</td>\n",
       "      <td></td>\n",
       "      <td>DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...</td>\n",
       "      <td>TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...</td>\n",
       "      <td>DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...</td>\n",
       "      <td>TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...</td>\n",
       "      <td>DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SKEMPI v2.0</td>\n",
       "      <td>1AK4:A, D::PMID=9223641</td>\n",
       "      <td>1AK4</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>D</td>\n",
       "      <td>Cyclophilin A</td>\n",
       "      <td>HIV-1 capsid protein</td>\n",
       "      <td>1.2e-05</td>\n",
       "      <td>SPR</td>\n",
       "      <td>...</td>\n",
       "      <td>8980234.0</td>\n",
       "      <td>1997-10-15</td>\n",
       "      <td>9223641</td>\n",
       "      <td>1997 Jun 27</td>\n",
       "      <td></td>\n",
       "      <td>MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...</td>\n",
       "      <td>PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...</td>\n",
       "      <td>MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...</td>\n",
       "      <td>PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...</td>\n",
       "      <td>MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Source Data Set                  Complex ID   PDB Mutations Ligand Chains  \\\n",
       "0     SKEMPI v2.0     1A22:A, B::PMID=7504735  1A22                       A   \n",
       "1     SKEMPI v2.0     1A4Y:A, B::PMID=9050852  1A4Y                       A   \n",
       "2     SKEMPI v2.0     1ACB:E, I::PMID=9048543  1ACB                       E   \n",
       "3     SKEMPI v2.0  1AHW:A, B, C::PMID=9480775  1AHW                    A, B   \n",
       "4     SKEMPI v2.0     1AK4:A, D::PMID=9223641  1AK4                       A   \n",
       "\n",
       "  Receptor Chains                Ligand Name         Receptor Name     KD(M)  \\\n",
       "0               B       Human growth hormone   hGH binding protein     9e-10   \n",
       "1               B     Ribonuclease inhibitor            Angiogenin     5e-16   \n",
       "2               I  Bovine alpha-chymotrypsin               Eglin c  1.49e-12   \n",
       "3               C     Immunoglobulin fab 5G9         Tissue factor   3.4e-09   \n",
       "4               D              Cyclophilin A  HIV-1 capsid protein   1.2e-05   \n",
       "\n",
       "  Affinity Method  ... PDB PubMed ID PDB Release Date Affinity PubMed ID  \\\n",
       "0             SPR  ...     9571026.0       1998-04-29            7504735   \n",
       "1           Other  ...     9311977.0       1998-10-14            9050852   \n",
       "2            IASP  ...     1583684.0       1993-10-31            9048543   \n",
       "3            IASP  ...     9480775.0       1998-02-25            9480775   \n",
       "4             SPR  ...     8980234.0       1997-10-15            9223641   \n",
       "\n",
       "  Affinity Release Date Subgroup  \\\n",
       "0            1993 Dec 5            \n",
       "1            1997 Mar 4            \n",
       "2           1997 Feb 18            \n",
       "3            1998 Feb 6            \n",
       "4           1997 Jun 27            \n",
       "\n",
       "                                    Ligand Sequences  \\\n",
       "0  FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...   \n",
       "1  SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...   \n",
       "2  CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...   \n",
       "3  DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...   \n",
       "4  MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...   \n",
       "\n",
       "                                  Receptor Sequences  \\\n",
       "0  PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...   \n",
       "1  QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...   \n",
       "2  KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...   \n",
       "3  TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...   \n",
       "4  PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...   \n",
       "\n",
       "                                              ligand  \\\n",
       "0  FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...   \n",
       "1  SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...   \n",
       "2  CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...   \n",
       "3  DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...   \n",
       "4  MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...   \n",
       "\n",
       "                                            receptor  \\\n",
       "0  PKFTKCRSPERETFSCHWTDEVHHGTKNLGPIQLFYTRRNTQEWTQ...   \n",
       "1  QDNSRYTHFLTQHYDAKPQGRDDRYCESIMRRRGLTSPCKDINTFI...   \n",
       "2  KSFPEVVGKTVDQAREYFTLHYPQYDVYFLPEGSPVTLDLRYNRVR...   \n",
       "3  TNTVAAYNLTWKSTNFKTILEWEPKPVNQVYTVQISTKSGDWKSKC...   \n",
       "4  PIVQNLQGQMVHQAISPRTLNAWVKVVEEKAFSPEVIPMFSALSEG...   \n",
       "\n",
       "                                            combined  \n",
       "0  FPTIPLSRLFDNAMLRAHRLHQLAFDTYQEFEEAYIPKEQKYSFLQ...  \n",
       "1  SLDIQSLDIQCEELSDARWAELLPLLQQCQVVRLDDCGLTEARCKD...  \n",
       "2  CGVPAIQPVLSGLSRIVNGEEAVPGSWPWQVSLQDKTGFHFCGGSL...  \n",
       "3  DIKMTQSPSSMYASLGERVTITCKASQDIRKYLNWYQQKPWKSPKT...  \n",
       "4  MVNPTVFFDIAVDGEPLGRVSFELFADKVPKTAENFRALSTGEKGF...  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"KD(M)\"] = pd.to_numeric(df[\"KD(M)\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dropped 2753 rows with ligand or receptor chain(s) shorter than 40\n",
      "Remaining rows: 9295\n"
     ]
    }
   ],
   "source": [
    "# filter out rows where any ligand or receptor chain is shorter than 40\n",
    "# amino acids\n",
    "minlen = 40\n",
    "f = lambda seqs: any([len(seq) < minlen for seq in seqs])\n",
    "min_len_mask = df[\"Ligand Sequences\"].str.split(\",\").apply(f) | df[\n",
    "    \"Receptor Sequences\"\n",
    "].str.split(\",\").apply(f)\n",
    "df = df[~min_len_mask].reset_index(drop=True)\n",
    "print(\n",
    "    f\"Dropped {min_len_mask.sum()} rows with ligand or receptor chain(s) \"\n",
    "    f\"shorter than {minlen}\"\n",
    ")\n",
    "print(f\"Remaining rows: {len(df)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# replace empty strings in \"Affinity Method\" with \"Unknown\"\n",
    "df[\"Affinity Method\"] = df[\"Affinity Method\"].apply(\n",
    "    lambda x: x if x != \"\" else \"Unknown\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Affinity Method\n",
       "Unknown    2931\n",
       "SPR        2418\n",
       "FL         1090\n",
       "Other       786\n",
       "ITC         543\n",
       "IASP        461\n",
       "SP          336\n",
       "RA          320\n",
       "ELISA       172\n",
       "IARA        125\n",
       "BLI         104\n",
       "IAGE          9\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check affinity method distribution in the dataset, we will use the Affinity\n",
    "# Method for handling duplicated sequences\n",
    "df[\"Affinity Method\"].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_duplicates_by_method(\n",
    "    df,\n",
    "    seq_col=\"combined\",\n",
    "    kd_col=\"KD(M)\",\n",
    "    method_col=\"Affinity Method\",\n",
    "):\n",
    "    \"\"\"\n",
    "    Group by 'seq_col' (the combined sequence) and resolve duplicates by method priority:\n",
    "      1. Within each group (all rows sharing the same 'combined' sequence), find the highest-priority method present.\n",
    "      2. Average the KD(M) values for all rows of that method.\n",
    "      3. Keep the *first row* from that subset for all other columns.\n",
    "    \"\"\"\n",
    "\n",
    "    # Priority list of methods to choose from\n",
    "    method_priority = [\n",
    "        \"SPR\",\n",
    "        \"ITC\",\n",
    "        \"BLI\",\n",
    "        \"RA\",\n",
    "        \"FL\",\n",
    "        \"SP\",\n",
    "        \"ELISA\",\n",
    "        \"IASP\",\n",
    "        \"IARA\",\n",
    "        \"IAGE\",\n",
    "        \"Other\",\n",
    "        \"Unknown\",\n",
    "    ]\n",
    "\n",
    "    def choose_best_method(group):\n",
    "        \"\"\"\n",
    "        For a subset of df with the same 'combined' sequence:\n",
    "          - Iterate over the 'method_priority' list in order.\n",
    "          - If the group contains that method, average KD(M) across those rows.\n",
    "          - Return the *first row* of that subset for all other columns.\n",
    "        \"\"\"\n",
    "        methods_in_group = group[method_col].unique()\n",
    "\n",
    "        for meth in method_priority:\n",
    "            if meth in methods_in_group:\n",
    "                # Get only the rows of the current top-priority method\n",
    "                subset = group[group[method_col] == meth]\n",
    "\n",
    "                # Average KD(M) and pKD for this subset\n",
    "                avg_kd = subset[kd_col].mean()\n",
    "\n",
    "                # Take the *first row* from THIS subset\n",
    "                # (the subset that contributes to the average)\n",
    "                first_row = subset.iloc[0].copy()\n",
    "\n",
    "                # Update the KD(M) and method to reflect the average and the chosen method\n",
    "                first_row[kd_col] = avg_kd\n",
    "                first_row[method_col] = meth\n",
    "\n",
    "                return first_row\n",
    "        raise ValueError(\n",
    "            \"None of the methods in the priority list were found in the group!\"\n",
    "        )\n",
    "\n",
    "    # Apply the 'choose_best_method' to each group of duplicated sequences\n",
    "    result_df = df.groupby(seq_col, as_index=False).apply(\n",
    "        choose_best_method, include_groups=False\n",
    "    )\n",
    "    return result_df.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Remaining rows after filtering duplicates: 8207\n"
     ]
    }
   ],
   "source": [
    "df = filter_duplicates_by_method(df)\n",
    "print(f\"Remaining rows after filtering duplicates: {len(df)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>combined</th>\n",
       "      <th>Source Data Set</th>\n",
       "      <th>Complex ID</th>\n",
       "      <th>PDB</th>\n",
       "      <th>Mutations</th>\n",
       "      <th>Ligand Chains</th>\n",
       "      <th>Receptor Chains</th>\n",
       "      <th>Ligand Name</th>\n",
       "      <th>Receptor Name</th>\n",
       "      <th>KD(M)</th>\n",
       "      <th>...</th>\n",
       "      <th>PDB PubMed ID</th>\n",
       "      <th>PDB Release Date</th>\n",
       "      <th>Affinity PubMed ID</th>\n",
       "      <th>Affinity Release Date</th>\n",
       "      <th>Subgroup</th>\n",
       "      <th>Ligand Sequences</th>\n",
       "      <th>Receptor Sequences</th>\n",
       "      <th>ligand</th>\n",
       "      <th>receptor</th>\n",
       "      <th>fasta_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>2WQZ:B, C::PMID=18093521</td>\n",
       "      <td>2WQZ</td>\n",
       "      <td></td>\n",
       "      <td>B</td>\n",
       "      <td>C</td>\n",
       "      <td>Neuroligin4, X-Linked</td>\n",
       "      <td>Neurexin-1-beta</td>\n",
       "      <td>1.320000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>18093521.0</td>\n",
       "      <td>2009-09-08</td>\n",
       "      <td>18093521</td>\n",
       "      <td>2007 Dec 20</td>\n",
       "      <td></td>\n",
       "      <td>AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...</td>\n",
       "      <td>HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...</td>\n",
       "      <td>AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...</td>\n",
       "      <td>HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...</td>\n",
       "      <td>2WQZ_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>5IIA:A, B, C, D::PMID=28622512</td>\n",
       "      <td>5IIA</td>\n",
       "      <td></td>\n",
       "      <td>B, D</td>\n",
       "      <td>A, C</td>\n",
       "      <td>sperm lysinR</td>\n",
       "      <td>red abalone egg VERL repeat 3 (VR3)</td>\n",
       "      <td>1.810000e-09</td>\n",
       "      <td>...</td>\n",
       "      <td>28622512.0</td>\n",
       "      <td>2017-06-14</td>\n",
       "      <td>28622512</td>\n",
       "      <td>2017 Jun 15</td>\n",
       "      <td></td>\n",
       "      <td>AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...</td>\n",
       "      <td>FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...</td>\n",
       "      <td>AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...</td>\n",
       "      <td>FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...</td>\n",
       "      <td>5IIA_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>4ZI2:A, C::PMID=26455799</td>\n",
       "      <td>4ZI2</td>\n",
       "      <td></td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>BART-like domain of BARTL1/CCDC104 1-113</td>\n",
       "      <td>Arl3FL-GppNHp</td>\n",
       "      <td>4.200000e-07</td>\n",
       "      <td>...</td>\n",
       "      <td>26455799.0</td>\n",
       "      <td>2015-11-11</td>\n",
       "      <td>26455799</td>\n",
       "      <td>2015 Nov 3</td>\n",
       "      <td></td>\n",
       "      <td>AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...</td>\n",
       "      <td>LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...</td>\n",
       "      <td>AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...</td>\n",
       "      <td>LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...</td>\n",
       "      <td>4ZI2_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>6SAZ:A, B::PMID=31604990</td>\n",
       "      <td>6SAZ</td>\n",
       "      <td></td>\n",
       "      <td>A</td>\n",
       "      <td>B</td>\n",
       "      <td>Crayfish Astacin</td>\n",
       "      <td>Cleaved human fetuin-b</td>\n",
       "      <td>1.400000e-10</td>\n",
       "      <td>...</td>\n",
       "      <td>31604990.0</td>\n",
       "      <td>2019-10-23</td>\n",
       "      <td>31604990</td>\n",
       "      <td>2019 Oct 11</td>\n",
       "      <td></td>\n",
       "      <td>AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...</td>\n",
       "      <td>ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...</td>\n",
       "      <td>AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...</td>\n",
       "      <td>ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...</td>\n",
       "      <td>6SAZ_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAKEGWLHFRPLVTDKGKRVGGSIRPWKQMYVVLRGTTPSEEEHSL...</td>\n",
       "      <td>PDBbind v2020</td>\n",
       "      <td>2J59:D, P::PMID=17347647</td>\n",
       "      <td>2J59</td>\n",
       "      <td></td>\n",
       "      <td>P</td>\n",
       "      <td>D</td>\n",
       "      <td>RHO-GTPASE ACTIVATING PROTEIN 10</td>\n",
       "      <td>ADP-RIBOSYLATION FACTOR 1</td>\n",
       "      <td>5.500000e-08</td>\n",
       "      <td>...</td>\n",
       "      <td>17347647.0</td>\n",
       "      <td>2007-02-20</td>\n",
       "      <td>17347647</td>\n",
       "      <td>2007 Apr 4</td>\n",
       "      <td></td>\n",
       "      <td>AAKEGWLHFRPLVTDKGKRVGGSIRPWKQMYVVLRGTTPSEEEHSL...</td>\n",
       "      <td>GSMRILMVGLDAAGKTTILYKLKLGEIVTTIPTIGFNVETVEYKNI...</td>\n",
       "      <td>AAKEGWLHFRPLVTDKGKRVGGSIRPWKQMYVVLRGTTPSEEEHSL...</td>\n",
       "      <td>GSMRILMVGLDAAGKTTILYKLKLGEIVTTIPTIGFNVETVEYKNI...</td>\n",
       "      <td>2J59_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            combined Source Data Set  \\\n",
       "0  AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...   PDBbind v2020   \n",
       "1  AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...   PDBbind v2020   \n",
       "2  AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...   PDBbind v2020   \n",
       "3  AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...   PDBbind v2020   \n",
       "4  AAKEGWLHFRPLVTDKGKRVGGSIRPWKQMYVVLRGTTPSEEEHSL...   PDBbind v2020   \n",
       "\n",
       "                       Complex ID   PDB Mutations Ligand Chains  \\\n",
       "0        2WQZ:B, C::PMID=18093521  2WQZ                       B   \n",
       "1  5IIA:A, B, C, D::PMID=28622512  5IIA                    B, D   \n",
       "2        4ZI2:A, C::PMID=26455799  4ZI2                       C   \n",
       "3        6SAZ:A, B::PMID=31604990  6SAZ                       A   \n",
       "4        2J59:D, P::PMID=17347647  2J59                       P   \n",
       "\n",
       "  Receptor Chains                               Ligand Name  \\\n",
       "0               C                     Neuroligin4, X-Linked   \n",
       "1            A, C                              sperm lysinR   \n",
       "2               A  BART-like domain of BARTL1/CCDC104 1-113   \n",
       "3               B                          Crayfish Astacin   \n",
       "4               D          RHO-GTPASE ACTIVATING PROTEIN 10   \n",
       "\n",
       "                         Receptor Name         KD(M)  ... PDB PubMed ID  \\\n",
       "0                      Neurexin-1-beta  1.320000e-07  ...    18093521.0   \n",
       "1  red abalone egg VERL repeat 3 (VR3)  1.810000e-09  ...    28622512.0   \n",
       "2                        Arl3FL-GppNHp  4.200000e-07  ...    26455799.0   \n",
       "3               Cleaved human fetuin-b  1.400000e-10  ...    31604990.0   \n",
       "4           ADP-RIBOSYLATION FACTOR 1   5.500000e-08  ...    17347647.0   \n",
       "\n",
       "  PDB Release Date Affinity PubMed ID Affinity Release Date Subgroup  \\\n",
       "0       2009-09-08           18093521           2007 Dec 20            \n",
       "1       2017-06-14           28622512           2017 Jun 15            \n",
       "2       2015-11-11           26455799            2015 Nov 3            \n",
       "3       2019-10-23           31604990           2019 Oct 11            \n",
       "4       2007-02-20           17347647            2007 Apr 4            \n",
       "\n",
       "                                    Ligand Sequences  \\\n",
       "0  AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...   \n",
       "1  AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...   \n",
       "2  AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...   \n",
       "3  AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...   \n",
       "4  AAKEGWLHFRPLVTDKGKRVGGSIRPWKQMYVVLRGTTPSEEEHSL...   \n",
       "\n",
       "                                  Receptor Sequences  \\\n",
       "0  HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...   \n",
       "1  FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...   \n",
       "2  LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...   \n",
       "3  ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...   \n",
       "4  GSMRILMVGLDAAGKTTILYKLKLGEIVTTIPTIGFNVETVEYKNI...   \n",
       "\n",
       "                                              ligand  \\\n",
       "0  AAAQYPVVNTNYGKIRGLRTPLPNEILGPVEQYLGVPYASPPTGER...   \n",
       "1  AADWDVYCSQDESIPAKFISRLVTSKDQALEKTEINCSNGLVPITQ...   \n",
       "2  AAEEEDEVEWVVESIAGFLRGPDWSIPILDFVEQKCEVFDDEEESK...   \n",
       "3  AAILGDEYLWSGGVIPYTFAGVSGADQSAILSGMQELEEKTCIRFV...   \n",
       "4  AAKEGWLHFRPLVTDKGKRVGGSIRPWKQMYVVLRGTTPSEEEHSL...   \n",
       "\n",
       "                                            receptor fasta_id  \n",
       "0  HAGTTYIFSKGGGQITYKWPPNDRPSTRADRLAIGFSTVQKEAVLV...   2WQZ_0  \n",
       "1  FLNKAFEVALKVQIIAGFDRGLVKWLRVHGRTLSTVQKKALYFVNR...   5IIA_1  \n",
       "2  LLSILRKLKSAPDQEVRILLLGLDNAGKTTLLKQLASEDISHITPT...   4ZI2_2  \n",
       "3  ALNPSALLSRGCNDSDVLAVAGFALRDINKDRKDGYVLRLNRVNDA...   6SAZ_3  \n",
       "4  GSMRILMVGLDAAGKTTILYKLKLGEIVTTIPTIGFNVETVEYKNI...   2J59_4  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fasta_id will be a unique identifier for each row to be used in fasta files\n",
    "# instead of complex id as it may contain spaces\n",
    "df[\"fasta_id\"] = df[\"PDB\"] + \"_\" + df.index.astype(str)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.makedirs(\"data_files\", exist_ok=True)\n",
    "with open(\"data_files/all_seqs.fasta\", \"w\") as f:\n",
    "    for i, row in df.iterrows():\n",
    "        fasta_id = row[\"fasta_id\"]\n",
    "        sequence = row[\"combined\"]\n",
    "        f.write(f\">{fasta_id}\\n{sequence}\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We cluster all the sequnces into as many clusters as it takes such that for each cluster the minimum sequence identity between the representative of the cluster and any other sequence in the cluster is at least 30%. We use default `easy-cluster` coverage parameters of both query and key sequences with minimum coverage of 80%. We also use the `connected component` algorithm to cluster the sequences to ensure that a single cluster can cover more remote homologs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data_files/all_seqs.fasta data_files/cluster tmp --min-seq-id 0.3 --remove-tmp-files --cluster-mode 1 --single-step-clustering \n",
      "\n",
      "MMseqs Version:                     \ta2815df9a6c6da173589fb65b3f71639ea08336d\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Mask lower letter repeating N times \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t128\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.3\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t1\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \ttrue\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Database type                       \t0\n",
      "Shuffle input database              \ttrue\n",
      "Createdb mode                       \t1\n",
      "Write lookup file                   \t0\n",
      "Offset of numeric ids               \t0\n",
      "\n",
      "createdb data_files/all_seqs.fasta tmp/1471709256671452136/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mShuffle database cannot be combined with --createdb-mode 0\n",
      "\u001b[39m\u001b[33mWe recompute with --shuffle 0\n",
      "\u001b[39mConverting sequences\n",
      "[8182] 0s 2ms\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\n",
      "Time for merging to input: 0h 0m 0s 0ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "Create directory tmp/1471709256671452136/clu_tmp\n",
      "cluster tmp/1471709256671452136/input tmp/1471709256671452136/clu tmp/1471709256671452136/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.3 --cluster-mode 1 --single-step-clustering 1 --remove-tmp-files 0 \n",
      "\n",
      "Set cluster sensitivity to -s 5.000000\n",
      "Set cluster iterations to 3\n",
      "clusthash tmp/1471709256671452136/input tmp/1471709256671452136/clu_tmp/16603263552163503201/aln_redundancy --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:3,nucl:5 --min-seq-id 0.99 --max-seq-len 65535 --db-load-mode 0 --threads 128 --compressed 0 -v 3 \n",
      "\n",
      "Reduced amino acid alphabet: (A B C D E G K N P Q R S T Z) (F H I J L M V W Y) (X) \n",
      "Hashing sequences...\n",
      "[=================================================================] 100.00% 8.21K 0s 19ms     \n",
      "Found 4532 unique hashes\n",
      "Time for merging to aln_redundancy: 0h 0m 0s 7ms==================] 155.11% 12.73K eta 0s       ======] 154.14% 12.65K eta 0s       \n",
      "Time for processing: 0h 0m 0s 99ms\n",
      "clust tmp/1471709256671452136/input tmp/1471709256671452136/clu_tmp/16603263552163503201/aln_redundancy tmp/1471709256671452136/clu_tmp/16603263552163503201/clu_redundancy --cluster-mode 1 --max-iterations 1000 --similarity-type 2 --threads 128 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Connected Component\n",
      "[=================================================================] 100.00% 8.21K 0s 10ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3389 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 8.21K 0s 14ms     \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 8.21K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 64ms\n",
      "connected component mode\n",
      "Total time: 0h 0m 0s 87ms\n",
      "\n",
      "Size of the sequence database: 8207\n",
      "Size of the alignment database: 8207\n",
      "Number of clusters: 4818\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 91ms\n",
      "createsubdb tmp/1471709256671452136/clu_tmp/16603263552163503201/clu_redundancy tmp/1471709256671452136/input tmp/1471709256671452136/clu_tmp/16603263552163503201/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/1471709256671452136/clu_tmp/16603263552163503201/input_step_redundancy tmp/1471709256671452136/clu_tmp/16603263552163503201/input_step_redundancy tmp/1471709256671452136/clu_tmp/16603263552163503201/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 128 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 4818 type: Aminoacid\n",
      "Estimated memory consumption: 1G\n",
      "Target database size: 4818 type: Aminoacid\n",
      "Index table k-mer threshold: 118 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 4.82K 0s 34ms     \n",
      "Index table: Masked residues: 64413\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index table: fill\n",
      "[=================================================================] 100.00% 4.82K 0s 26ms     \n",
      "Index statistics\n",
      "Entries:          2184101\n",
      "DB size:          500 MB\n",
      "Avg k-mer size:   0.034127\n",
      "Top 10 k-mers\n",
      "    VHFAQS\t1149\n",
      "    VKYPVT\t1139\n",
      "    WNGLGV\t1114\n",
      "    LYLSTV\t1072\n",
      "    SVCLFY\t934\n",
      "    STSSLT\t915\n",
      "    TVWSLT\t854\n",
      "    AAGLDY\t851\n",
      "    YFEVSW\t847\n",
      "    PALSLY\t847\n",
      "Time for index table init: 0h 0m 0s 424ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 118\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 4818\n",
      "Target db start 1 to 4818\n",
      "[=================================================================] 100.00% 4.82K 0s 295ms    \n",
      "\n",
      "204.596370 k-mers per position\n",
      "62098 DB matches per sequence\n",
      "0 overflows\n",
      "18 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 6ms\n",
      "Time for processing: 0h 0m 0s 961ms\n",
      "align tmp/1471709256671452136/clu_tmp/16603263552163503201/input_step_redundancy tmp/1471709256671452136/clu_tmp/16603263552163503201/input_step_redundancy tmp/1471709256671452136/clu_tmp/16603263552163503201/pref tmp/1471709256671452136/clu_tmp/16603263552163503201/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 128 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 4818 type: Aminoacid\n",
      "Target database size: 4818 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 4.82K 0s 279ms    \n",
      "Time for merging to aln: 0h 0m 0s 7ms\n",
      "57749 alignments calculated\n",
      "36460 sequence pairs passed the thresholds (0.631353 of overall calculated)\n",
      "7.567455 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 360ms\n",
      "clust tmp/1471709256671452136/clu_tmp/16603263552163503201/input_step_redundancy tmp/1471709256671452136/clu_tmp/16603263552163503201/aln tmp/1471709256671452136/clu_tmp/16603263552163503201/clu_step0 --cluster-mode 1 --max-iterations 1000 --similarity-type 2 --threads 128 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Connected Component\n",
      "[=================================================================] 100.00% 4.82K 0s 14ms     \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 7428 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 4.82K 0s 18ms     =======================>                           ] 57.63% 2.78K eta 0s       \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 4.82K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 75ms\n",
      "connected component mode\n",
      "Total time: 0h 0m 0s 94ms\n",
      "\n",
      "Size of the sequence database: 4818\n",
      "Size of the alignment database: 4818\n",
      "Number of clusters: 1572\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 97ms\n",
      "mergeclusters tmp/1471709256671452136/input tmp/1471709256671452136/clu tmp/1471709256671452136/clu_tmp/16603263552163503201/clu_redundancy tmp/1471709256671452136/clu_tmp/16603263552163503201/clu_step0 --threads 128 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 4.82K 0s 10ms     \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 1.57K 0s 32ms     \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 8.21K 0s 95ms     \n",
      "Time for merging to clu: 0h 0m 0s 8ms\n",
      "Time for processing: 0h 0m 0s 113ms\n",
      "createtsv tmp/1471709256671452136/input tmp/1471709256671452136/input tmp/1471709256671452136/clu tmp/1471709256671452136/cluster.tsv --threads 128 -v 3 \n",
      "\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 9ms\n",
      "Time for processing: 0h 0m 0s 88ms\n",
      "result2repseq tmp/1471709256671452136/input tmp/1471709256671452136/clu tmp/1471709256671452136/clu_rep --db-load-mode 0 --compressed 0 --threads 128 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 1.57K 0s 10ms     \n",
      "Time for merging to clu_rep: 0h 0m 0s 9ms\n",
      "Time for processing: 0h 0m 0s 97ms\n",
      "result2flat tmp/1471709256671452136/input tmp/1471709256671452136/input tmp/1471709256671452136/clu_rep tmp/1471709256671452136/rep_seq.fasta --use-fasta-header -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 4ms\n",
      "createseqfiledb tmp/1471709256671452136/input tmp/1471709256671452136/clu tmp/1471709256671452136/clu_seqs --threads 128 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 1.57K 0s 14ms     \n",
      "Time for merging to clu_seqs: 0h 0m 0s 4ms\n",
      "Time for processing: 0h 0m 0s 97ms\n",
      "result2flat tmp/1471709256671452136/input tmp/1471709256671452136/input tmp/1471709256671452136/clu_seqs tmp/1471709256671452136/all_seqs.fasta -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 14ms\n"
     ]
    }
   ],
   "source": [
    "!mmseqs easy-cluster data_files/all_seqs.fasta data_files/cluster tmp --min-seq-id 0.3 --remove-tmp-files --cluster-mode 1 --single-step-clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_rep_id</th>\n",
       "      <th>fasta_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5IIA_1</td>\n",
       "      <td>5IIA_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6SAZ_3</td>\n",
       "      <td>6SAZ_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2J59_6</td>\n",
       "      <td>2J59_6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2J59_6</td>\n",
       "      <td>2J59_4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2J59_6</td>\n",
       "      <td>2J59_7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8202</th>\n",
       "      <td>4IDJ_6476</td>\n",
       "      <td>4IDJ_6476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8203</th>\n",
       "      <td>3FKU_6477</td>\n",
       "      <td>3FKU_6477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8204</th>\n",
       "      <td>3FKU_6477</td>\n",
       "      <td>3FKU_6479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8205</th>\n",
       "      <td>3FKU_6477</td>\n",
       "      <td>3FKU_6480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8206</th>\n",
       "      <td>3FKU_6477</td>\n",
       "      <td>3FKU_6482</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8207 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     cluster_rep_id   fasta_id\n",
       "0            5IIA_1     5IIA_1\n",
       "1            6SAZ_3     6SAZ_3\n",
       "2            2J59_6     2J59_6\n",
       "3            2J59_6     2J59_4\n",
       "4            2J59_6     2J59_7\n",
       "...             ...        ...\n",
       "8202      4IDJ_6476  4IDJ_6476\n",
       "8203      3FKU_6477  3FKU_6477\n",
       "8204      3FKU_6477  3FKU_6479\n",
       "8205      3FKU_6477  3FKU_6480\n",
       "8206      3FKU_6477  3FKU_6482\n",
       "\n",
       "[8207 rows x 2 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_df = pd.read_csv(\n",
    "    \"data_files/cluster_cluster.tsv\", sep=\"\\t\", header=None\n",
    ")\n",
    "cluster_df = cluster_df.rename(columns={0: \"cluster_rep_id\", 1: \"fasta_id\"})\n",
    "cluster_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_rep_id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1R0R_374</th>\n",
       "      <td>311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3SGB_3801</th>\n",
       "      <td>298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1DEE_6384</th>\n",
       "      <td>276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1A22_2752</th>\n",
       "      <td>254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1PPF_4211</th>\n",
       "      <td>247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6PLK_8036</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6PPG_2463</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6QB3_6603</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6QB6_6602</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6R2S_6837</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1572 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                cluster_size\n",
       "cluster_rep_id              \n",
       "1R0R_374                 311\n",
       "3SGB_3801                298\n",
       "1DEE_6384                276\n",
       "1A22_2752                254\n",
       "1PPF_4211                247\n",
       "...                      ...\n",
       "6PLK_8036                  1\n",
       "6PPG_2463                  1\n",
       "6QB3_6603                  1\n",
       "6QB6_6602                  1\n",
       "6R2S_6837                  1\n",
       "\n",
       "[1572 rows x 1 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_df_count = cluster_df.groupby([\"cluster_rep_id\"]).count().sort_values(\n",
    "    \"fasta_id\", ascending=False\n",
    ").rename(columns={\"fasta_id\": \"cluster_size\"})\n",
    "cluster_df_count"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We construct the training set by assigning the largest clusters first to the training set until the training set lengh is approx. 75% of the total number of sequences. We don't want to reach 80% right now, because later we will perform several iterations to transfer some sequences from test to train set if they are too similar to the training set. Also, we found that some entries in the data may have the same PDB but the ligand and receptor chains are swapped. Such entries are not guaranteed to be clustered together, but they must not end in two different sets. Therefore, for each cluster member when we assign it to the training set, we also assign all other members of the same PDB to the training set. This way we ensure that all entries with the same PDB are in the same set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "split\n",
       "train    0.75\n",
       "test     0.25\n",
       "Name: count, dtype: float64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_fasta_ids = set()\n",
    "target_train_count = round(0.75 * len(df))\n",
    "train_clusters = set()\n",
    "for cluster_rep_id, cluster_size in cluster_df_count.iterrows():\n",
    "    # add largest cluster first to the training set\n",
    "    train_clusters.add(cluster_rep_id)\n",
    "    cluster_members = cluster_df[\n",
    "        cluster_df[\"cluster_rep_id\"] == cluster_rep_id\n",
    "    ][\"fasta_id\"].tolist()\n",
    "    for member in cluster_members:\n",
    "        if member in train_fasta_ids:\n",
    "            continue\n",
    "        train_fasta_ids.add(member)\n",
    "        member_pdb = df[df[\"fasta_id\"] == member][\"PDB\"].values[0]\n",
    "        [\n",
    "            train_fasta_ids.add(fasta_id)\n",
    "            for fasta_id in df[df[\"PDB\"] == member_pdb][\"fasta_id\"].tolist()\n",
    "        ]\n",
    "    if len(train_fasta_ids) >= target_train_count:\n",
    "        break\n",
    "\n",
    "df[\"split\"] = df[\"fasta_id\"].apply(\n",
    "    lambda x: \"train\" if x in train_fasta_ids else \"test\"\n",
    ")\n",
    "(df[\"split\"].value_counts() / len(df)).round(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next we perform as many iterations as needed to transfer sequences from the test set to the training set if they are too similar to the training set. In each iteration we do the following:\n",
    "1. Calculate the sequence identity between the training and test set.\n",
    "2. Transfer all sequences in test set that have sequence identity of at least 30% with any sequence in the training set to the training set along with all other sequences from the same PDB (for the same reason described above).\n",
    "3. Repeat until no more sequences are transferred.\n",
    "\n",
    "Finally, we remove all sequences that are too similar to the training set from the test set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "from subprocess import run\n",
    "\n",
    "\n",
    "def do_split_iteration(df, splits):\n",
    "    split1, split2 = splits\n",
    "    with open(f\"data_files/{split1}.fasta\", \"w\") as f:\n",
    "        for _, row in df[df[\"split\"] == split1].iterrows():\n",
    "            fasta_id = row[\"fasta_id\"]\n",
    "            seq = row[\"combined\"]\n",
    "            f.write(f\">{fasta_id}\\n{seq}\\n\")\n",
    "\n",
    "    with open(f\"data_files/{split2}.fasta\", \"w\") as f:\n",
    "        for _, row in df[df[\"split\"] == split2].iterrows():\n",
    "            fasta_id = row[\"fasta_id\"]\n",
    "            seq = row[\"combined\"]\n",
    "            f.write(f\">{fasta_id}\\n{seq}\\n\")\n",
    "\n",
    "    command = [\n",
    "        \"mmseqs\",\n",
    "        \"easy-search\",\n",
    "        f\"data_files/{split1}.fasta\",\n",
    "        f\"data_files/{split2}.fasta\",\n",
    "        \"data_files/matches.tsv\",\n",
    "        \"tmp\",\n",
    "        \"--min-seq-id\",\n",
    "        \"0.3\",\n",
    "        \"-c\",\n",
    "        \"0.8\",\n",
    "    ]\n",
    "\n",
    "    result = run(command, capture_output=True, text=True)\n",
    "    if result.returncode != 0:\n",
    "        print(result.stderr)\n",
    "        raise ValueError(\"Error running mmseqs\")\n",
    "    try:\n",
    "        matches_df = pd.read_csv(\n",
    "            \"data_files/matches.tsv\", sep=\"\\t\", header=None\n",
    "        )\n",
    "    except pd.errors.EmptyDataError:\n",
    "        print(\"No matches found\")\n",
    "        return df, 0\n",
    "    matches_df = matches_df.rename(\n",
    "        columns={0: f\"{split1}_id\", 1: f\"{split2}_id\", 2: \"pident\"}\n",
    "    )\n",
    "    fasta_ids_to_change_split = set(matches_df[f\"{split2}_id\"].unique())\n",
    "    pdbs_to_change_split = df[\n",
    "        df[\"fasta_id\"].isin(fasta_ids_to_change_split)\n",
    "    ][\"PDB\"].unique()\n",
    "    for pdb in pdbs_to_change_split.tolist():\n",
    "        fasta_ids_to_change_split.update(\n",
    "            df[df[\"PDB\"] == pdb][\"fasta_id\"].tolist()\n",
    "        )\n",
    "    print(\n",
    "        f\"Changing {len(fasta_ids_to_change_split)} sequences from {split2} \"\n",
    "        f\"to {split1}\"\n",
    "    )\n",
    "    df[\"split\"] = df.apply(\n",
    "        lambda row: (\n",
    "            split1\n",
    "            if row[\"fasta_id\"] in fasta_ids_to_change_split\n",
    "            else row[\"split\"]\n",
    "        ),\n",
    "        axis=1,\n",
    "    )\n",
    "    return df, len(fasta_ids_to_change_split)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Changing 273 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.78\n",
      "test     0.22\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 47 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.79\n",
      "test     0.21\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "Changing 8 sequences from test to train\n",
      "Current split ratios: split\n",
      "train    0.79\n",
      "test     0.21\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n",
      "No matches found\n",
      "Current split ratios: split\n",
      "train    0.79\n",
      "test     0.21\n",
      "Name: count, dtype: float64\n",
      "********************************************************************************\n"
     ]
    }
   ],
   "source": [
    "while True:\n",
    "    df, total_changed = do_split_iteration(df, (\"train\", \"test\"))\n",
    "    print(\n",
    "        f\"Current split ratios: {(df['split'].value_counts() / len(df)).round(2)}\"\n",
    "    )\n",
    "    print(\"*\" * 80)\n",
    "    if (df[\"split\"] == \"test\").sum() == 0 or total_changed == 0:\n",
    "        break"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now the train and test sets are properly separated. We now need to split the test set into two approximately equal parts: validation and test set. We cluster the test set using the same parameters as before and start assigning the largest cluster to the test set and the second largest to the validation set and so on untill all clusters are assigned. We also ensure that all entries with the same PDB are in the same set as before."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"data_files/test_seqs.fasta\", \"w\") as f:\n",
    "    for i, row in df[df[\"split\"] == \"test\"].iterrows():\n",
    "        fasta_id = row[\"fasta_id\"]\n",
    "        sequence = row[\"combined\"]\n",
    "        f.write(f\">{fasta_id}\\n{sequence}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data_files/test_seqs.fasta data_files/cluster tmp --min-seq-id 0.3 --remove-tmp-files --cluster-mode 1 --single-step-clustering \n",
      "\n",
      "MMseqs Version:                     \ta2815df9a6c6da173589fb65b3f71639ea08336d\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Mask lower letter repeating N times \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t128\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.3\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t1\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \ttrue\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \tfalse\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Database type                       \t0\n",
      "Shuffle input database              \ttrue\n",
      "Createdb mode                       \t1\n",
      "Write lookup file                   \t0\n",
      "Offset of numeric ids               \t0\n",
      "\n",
      "createdb data_files/test_seqs.fasta tmp/17467532586964142270/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mShuffle database cannot be combined with --createdb-mode 0\n",
      "\u001b[39m\u001b[33mWe recompute with --shuffle 0\n",
      "\u001b[39mConverting sequences\n",
      "[1718] 0s 0ms\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\n",
      "Time for merging to input: 0h 0m 0s 0ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Create directory tmp/17467532586964142270/clu_tmp\n",
      "cluster tmp/17467532586964142270/input tmp/17467532586964142270/clu tmp/17467532586964142270/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.3 --cluster-mode 1 --single-step-clustering 1 --remove-tmp-files 0 \n",
      "\n",
      "Set cluster sensitivity to -s 5.000000\n",
      "Set cluster iterations to 3\n",
      "clusthash tmp/17467532586964142270/input tmp/17467532586964142270/clu_tmp/16210148996865129276/aln_redundancy --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:3,nucl:5 --min-seq-id 0.99 --max-seq-len 65535 --db-load-mode 0 --threads 128 --compressed 0 -v 3 \n",
      "\n",
      "Reduced amino acid alphabet: (A B C D E G K N P Q R S T Z) (F H I J L M V W Y) (X) \n",
      "Hashing sequences...\n",
      "[=================================================================] 100.00% 1.72K 0s 7ms      \n",
      "Found 1665 unique hashes\n",
      "Time for merging to aln_redundancy: 0h 0m 0s 6ms==================] 196.51% 3.38K eta 0s       \n",
      "Time for processing: 0h 0m 0s 83ms\n",
      "clust tmp/17467532586964142270/input tmp/17467532586964142270/clu_tmp/16210148996865129276/aln_redundancy tmp/17467532586964142270/clu_tmp/16210148996865129276/clu_redundancy --cluster-mode 1 --max-iterations 1000 --similarity-type 2 --threads 128 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Connected Component\n",
      "[=================================================================] 100.00% 1.72K 0s 9ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 41 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 1.72K 0s 7ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 1.72K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 60ms\n",
      "connected component mode\n",
      "Total time: 0h 0m 0s 80ms\n",
      "\n",
      "Size of the sequence database: 1722\n",
      "Size of the alignment database: 1722\n",
      "Number of clusters: 1681\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 83ms\n",
      "createsubdb tmp/17467532586964142270/clu_tmp/16210148996865129276/clu_redundancy tmp/17467532586964142270/input tmp/17467532586964142270/clu_tmp/16210148996865129276/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/17467532586964142270/clu_tmp/16210148996865129276/input_step_redundancy tmp/17467532586964142270/clu_tmp/16210148996865129276/input_step_redundancy tmp/17467532586964142270/clu_tmp/16210148996865129276/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 128 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 1681 type: Aminoacid\n",
      "Estimated memory consumption: 993M\n",
      "Target database size: 1681 type: Aminoacid\n",
      "Index table k-mer threshold: 118 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 1.68K 0s 15ms     \n",
      "Index table: Masked residues: 13500\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 1.68K 0s 18ms     \n",
      "Index statistics\n",
      "Entries:          702594\n",
      "DB size:          492 MB\n",
      "Avg k-mer size:   0.010978\n",
      "Top 10 k-mers\n",
      "    GQTVSS\t167\n",
      "    YWQTTV\t165\n",
      "    NTYQSL\t122\n",
      "    QLEGLV\t121\n",
      "    GSRSAS\t117\n",
      "    DYGGVT\t110\n",
      "    DSKRIS\t102\n",
      "    RQPKEW\t98\n",
      "    VHFAQS\t96\n",
      "    VKYPVT\t96\n",
      "Time for index table init: 0h 0m 0s 395ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 118\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 1681\n",
      "Target db start 1 to 1681\n",
      "[=================================================================] 100.00% 1.68K 0s 99ms     \n",
      "\n",
      "201.134259 k-mers per position\n",
      "4767 DB matches per sequence\n",
      "0 overflows\n",
      "14 sequences passed prefiltering per query sequence\n",
      "17 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref: 0h 0m 0s 7ms\n",
      "Time for processing: 0h 0m 0s 707ms\n",
      "align tmp/17467532586964142270/clu_tmp/16210148996865129276/input_step_redundancy tmp/17467532586964142270/clu_tmp/16210148996865129276/input_step_redundancy tmp/17467532586964142270/clu_tmp/16210148996865129276/pref tmp/17467532586964142270/clu_tmp/16210148996865129276/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 128 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 1681 type: Aminoacid\n",
      "Target database size: 1681 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 1.68K 0s 43ms     \n",
      "Time for merging to aln: 0h 0m 0s 7ms\n",
      "9932 alignments calculated\n",
      "3494 sequence pairs passed the thresholds (0.351792 of overall calculated)\n",
      "2.078525 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 115ms\n",
      "clust tmp/17467532586964142270/clu_tmp/16210148996865129276/input_step_redundancy tmp/17467532586964142270/clu_tmp/16210148996865129276/aln tmp/17467532586964142270/clu_tmp/16210148996865129276/clu_step0 --cluster-mode 1 --max-iterations 1000 --similarity-type 2 --threads 128 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Connected Component\n",
      "[=================================================================] 100.00% 1.68K 0s 10ms     .50K eta 0s       \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 73 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 1.68K 0s 9ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 1.68K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 60ms\n",
      "connected component mode\n",
      "Total time: 0h 0m 0s 80ms\n",
      "\n",
      "Size of the sequence database: 1681\n",
      "Size of the alignment database: 1681\n",
      "Number of clusters: 1071\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 83ms\n",
      "mergeclusters tmp/17467532586964142270/input tmp/17467532586964142270/clu tmp/17467532586964142270/clu_tmp/16210148996865129276/clu_redundancy tmp/17467532586964142270/clu_tmp/16210148996865129276/clu_step0 --threads 128 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 1.68K 0s 10ms     \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 1.07K 0s 31ms     \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 1.72K 0s 85ms     \n",
      "Time for merging to clu: 0h 0m 0s 5ms\n",
      "Time for processing: 0h 0m 0s 106ms\n",
      "createtsv tmp/17467532586964142270/input tmp/17467532586964142270/input tmp/17467532586964142270/clu tmp/17467532586964142270/cluster.tsv --threads 128 -v 3 \n",
      "\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 7ms\n",
      "Time for processing: 0h 0m 0s 93ms\n",
      "result2repseq tmp/17467532586964142270/input tmp/17467532586964142270/clu tmp/17467532586964142270/clu_rep --db-load-mode 0 --compressed 0 --threads 128 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 1.07K 0s 11ms     \n",
      "Time for merging to clu_rep: 0h 0m 0s 8ms\n",
      "Time for processing: 0h 0m 0s 83ms\n",
      "result2flat tmp/17467532586964142270/input tmp/17467532586964142270/input tmp/17467532586964142270/clu_rep tmp/17467532586964142270/rep_seq.fasta --use-fasta-header -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "createseqfiledb tmp/17467532586964142270/input tmp/17467532586964142270/clu tmp/17467532586964142270/clu_seqs --threads 128 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 1.07K 0s 6ms      \n",
      "Time for merging to clu_seqs: 0h 0m 0s 5ms\n",
      "Time for processing: 0h 0m 0s 73ms\n",
      "result2flat tmp/17467532586964142270/input tmp/17467532586964142270/input tmp/17467532586964142270/clu_seqs tmp/17467532586964142270/all_seqs.fasta -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 4ms\n"
     ]
    }
   ],
   "source": [
    "!mmseqs easy-cluster data_files/test_seqs.fasta data_files/cluster tmp --min-seq-id 0.3 --remove-tmp-files --cluster-mode 1 --single-step-clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_rep_id</th>\n",
       "      <th>fasta_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2AZE_2521</td>\n",
       "      <td>2AZE_2521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5JHL_2524</td>\n",
       "      <td>5JHL_2524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5D50_2525</td>\n",
       "      <td>5D50_2525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4DXA_2526</td>\n",
       "      <td>4DXA_2526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4NQW_2527</td>\n",
       "      <td>4NQW_2527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1717</th>\n",
       "      <td>6FE4_6240</td>\n",
       "      <td>6FE4_6242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1718</th>\n",
       "      <td>6FE4_6240</td>\n",
       "      <td>6FE4_6243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1719</th>\n",
       "      <td>5TOJ_6246</td>\n",
       "      <td>5TOJ_6246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1720</th>\n",
       "      <td>5TOJ_6246</td>\n",
       "      <td>5TOJ_6244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1721</th>\n",
       "      <td>5TOJ_6246</td>\n",
       "      <td>5TOJ_6245</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1722 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     cluster_rep_id   fasta_id\n",
       "0         2AZE_2521  2AZE_2521\n",
       "1         5JHL_2524  5JHL_2524\n",
       "2         5D50_2525  5D50_2525\n",
       "3         4DXA_2526  4DXA_2526\n",
       "4         4NQW_2527  4NQW_2527\n",
       "...             ...        ...\n",
       "1717      6FE4_6240  6FE4_6242\n",
       "1718      6FE4_6240  6FE4_6243\n",
       "1719      5TOJ_6246  5TOJ_6246\n",
       "1720      5TOJ_6246  5TOJ_6244\n",
       "1721      5TOJ_6246  5TOJ_6245\n",
       "\n",
       "[1722 rows x 2 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_df = pd.read_csv(\n",
    "    \"data_files/cluster_cluster.tsv\", sep=\"\\t\", header=None\n",
    ")\n",
    "cluster_df = cluster_df.rename(columns={0: \"cluster_rep_id\", 1: \"fasta_id\"})\n",
    "cluster_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster_size</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster_rep_id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6EWB_6311</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5FV2_2194</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2C7N_5023</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6FE4_6240</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1TH1_5987</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1B6C_3200</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6J14_5441</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6JDJ_896</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6JHW_2954</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6JO8_3176</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1071 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                cluster_size\n",
       "cluster_rep_id              \n",
       "6EWB_6311                  7\n",
       "5FV2_2194                  6\n",
       "2C7N_5023                  6\n",
       "6FE4_6240                  6\n",
       "1TH1_5987                  6\n",
       "...                      ...\n",
       "1B6C_3200                  1\n",
       "6J14_5441                  1\n",
       "6JDJ_896                   1\n",
       "6JHW_2954                  1\n",
       "6JO8_3176                  1\n",
       "\n",
       "[1071 rows x 1 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster_df_count = cluster_df.groupby([\"cluster_rep_id\"]).count().sort_values(\n",
    "    \"fasta_id\", ascending=False\n",
    ").rename(columns={\"fasta_id\": \"cluster_size\"})\n",
    "cluster_df_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "split\n",
       "train    0.79\n",
       "val      0.12\n",
       "test     0.09\n",
       "Name: count, dtype: float64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "val_fasta_ids = set()\n",
    "test_fasta_ids = set(df[df[\"split\"] == \"test\"][\"fasta_id\"].tolist())\n",
    "add_to_val = False\n",
    "for cluster_rep_id, cluster_size in cluster_df_count.iterrows():\n",
    "    cluster_members = cluster_df[\n",
    "        cluster_df[\"cluster_rep_id\"] == cluster_rep_id\n",
    "    ][\"fasta_id\"].tolist()\n",
    "    if add_to_val:\n",
    "        for member in cluster_members:\n",
    "            val_fasta_ids.add(member)\n",
    "            member_pdb = df[df[\"fasta_id\"] == member][\"PDB\"].values[0]\n",
    "            [\n",
    "                val_fasta_ids.add(fasta_id)\n",
    "                for fasta_id in df[df[\"PDB\"] == member_pdb][\n",
    "                    \"fasta_id\"\n",
    "                ].tolist()\n",
    "            ]\n",
    "    add_to_val = not add_to_val\n",
    "test_fasta_ids = test_fasta_ids - val_fasta_ids\n",
    "df[\"split\"] = df.apply(\n",
    "    lambda x: \"val\" if x[\"fasta_id\"] in val_fasta_ids else x[\"split\"], axis=1\n",
    ")\n",
    "df[\"split\"] = df.apply(\n",
    "    lambda x: \"test\" if x[\"fasta_id\"] in test_fasta_ids else x[\"split\"], axis=1\n",
    ")\n",
    "(df[\"split\"].value_counts() / len(df)).round(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['combined', 'Source Data Set', 'Complex ID', 'PDB', 'Mutations',\n",
       "       'Ligand Chains', 'Receptor Chains', 'Ligand Name', 'Receptor Name',\n",
       "       'KD(M)', 'Affinity Method', 'Structure Method', 'Temperature(K)',\n",
       "       'Resolution(Å)', 'PDB PubMed ID', 'PDB Release Date',\n",
       "       'Affinity PubMed ID', 'Affinity Release Date', 'Subgroup',\n",
       "       'Ligand Sequences', 'Receptor Sequences', 'ligand', 'receptor',\n",
       "       'fasta_id', 'split'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.drop(\n",
    "    columns=[\n",
    "        \"ligand\",\n",
    "        \"receptor\",\n",
    "        \"combined\",\n",
    "        \"fasta_id\",\n",
    "    ]\n",
    ")\n",
    "df.to_csv(\"PPB_Affinity_processed_filtered.csv\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
